<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Costa, W. G." />

<meta name="date" content="2025-01-17" />

<title>Genomic Selection - Prepare Data</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomic Selection for Drought Tolerance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="phenotype.html">Phenotype</a>
</li>
<li>
  <a href="mixed_models.html">GxE Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genomic Selection
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="GWS.html">Prepare Data</a>
    </li>
    <li>
      <a href="GWS_RR-BLUP.html">RR-BLUP</a>
    </li>
    <li>
      <a href="GWS_G-BLUP.html">G-BLUP (additive and additive-dominant)</a>
    </li>
    <li>
      <a href="GWS_BayesA.html">Bayes A</a>
    </li>
    <li>
      <a href="GWS_BayesB.html">Bayes B</a>
    </li>
    <li>
      <a href="GWS_RKHS.html">RKHS</a>
    </li>
    <li>
      <a href="GWS_RF.html">Random Forest</a>
    </li>
  </ul>
</li>
<li>
  <a href="clone_selection.html">Clone Selection</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Genomic Selection - Prepare Data</h1>
<h4 class="author">Costa, W. G.<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a></h4>
<h4 class="date">2025-01-17</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-17
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221020code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221020)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221020code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221020)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree75e71b86b709b5628f10f9e8b9d58716cb353e38targetblank75e71b8a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/75e71b86b709b5628f10f9e8b9d58716cb353e38" target="_blank">75e71b8</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree75e71b86b709b5628f10f9e8b9d58716cb353e38targetblank75e71b8a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/75e71b86b709b5628f10f9e8b9d58716cb353e38" target="_blank">75e71b8</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/allchrAR08.txt
    Ignored:    output/BLUPS_density_med_row_col.png
    Ignored:    output/Density_residual_row_col.tiff
    Ignored:    output/Residuals_vs_fitted_row_col.tiff
    Ignored:    output/varcomp_row_col.tiff

Untracked files:
    Untracked:  analysis/GWS_BayesA.Rmd
    Untracked:  analysis/GWS_BayesB.Rmd
    Untracked:  analysis/GWS_G-BLUP.Rmd
    Untracked:  analysis/GWS_RF.Rmd
    Untracked:  analysis/GWS_RKHS.Rmd
    Untracked:  analysis/GWS_RR-BLUP.Rmd
    Untracked:  analysis/clone_selection.Rmd
    Untracked:  analysis/figure/GWS.Rmd/
    Untracked:  data/Artigo/
    Untracked:  data/geno.rds
    Untracked:  data/pheno.rds

Unstaged changes:
    Modified:   analysis/GWS.Rmd
    Modified:   analysis/mixed_models.Rmd
    Modified:   data/SNPs.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/GWS.Rmd</code>) and HTML
(<code>docs/GWS.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/9cd149396ec6b79a9f5417ba8236757ef358790b/analysis/GWS.Rmd" target="_blank">9cd1493</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2025-01-17
</td>
<td>
Update GWS.Rmd
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/286b4922972c00dd9062e8c6a1848a867118378a/analysis/GWS.Rmd" target="_blank">286b492</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
Update Scripts and README
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/286b4922972c00dd9062e8c6a1848a867118378a/docs/GWS.html" target="_blank">286b492</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
Update Scripts and README
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/90dc11276f860bdf4c29fc2cbf7a7d66d049d277/analysis/GWS.Rmd" target="_blank">90dc112</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/6cc4d2326a79681446b1ea653122061a959c2ffe/analysis/GWS.Rmd" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/6cc4d2326a79681446b1ea653122061a959c2ffe/docs/GWS.html" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/1936c513a4d0c67c245b9b4f90722de1fe6f8b16/analysis/GWS.Rmd" target="_blank">1936c51</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/1936c513a4d0c67c245b9b4f90722de1fe6f8b16/docs/GWS.html" target="_blank">1936c51</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/62bf2ac6d936e72f793ae07e32b74331a91405bf/analysis/GWS.Rmd" target="_blank">62bf2ac</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/62bf2ac6d936e72f793ae07e32b74331a91405bf/docs/GWS.html" target="_blank">62bf2ac</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/5988c27353e95ed989eeda03e0ca0091cccc9bf0/analysis/GWS.Rmd" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/analysis/GWS.Rmd" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/GWS.html" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="configurations-and-packages" class="section level2">
<h2>Configurations and packages</h2>
<p>To perform the analyses, we will need the following packages:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#devtools::install_github(&quot;wolfemd/genomicMateSelectR&quot;, ref = &#39;master&#39;) </span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(genomicMateSelectR)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(AGHmatrix)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(ComplexHeatmap)</span></code></pre></div>
<p><br></p>
</div>
<div id="data" class="section level2">
<h2>Data</h2>
<p>The data set is based in genotypes evalueted in five years (2016 to
2020), each year was considered as environment.</p>
<div id="names-marker-data" class="section level3">
<h3>Names marker data</h3>
<p>First let’s get the marker IDs for each clone.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>names <span class="ot">&lt;-</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="fu">read_excel</span>(<span class="st">&quot;data/Phenotyping2.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="st">&quot;GBS&quot;</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Clone =</span> <span class="st">`</span><span class="at">Names trials Petrolina</span><span class="st">`</span>, <span class="at">ID_Clone =</span> <span class="st">`</span><span class="at">Nome GBS</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID_Clone =</span> <span class="fu">str_replace_all</span>(ID_Clone, <span class="st">&quot;:&quot;</span>, <span class="st">&quot;.&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a>  <span class="fu">select</span>(Clone, ID_Clone)</span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>names <span class="sc">|&gt;</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span> </span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">|&gt;</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Clone
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ID_Clone
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
4271
</td>
<td style="text-align:center;">
4271.250494233
</td>
</tr>
<tr>
<td style="text-align:center;">
9624-09
</td>
<td style="text-align:center;">
962409.250370255
</td>
</tr>
<tr>
<td style="text-align:center;">
Aipim Abacate
</td>
<td style="text-align:center;">
ERETA.250437577
</td>
</tr>
<tr>
<td style="text-align:center;">
Alagoana
</td>
<td style="text-align:center;">
Alagoana363.250437472
</td>
</tr>
<tr>
<td style="text-align:center;">
BGM-0004
</td>
<td style="text-align:center;">
CNPMF4.250370278
</td>
</tr>
<tr>
<td style="text-align:center;">
BGM-0019
</td>
<td style="text-align:center;">
CNPMF19.250370327
</td>
</tr>
</tbody>
</table>
</div>
<div id="phenotypic-data" class="section level3">
<h3>Phenotypic data</h3>
<p>Now let’s group the marker IDs with the clone names.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;output/BLUPS_row_col_random.csv&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Clone =</span> <span class="fu">str_split_i</span>(Clone, <span class="st">&quot;[.]&quot;</span>, <span class="sc">-</span><span class="dv">1</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>  <span class="fu">inner_join</span>(names) <span class="sc">|&gt;</span> <span class="co"># Join Phenotypic with Genotypic datas</span></span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Clone =</span> <span class="fu">factor</span>(Clone), <span class="at">ID_Clone =</span> <span class="fu">factor</span>(ID_Clone))</span></code></pre></div>
<pre><code>Joining with `by = join_by(Clone)`</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>pheno <span class="sc">|&gt;</span> </span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">|&gt;</span></span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">|&gt;</span></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
trait
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Clone
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
BLUP
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
BLUPS_mean
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
drgBLUP_mean
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ID_Clone
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
4271
</td>
<td style="text-align:center;">
-0.8918756
</td>
<td style="text-align:center;">
3.401067
</td>
<td style="text-align:center;">
2.715979
</td>
<td style="text-align:center;">
4271.250494233
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
9624-09
</td>
<td style="text-align:center;">
1.8104038
</td>
<td style="text-align:center;">
6.103346
</td>
<td style="text-align:center;">
9.106536
</td>
<td style="text-align:center;">
962409.250370255
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
Aipim Abacate
</td>
<td style="text-align:center;">
1.5033015
</td>
<td style="text-align:center;">
5.796244
</td>
<td style="text-align:center;">
6.952723
</td>
<td style="text-align:center;">
ERETA.250437577
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
Alagoana
</td>
<td style="text-align:center;">
-0.3341086
</td>
<td style="text-align:center;">
3.958834
</td>
<td style="text-align:center;">
3.703014
</td>
<td style="text-align:center;">
Alagoana363.250437472
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
BGM-0004
</td>
<td style="text-align:center;">
-0.6240384
</td>
<td style="text-align:center;">
3.668904
</td>
<td style="text-align:center;">
3.190100
</td>
<td style="text-align:center;">
CNPMF4.250370278
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
BGM-0019
</td>
<td style="text-align:center;">
-1.6545523
</td>
<td style="text-align:center;">
2.638390
</td>
<td style="text-align:center;">
1.201559
</td>
<td style="text-align:center;">
CNPMF19.250370327
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">levels</span>(<span class="fu">factor</span>(pheno<span class="sc">$</span>trait))</span></code></pre></div>
</div>
<div id="genotypic-data" class="section level3">
<h3>Genotypic data</h3>
<p>Now let’s load the genotypic data from the GBS markers and correct
the base pair values. Furthermore, we will also divide the alleles
column into two columns, for the reference allele and the recessive
allele. And we will select the columns with the names of the markers,
reference alleles and the columns with the clone IDs according to the
BLUPs data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/allchrAR08.txt&quot;</span>, <span class="at">header =</span> T) <span class="sc">|&gt;</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="dv">12</span><span class="sc">:</span><span class="dv">3365</span>, <span class="sc">~</span>{</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>    <span class="fu">case_when</span>(</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;A&quot;</span> <span class="sc">~</span> <span class="st">&quot;AA&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;R&quot;</span> <span class="sc">~</span> <span class="st">&quot;AG&quot;</span>,</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;W&quot;</span> <span class="sc">~</span> <span class="st">&quot;AT&quot;</span>,</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;M&quot;</span> <span class="sc">~</span> <span class="st">&quot;AC&quot;</span>,</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;C&quot;</span> <span class="sc">~</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;S&quot;</span> <span class="sc">~</span> <span class="st">&quot;CG&quot;</span>,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;Y&quot;</span> <span class="sc">~</span> <span class="st">&quot;CT&quot;</span>,</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;G&quot;</span> <span class="sc">~</span> <span class="st">&quot;GG&quot;</span>,</span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;K&quot;</span> <span class="sc">~</span> <span class="st">&quot;GT&quot;</span>,</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;T&quot;</span> <span class="sc">~</span> <span class="st">&quot;TT&quot;</span>)</span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>  })) <span class="sc">|&gt;</span></span>
<span id="cb7-15"><a href="#cb7-15" tabindex="-1"></a>  <span class="fu">separate</span>(alleles, <span class="fu">c</span>(<span class="st">&quot;reference&quot;</span>, <span class="st">&quot;recess&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb7-16"><a href="#cb7-16" tabindex="-1"></a>  <span class="fu">select</span>(rs, reference, recess, <span class="fu">any_of</span>(pheno<span class="sc">$</span>ID_Clone))</span>
<span id="cb7-17"><a href="#cb7-17" tabindex="-1"></a></span>
<span id="cb7-18"><a href="#cb7-18" tabindex="-1"></a>geno[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>] <span class="sc">|&gt;</span></span>
<span id="cb7-19"><a href="#cb7-19" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">|&gt;</span></span>
<span id="cb7-20"><a href="#cb7-20" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
rs
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
reference
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
recess
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ERETA.250437577
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Alagoana363.250437472
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF4.250370278
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF19.250370327
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF30.250370293
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF32.250370305
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF36.250370328
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF46.250370283
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF48.250370295
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF56.250370248
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF65.250370285
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF66.250370296
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF74.250370321
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF80.250370250
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF84.250370263
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF89.250370287
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF91.250370299
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
S1_84637
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
G
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AG
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_84843
</td>
<td style="text-align:center;">
C
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
AC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_126260
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
C
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_126261
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
G
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_126264
</td>
<td style="text-align:center;">
T
</td>
<td style="text-align:center;">
G
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
</tr>
</tbody>
</table>
<p>Now we need to do the base pair conversion for allelic dosage
according to the reference allele. I will also add the rs column as the
column name. Then I will exclude the reference and recess allele
columns. To convert into the format to perform GWS analyses, we have to
transpose the marker matrix.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno <span class="sc">|&gt;</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(geno), <span class="sc">~</span>{</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>      <span class="fu">case_when</span>(</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>        . <span class="sc">==</span> <span class="fu">paste</span>(reference, reference, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>        . <span class="sc">==</span> <span class="fu">paste</span>(recess, recess, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="dv">1</span></span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>      )</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a>  })) <span class="sc">|&gt;</span></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(reference, recess)) <span class="sc">|&gt;</span> </span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="at">var =</span><span class="st">&quot;rs&quot;</span>) <span class="sc">|&gt;</span> </span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  </span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>geno[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="sc">|&gt;</span></span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">|&gt;</span></span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_84637
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_84843
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_126260
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_126261
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_126264
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ERETA.250437577
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Alagoana363.250437472
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
CNPMF4.250370278
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
CNPMF19.250370327
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
CNPMF30.250370293
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
</tbody>
</table>
<p>Let’s check how many clones present data genotyped with the
markers.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>geno <span class="sc">|&gt;</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">dim</span>() <span class="sc">|&gt;</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">|&gt;</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>,</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>      <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Number of Clones&quot;</span>, <span class="st">&quot;Number of markers&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of Clones
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of markers
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
415
</td>
<td style="text-align:center;">
27045
</td>
</tr>
</tbody>
</table>
<p>By filtering common genotypes, we have 415 genotypes and 27045
brands.</p>
<p>Now let’s filter the SNPS using MAF of 0.01 and check how many
markers will remain.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">maf_filter</span>(geno, <span class="at">thresh =</span> <span class="fl">0.01</span>)</span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>geno <span class="sc">|&gt;</span></span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>  <span class="fu">dim</span>() <span class="sc">|&gt;</span></span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">|&gt;</span></span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>,</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a>      <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Number of Clones&quot;</span>, <span class="st">&quot;Number of markers&quot;</span>)) <span class="sc">|&gt;</span></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of Clones
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of markers
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
415
</td>
<td style="text-align:center;">
22779
</td>
</tr>
</tbody>
</table>
<p>With the MAF filter at 1%, 22779 markers remained. I’m going to save
the matrix now so we can load it if necessary.</p>
</div>
</div>
<div id="genomic-selection" class="section level2">
<h2>Genomic selection</h2>
<p>For this purpose, we will use only individuals with BLUps and SNPs
available.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">|&gt;</span> </span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="fu">select</span>(ID_Clone, trait, BLUP) <span class="sc">%&gt;%</span> </span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> trait, <span class="at">values_from =</span> BLUP, <span class="at">id_cols =</span> <span class="dv">1</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  <span class="fu">filter</span>(ID_Clone <span class="sc">%in%</span> <span class="fu">rownames</span>(geno)) <span class="sc">|&gt;</span></span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>  <span class="fu">droplevels</span>()</span>
<span id="cb11-6"><a href="#cb11-6" tabindex="-1"></a></span>
<span id="cb11-7"><a href="#cb11-7" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb11-8"><a href="#cb11-8" tabindex="-1"></a></span>
<span id="cb11-9"><a href="#cb11-9" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno[<span class="fu">order</span>(pheno<span class="sc">$</span>ID_Clone, <span class="at">decreasing =</span> F),]</span>
<span id="cb11-10"><a href="#cb11-10" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno[<span class="fu">order</span>(<span class="fu">row.names</span>(geno)),]</span>
<span id="cb11-11"><a href="#cb11-11" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(geno) <span class="sc">==</span> pheno<span class="sc">$</span>ID_Clone)</span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
<div id="building-the-g-matrix" class="section level3">
<h3>Building the G matrix</h3>
<p>Again, we will use the <a
href="https://cran.r-project.org/web/packages/AGHmatrix/vignettes/Tutorial_AGHmatrix.html">AGHmatrix</a>
package <span class="citation">[@amadeu_aghmatrix_2016]</span> to build
the G matrix:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a>G_matrix <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(geno,</span>
<span id="cb13-2"><a href="#cb13-2" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;VanRaden&quot;</span>,</span>
<span id="cb13-3"><a href="#cb13-3" tabindex="-1"></a>                    <span class="at">ploidy =</span> <span class="dv">2</span>,</span>
<span id="cb13-4"><a href="#cb13-4" tabindex="-1"></a>                    <span class="at">missingValue =</span> <span class="cn">NA</span>)</span></code></pre></div>
<pre><code>Initial data: 
    Number of Individuals: 415 
    Number of Markers: 22779 

Missing data check: 
    Total SNPs: 22779 
     0 SNPs dropped due to missing data threshold of 0.5 
    Total of: 22779  SNPs 

MAF check: 
    No SNPs with MAF below 0 

Heterozigosity data check: 
    No SNPs with heterozygosity, missing threshold of =  0 

Summary check: 
    Initial:  22779 SNPs 
    Final:  22779  SNPs ( 0  SNPs removed) 
 
Completed! Time = 9.88  seconds </code></pre>
<p>Now we have the whole G matrix (414 x 414), which we can represent
using a heatmap:</p>
<div id="heatmap-of-the-genomic-kinship-matrix-between-clones"
class="section level11">
<p class="heading">Heatmap of the genomic kinship matrix between
clones</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  G_matrix,</span>
<span id="cb15-3"><a href="#cb15-3" tabindex="-1"></a>  <span class="at">show_row_names =</span> F,</span>
<span id="cb15-4"><a href="#cb15-4" tabindex="-1"></a>  <span class="at">show_column_names =</span> F,</span>
<span id="cb15-5"><a href="#cb15-5" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;Res&quot;</span>)</span>
<span id="cb15-6"><a href="#cb15-6" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figure/GWS.Rmd/heatmap-G-matrix-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>“Res” in the heatmap legend title is for “Resemblance”.</p>
<p>Let’s now move on to processing each model. Since this can be
time-consuming, I’ve separated it into parallel processing and each
model into each script:</p>
<ul>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RR-BLUP.html"><strong>RR-BLUP</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_G-BLUP.html"><strong>G-BLUP
(additive and additive-dominant)</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_BayesA.html"><strong>Bayes
A</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_BayesB.html"><strong>Bayes
B</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RKHS.html"><strong>RKHS</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RF.html"><strong>Random
Forest</strong></a></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.3.3 (2024-02-29 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=Portuguese_Brazil.utf8  LC_CTYPE=Portuguese_Brazil.utf8   
[3] LC_MONETARY=Portuguese_Brazil.utf8 LC_NUMERIC=C                      
[5] LC_TIME=Portuguese_Brazil.utf8    

time zone: America/Sao_Paulo
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] ComplexHeatmap_2.16.0    AGHmatrix_2.1.4          genomicMateSelectR_0.2.0
 [4] kableExtra_1.4.0         lubridate_1.9.4          forcats_1.0.0           
 [7] stringr_1.5.1            dplyr_1.1.4              purrr_1.0.2             
[10] readr_2.1.5              tidyr_1.3.1              tibble_3.2.1            
[13] ggplot2_3.5.1            tidyverse_2.0.0          readxl_1.4.3            

loaded via a namespace (and not attached):
 [1] shape_1.4.6.1       circlize_0.4.16     gtable_0.3.6       
 [4] rjson_0.2.23        xfun_0.50           bslib_0.8.0        
 [7] GlobalOptions_0.1.2 lattice_0.22-6      tzdb_0.4.0         
[10] vctrs_0.6.5         tools_4.3.3         generics_0.1.3     
[13] stats4_4.3.3        parallel_4.3.3      cluster_2.1.8      
[16] pkgconfig_2.0.3     Matrix_1.6-1        RColorBrewer_1.1-3 
[19] S4Vectors_0.38.1    lifecycle_1.0.4     compiler_4.3.3     
[22] git2r_0.35.0        munsell_0.5.1       codetools_0.2-20   
[25] clue_0.3-66         httpuv_1.6.15       htmltools_0.5.8.1  
[28] sass_0.4.9          yaml_2.3.10         crayon_1.5.3       
[31] later_1.4.1         pillar_1.10.1       jquerylib_0.1.4    
[34] whisker_0.4.1       cachem_1.1.0        iterators_1.0.14   
[37] foreach_1.5.2       tidyselect_1.2.1    digest_0.6.37      
[40] stringi_1.8.4       rprojroot_2.0.4     fastmap_1.2.0      
[43] colorspace_2.1-1    cli_3.6.3           magrittr_2.0.3     
[46] withr_3.0.2         scales_1.3.0        promises_1.3.2     
[49] timechange_0.3.0    rmarkdown_2.29      matrixStats_1.5.0  
[52] workflowr_1.7.1     cellranger_1.1.0    GetoptLong_1.0.5   
[55] png_0.1-8           zoo_1.8-12          hms_1.1.3          
[58] evaluate_1.0.3      knitr_1.49          IRanges_2.34.1     
[61] doParallel_1.0.17   viridisLite_0.4.2   rlang_1.1.4        
[64] Rcpp_1.0.14         glue_1.8.0          xml2_1.3.6         
[67] BiocGenerics_0.46.0 svglite_2.1.3       rstudioapi_0.17.1  
[70] jsonlite_1.8.9      R6_2.5.1            systemfonts_1.1.0  
[73] fs_1.6.5           </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Weverton Gomes da Costa, Pós-Doutorando, Embrapa
Mandioca e Fruticultura, <a href="mailto:wevertonufv@gmail.com"
class="email">wevertonufv@gmail.com</a><a href="#fnref1"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
