<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Costa, W. G." />

<meta name="date" content="2022-11-10" />

<title>GWS</title>

<script src="site_libs/header-attrs-2.17/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomic Selection for Drought Tolerance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="phenotype.html">Phenotype</a>
</li>
<li>
  <a href="GWS.html">GWS</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">GWS</h1>
<h4 class="author">Costa, W. G.<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a></h4>
<h4 class="date">2022-11-10</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-11-10
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has staged changes. To know which version of the
R Markdown file created these results, you’ll want to first commit it to
the Git repo. If you’re still working on the analysis, you can ignore
this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221020code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221020)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221020code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221020)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatreeb78c842a0b90fe20a5a7c904c9efb5deb23fdcb5targetblankb78c842a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/b78c842a0b90fe20a5a7c904c9efb5deb23fdcb5" target="_blank">b78c842</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatreeb78c842a0b90fe20a5a7c904c9efb5deb23fdcb5targetblankb78c842a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/b78c842a0b90fe20a5a7c904c9efb5deb23fdcb5" target="_blank">b78c842</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/figure/
    Untracked:  data/AllGBSandDArTClones_ReadyForGP_Dos.rds
    Untracked:  data/DCas22_GBSandDArT_ReadyForGP_Dos.rds
    Untracked:  data/allchrAR08.txt
    Untracked:  data/convet_haplo_diplo.txt
    Untracked:  output/BLUPS.RDS
    Untracked:  output/BLUPS.csv
    Untracked:  output/BLUPS_par.Rdata
    Untracked:  output/herdabilidade.csv
    Untracked:  output/media_pheno.csv
    Untracked:  output/resultMM.Rdata

Unstaged changes:
    Deleted:    BLUPS.RDS
    Deleted:    BLUPS.csv
    Deleted:    BLUPs.Rdata
    Deleted:    RR-BLUP-method.pdf
    Modified:   analysis/GWS.Rmd
    Modified:   analysis/phenotype.Rmd
    Deleted:    cor-methods-RR_BLUP.pdf
    Modified:   data/SNPs.rds
    Deleted:    deregress.Rdata
    Deleted:    geno.txt
    Deleted:    geno2.txt
    Deleted:    herdabilidade.csv

Staged changes:
    New:        BLUPS.RDS
    New:        BLUPS.csv
    New:        BLUPs.Rdata
    New:        RR-BLUP-method.pdf
    New:        analysis/GWS.Rmd
    Modified:   analysis/_site.yml
    Modified:   analysis/about.Rmd
    Modified:   analysis/index.Rmd
    Modified:   analysis/license.Rmd
    New:        analysis/phenotype.Rmd
    New:        cor-methods-RR_BLUP.pdf
    New:        data/Fenótipos Desregressados GBS Todos.RDS
    New:        data/Genomic Selection for Drought Tolerance Using Genome-Wide SNPs in Maize.pdf
    New:        data/Phenotyping v2.xlsx
    New:        data/Phenotyping.xlsx
    New:        data/SNPs.rds
    New:        data/deregress.Rdata
    New:        data/geno.rds
    New:        data/geno.txt
    New:        deregress.Rdata
    New:        geno.txt
    New:        geno2.txt
    New:        herdabilidade.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with
<code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<p>Yesterday, we learned how to perform genomic selection in a single
environment trial. Today, we will expand the models to a
multiple-environment context.</p>
<p>To perform the analyses, we will need the following packages:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(janitor)</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(genomicMateSelectR)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(AGHmatrix)</span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(ComplexHeatmap)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">require</span>(rrBLUP)</span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpmisc)</span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cvTools)</span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span></code></pre></div>
<p><br></p>
<div id="data" class="section level2">
<h2>Data</h2>
<p>The data set is based in genotypes evalueted in five years (2016 to
2020), each year was considered as environment.</p>
<div id="names-marker-data" class="section level3">
<h3>Names marker data</h3>
<p>Primeiro vamos buscar os ID dos marcadores para cada clone.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>names <span class="ot">&lt;-</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">read_excel</span>(<span class="st">&quot;data/Phenotyping.xlsx&quot;</span>, <span class="at">sheet =</span> <span class="st">&quot;GBS&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Clone =</span> <span class="st">`</span><span class="at">Names trials Petrolina</span><span class="st">`</span>,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">ID_Clone =</span> <span class="st">`</span><span class="at">Nome GBS</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">ID_Clone =</span> <span class="fu">str_replace_all</span>(ID_Clone, <span class="st">&quot;:&quot;</span>, <span class="st">&quot;.&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Clone, ID_Clone)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>names <span class="sc">%&gt;%</span></span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">%&gt;%</span></span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Clone
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ID_Clone
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
4271
</td>
<td style="text-align:center;">
4271.250494233
</td>
</tr>
<tr>
<td style="text-align:center;">
9624-09
</td>
<td style="text-align:center;">
962409.250370255
</td>
</tr>
<tr>
<td style="text-align:center;">
Aipim Abacate
</td>
<td style="text-align:center;">
ERETA.250437577
</td>
</tr>
<tr>
<td style="text-align:center;">
Alagoana
</td>
<td style="text-align:center;">
Alagoana363.250437472
</td>
</tr>
<tr>
<td style="text-align:center;">
BGM-0004
</td>
<td style="text-align:center;">
CNPMF4.250370278
</td>
</tr>
<tr>
<td style="text-align:center;">
BGM-0019
</td>
<td style="text-align:center;">
CNPMF19.250370327
</td>
</tr>
</tbody>
</table>
</div>
<div id="phenotypic-data" class="section level3">
<h3>Phenotypic data</h3>
<p>Agora vamos agrupar os IDs dos marcadores com os nomes dos
clones.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;output/BLUPS.RDS&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">inner_join</span>(names) <span class="sc">%&gt;%</span> <span class="co"># Join Phenotypic data with Genotypics names (ID_Clones) of the Clones</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Clone =</span> <span class="fu">as.factor</span>(Clone),</span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">ID_Clone =</span> <span class="fu">as.factor</span>(ID_Clone))</span></code></pre></div>
<pre><code>Joining, by = &quot;Clone&quot;</code></pre>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">head</span>() <span class="sc">%&gt;%</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Clone
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
NR.P
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
PTR
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
PPA
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
MS
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
PROD.AMD
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
AP
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
HI
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
AMD
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Porte
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
RF
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CR
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
DR
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
DCaule
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ácaro
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Nº Hastes
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Stand_Final
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Branching_Level
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Staygreen
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ID_Clone
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
4271
</td>
<td style="text-align:center;">
-1.0991002
</td>
<td style="text-align:center;">
-0.9232850
</td>
<td style="text-align:center;">
1.2370042
</td>
<td style="text-align:center;">
4.1458588
</td>
<td style="text-align:center;">
0.0261588
</td>
<td style="text-align:center;">
-0.0519819
</td>
<td style="text-align:center;">
-4.7190501
</td>
<td style="text-align:center;">
4.1704796
</td>
<td style="text-align:center;">
0.1990073
</td>
<td style="text-align:center;">
-0.0646686
</td>
<td style="text-align:center;">
0.3402402
</td>
<td style="text-align:center;">
-1.684985
</td>
<td style="text-align:center;">
0.1064159
</td>
<td style="text-align:center;">
0.0098488
</td>
<td style="text-align:center;">
-0.2106357
</td>
<td style="text-align:center;">
0.2585191
</td>
<td style="text-align:center;">
0.2189612
</td>
<td style="text-align:center;">
-0.0017225
</td>
<td style="text-align:center;">
4271.250494233
</td>
</tr>
<tr>
<td style="text-align:center;">
9624-09
</td>
<td style="text-align:center;">
2.5448509
</td>
<td style="text-align:center;">
-1.2179385
</td>
<td style="text-align:center;">
2.1813479
</td>
<td style="text-align:center;">
0.0766224
</td>
<td style="text-align:center;">
-0.4243017
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-5.3432293
</td>
<td style="text-align:center;">
-0.0078057
</td>
<td style="text-align:center;">
0.6737550
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-0.2501030
</td>
<td style="text-align:center;">
-3.687546
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-0.2181686
</td>
<td style="text-align:center;">
-0.0500388
</td>
<td style="text-align:center;">
962409.250370255
</td>
</tr>
<tr>
<td style="text-align:center;">
Aipim Abacate
</td>
<td style="text-align:center;">
1.0022448
</td>
<td style="text-align:center;">
1.0140978
</td>
<td style="text-align:center;">
1.1807247
</td>
<td style="text-align:center;">
0.9060262
</td>
<td style="text-align:center;">
0.2158898
</td>
<td style="text-align:center;">
0.3327817
</td>
<td style="text-align:center;">
4.4575307
</td>
<td style="text-align:center;">
0.9313454
</td>
<td style="text-align:center;">
-0.7791874
</td>
<td style="text-align:center;">
0.2490219
</td>
<td style="text-align:center;">
0.8804850
</td>
<td style="text-align:center;">
-1.147279
</td>
<td style="text-align:center;">
-0.0315661
</td>
<td style="text-align:center;">
-0.1144109
</td>
<td style="text-align:center;">
1.0019293
</td>
<td style="text-align:center;">
-0.3944873
</td>
<td style="text-align:center;">
-0.3775248
</td>
<td style="text-align:center;">
-0.0017225
</td>
<td style="text-align:center;">
ERETA.250437577
</td>
</tr>
<tr>
<td style="text-align:center;">
Alagoana
</td>
<td style="text-align:center;">
-0.6253802
</td>
<td style="text-align:center;">
0.1014609
</td>
<td style="text-align:center;">
0.1866389
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-0.0272367
</td>
<td style="text-align:center;">
0.2111938
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-0.0402352
</td>
<td style="text-align:center;">
0.3013914
</td>
<td style="text-align:center;">
2.457162
</td>
<td style="text-align:center;">
-0.0021300
</td>
<td style="text-align:center;">
-0.1964558
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
Alagoana363.250437472
</td>
</tr>
<tr>
<td style="text-align:center;">
BGM-0004
</td>
<td style="text-align:center;">
-0.6980727
</td>
<td style="text-align:center;">
-3.3388524
</td>
<td style="text-align:center;">
-2.2929597
</td>
<td style="text-align:center;">
0.4748982
</td>
<td style="text-align:center;">
-1.0998681
</td>
<td style="text-align:center;">
-0.1925518
</td>
<td style="text-align:center;">
-13.6431414
</td>
<td style="text-align:center;">
0.4197789
</td>
<td style="text-align:center;">
-0.3558847
</td>
<td style="text-align:center;">
0.2427626
</td>
<td style="text-align:center;">
-6.1505681
</td>
<td style="text-align:center;">
-4.975497
</td>
<td style="text-align:center;">
-0.1898330
</td>
<td style="text-align:center;">
0.1316063
</td>
<td style="text-align:center;">
0.8123351
</td>
<td style="text-align:center;">
0.1444067
</td>
<td style="text-align:center;">
-0.0193399
</td>
<td style="text-align:center;">
0.0242331
</td>
<td style="text-align:center;">
CNPMF4.250370278
</td>
</tr>
<tr>
<td style="text-align:center;">
BGM-0019
</td>
<td style="text-align:center;">
-2.8554890
</td>
<td style="text-align:center;">
-2.4587770
</td>
<td style="text-align:center;">
-5.7454297
</td>
<td style="text-align:center;">
-3.5447503
</td>
<td style="text-align:center;">
-0.1443845
</td>
<td style="text-align:center;">
-0.1536869
</td>
<td style="text-align:center;">
-14.0193447
</td>
<td style="text-align:center;">
-3.5290772
</td>
<td style="text-align:center;">
0.3620398
</td>
<td style="text-align:center;">
-0.0646686
</td>
<td style="text-align:center;">
-5.4186360
</td>
<td style="text-align:center;">
-3.088693
</td>
<td style="text-align:center;">
-0.2775123
</td>
<td style="text-align:center;">
-0.1144109
</td>
<td style="text-align:center;">
-0.7880476
</td>
<td style="text-align:center;">
-0.0679841
</td>
<td style="text-align:center;">
0.4177899
</td>
<td style="text-align:center;">
-0.0017225
</td>
<td style="text-align:center;">
CNPMF19.250370327
</td>
</tr>
</tbody>
</table>
</div>
<div id="genotypic-data" class="section level3">
<h3>Genotypic data</h3>
<p>Agora vamos carregar os dados genotípicos dos marcadores GBS e
corrigir os valores dos pares de base. Além disso, também vamos dividir
a coluna alleles em duas colunas, para o alelo de referencia e o
recessivo. E vamos selecionar as colunas com os nomes dos marcadores,
alelos de referencia e as colunas com os IDs dos clones de acordo com os
dados dos BLUPs.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/allchrAR08.txt&quot;</span>, <span class="at">header =</span> T)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno <span class="sc">%&gt;%</span></span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="dv">12</span><span class="sc">:</span><span class="fu">ncol</span>(geno)), <span class="fu">funs</span>(</span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>    <span class="fu">case_when</span>(</span>
<span id="cb6-6"><a href="#cb6-6" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;A&quot;</span> <span class="sc">~</span> <span class="st">&quot;AA&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;R&quot;</span> <span class="sc">~</span> <span class="st">&quot;AG&quot;</span>,</span>
<span id="cb6-8"><a href="#cb6-8" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;W&quot;</span> <span class="sc">~</span> <span class="st">&quot;AT&quot;</span>,</span>
<span id="cb6-9"><a href="#cb6-9" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;M&quot;</span> <span class="sc">~</span> <span class="st">&quot;AC&quot;</span>,</span>
<span id="cb6-10"><a href="#cb6-10" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;C&quot;</span> <span class="sc">~</span> <span class="st">&quot;CC&quot;</span>,</span>
<span id="cb6-11"><a href="#cb6-11" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;S&quot;</span> <span class="sc">~</span> <span class="st">&quot;CG&quot;</span>,</span>
<span id="cb6-12"><a href="#cb6-12" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;Y&quot;</span> <span class="sc">~</span> <span class="st">&quot;CT&quot;</span>,</span>
<span id="cb6-13"><a href="#cb6-13" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;G&quot;</span> <span class="sc">~</span> <span class="st">&quot;GG&quot;</span>,</span>
<span id="cb6-14"><a href="#cb6-14" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;K&quot;</span> <span class="sc">~</span> <span class="st">&quot;GT&quot;</span>,</span>
<span id="cb6-15"><a href="#cb6-15" aria-hidden="true" tabindex="-1"></a>      . <span class="sc">==</span> <span class="st">&quot;T&quot;</span> <span class="sc">~</span> <span class="st">&quot;TT&quot;</span>,</span>
<span id="cb6-16"><a href="#cb6-16" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb6-17"><a href="#cb6-17" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span></span>
<span id="cb6-18"><a href="#cb6-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">separate</span>(alleles, <span class="fu">c</span>(<span class="st">&quot;reference&quot;</span>, <span class="st">&quot;recess&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb6-19"><a href="#cb6-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(rs, reference, recess, <span class="fu">any_of</span>(pheno<span class="sc">$</span>ID_Clone))</span>
<span id="cb6-20"><a href="#cb6-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-21"><a href="#cb6-21" aria-hidden="true" tabindex="-1"></a>geno[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>] <span class="sc">%&gt;%</span></span>
<span id="cb6-22"><a href="#cb6-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-23"><a href="#cb6-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb6-24"><a href="#cb6-24" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb6-25"><a href="#cb6-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb6-26"><a href="#cb6-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb6-27"><a href="#cb6-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb6-28"><a href="#cb6-28" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
rs
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
reference
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
recess
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ERETA.250437577
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Alagoana363.250437472
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF4.250370278
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF19.250370327
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF30.250370293
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF32.250370305
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF36.250370328
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF46.250370283
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF48.250370295
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF56.250370248
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF65.250370285
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF66.250370296
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF74.250370321
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF80.250370250
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF84.250370263
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF89.250370287
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF91.250370299
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
S1_84637
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
G
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AG
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_84843
</td>
<td style="text-align:center;">
C
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
AC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
<td style="text-align:center;">
CC
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_126260
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
C
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_126261
</td>
<td style="text-align:center;">
A
</td>
<td style="text-align:center;">
G
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
<td style="text-align:center;">
AA
</td>
</tr>
<tr>
<td style="text-align:center;">
S1_126264
</td>
<td style="text-align:center;">
T
</td>
<td style="text-align:center;">
G
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
<td style="text-align:center;">
TT
</td>
</tr>
</tbody>
</table>
<p>Agora precisamos fazer a conversão de pares de base para dosagem
alélica de acordo com o alelo de referência. Também vou adcionar a
coluna rs como nome das colunas. Depois vou excluir as colunas dos
alelos de reference e recess.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate_at</span>(<span class="fu">vars</span>(<span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(geno)), <span class="fu">funs</span>(<span class="fu">case_when</span>(</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>    . <span class="sc">==</span> <span class="fu">paste</span>(reference, reference, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">~</span> <span class="dv">2</span>,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>    . <span class="sc">==</span> <span class="fu">paste</span>(recess, recess, <span class="at">sep =</span> <span class="st">&quot;&quot;</span>) <span class="sc">~</span> <span class="dv">0</span>,</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>    T <span class="sc">~</span> <span class="dv">1</span></span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>  ))) <span class="sc">%&gt;%</span></span>
<span id="cb7-7"><a href="#cb7-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_to_rownames</span>(<span class="st">&quot;rs&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-8"><a href="#cb7-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(reference, recess))</span>
<span id="cb7-9"><a href="#cb7-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-10"><a href="#cb7-10" aria-hidden="true" tabindex="-1"></a>geno[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">15</span>] <span class="sc">%&gt;%</span></span>
<span id="cb7-11"><a href="#cb7-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb7-12"><a href="#cb7-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb7-13"><a href="#cb7-13" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb7-14"><a href="#cb7-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb7-15"><a href="#cb7-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb7-16"><a href="#cb7-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb7-17"><a href="#cb7-17" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ERETA.250437577
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Alagoana363.250437472
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF4.250370278
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF19.250370327
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF30.250370293
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF32.250370305
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF36.250370328
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF46.250370283
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF48.250370295
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF56.250370248
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF65.250370285
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF66.250370296
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF74.250370321
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF80.250370250
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
CNPMF84.250370263
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
S1_84637
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
S1_84843
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
S1_126260
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
S1_126261
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
S1_126264
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
</tbody>
</table>
<p>Para converter no formato para realizar as análises de GWS temos que
transpor a matriz de marcadores.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>()</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a>geno[<span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>, <span class="dv">1</span><span class="sc">:</span><span class="dv">5</span>] <span class="sc">%&gt;%</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-6"><a href="#cb8-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb8-7"><a href="#cb8-7" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb8-8"><a href="#cb8-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb8-9"><a href="#cb8-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb8-10"><a href="#cb8-10" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb8-11"><a href="#cb8-11" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_84637
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_84843
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_126260
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_126261
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
S1_126264
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
ERETA.250437577
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
Alagoana363.250437472
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
CNPMF4.250370278
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
CNPMF19.250370327
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
<tr>
<td style="text-align:left;">
CNPMF30.250370293
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
2
</td>
</tr>
</tbody>
</table>
<p>Vamos verificar quantos clones apresentam dados genotipados com os
marcadores.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>geno <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(</span>
<span id="cb9-5"><a href="#cb9-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">escape =</span> F,</span>
<span id="cb9-6"><a href="#cb9-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">&#39;c&#39;</span>,</span>
<span id="cb9-7"><a href="#cb9-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Number of Clones&quot;</span>, <span class="st">&quot;Number of markers&quot;</span>)</span>
<span id="cb9-8"><a href="#cb9-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb9-9"><a href="#cb9-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb9-10"><a href="#cb9-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb9-11"><a href="#cb9-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb9-12"><a href="#cb9-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb9-13"><a href="#cb9-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb9-14"><a href="#cb9-14" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of Clones
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of markers
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
414
</td>
<td style="text-align:center;">
27045
</td>
</tr>
</tbody>
</table>
<p>Com a filtragem dos genótipos em comum, temos 414 genótipos e 27405
marcas.</p>
<p>Agora vamos filtrar os SNPS usando MAF de 0.01 e verificar quantos
marcadores se manterão.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> <span class="fu">maf_filter</span>(geno, <span class="at">thresh =</span> <span class="fl">0.01</span>)</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>geno <span class="sc">%&gt;%</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dim</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">escape =</span> F,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">&#39;c&#39;</span>,</span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;Number of Clones&quot;</span>, <span class="st">&quot;Number of markers&quot;</span>)</span>
<span id="cb10-10"><a href="#cb10-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb10-11"><a href="#cb10-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb10-12"><a href="#cb10-12" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb10-13"><a href="#cb10-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb10-14"><a href="#cb10-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb10-15"><a href="#cb10-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb10-16"><a href="#cb10-16" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of Clones
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Number of markers
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
414
</td>
<td style="text-align:center;">
22779
</td>
</tr>
</tbody>
</table>
<p>Com o filtro de MAF a 1% restaram 22779 marcadores. Vou salvar a
matriz agora para que possamos carregar ela caso seja necessário.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(geno, <span class="st">&quot;data/SNPs.rds&quot;</span>)</span></code></pre></div>
</div>
</div>
<div id="building-the-g-matrix" class="section level2">
<h2>Building the G matrix</h2>
<p>Again, we will use the <a
href="https://cran.r-project.org/web/packages/AGHmatrix/vignettes/Tutorial_AGHmatrix.html">AGHmatrix</a>
package <span class="citation">[@amadeu_aghmatrix_2016]</span> to build
the G matrix:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>G_matrix <span class="ot">&lt;-</span> <span class="fu">Gmatrix</span>(geno,</span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a>                    <span class="at">method =</span> <span class="st">&quot;VanRaden&quot;</span>,</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>                    <span class="at">ploidy =</span> <span class="dv">2</span>,</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>                    <span class="at">missingValue =</span> <span class="cn">NA</span>)</span></code></pre></div>
<pre><code>Initial data: 
    Number of Individuals: 414 
    Number of Markers: 22779 

Missing data check: 
    Total SNPs: 22779 
     0 SNPs dropped due to missing data threshold of 1 
    Total of: 22779  SNPs 
MAF check: 
    No SNPs with MAF below 0 
Monomorphic check: 
    No monomorphic SNPs 
Summary check: 
    Initial:  22779 SNPs 
    Final:  22779  SNPs ( 0  SNPs removed) 
 
Completed! Time = 23.2  seconds </code></pre>
<p>Now we have the whole G matrix (414 x 414), which we can represent
using a heatmap:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  G_matrix,</span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_row_names =</span> F,</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">show_column_names =</span> F,</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;Res&quot;</span>)</span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figure/GWS.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>“Res” in the heatmap legend title is for “Resemblance”.</p>
</div>
<div id="rrblup" class="section level2">
<h2>RRBLUP</h2>
<p>The Ridge Regression BLUP, or RRBLUP, will predict the marker effect.
In the RRBLUP, we will use the matrix of markers directly. For this
purpose, we will use only individuals with BLUps and SNPs available.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(ID_Clone <span class="sc">%in%</span> <span class="fu">rownames</span>(geno)) <span class="sc">%&gt;%</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">droplevels</span>()</span></code></pre></div>
<p>For this purpose, we will use the <a
href="https://cran.r-project.org/web/packages/rrBLUP/rrBLUP.pdf">rrBLUP</a>
package <span class="citation">[@endelman_2011]</span>. In the code
below, <em>y</em> is for the vector with the means, <em>Z</em> is where
we will insert the SNPs matrix, <em>K</em> is for a covariance matrix
for the random effects, which will be and identity matrix by default;
and <em>X</em> is a design matrix for the fixed effects, which will be a
vector of ones (<strong>1</strong>) by default. Note that we are
returning to the “1, 0, -1” codification in the SNPs matrix. This is a
requirement of the <em>rrBlUP</em> package.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>result_RR_BLUP <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Clone =</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>Clone),</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID_Clone =</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>ID_Clone),</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">stringsAsFactors =</span> F</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[<span class="dv">2</span><span class="sc">:</span><span class="dv">19</span>]</span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> traits) {</span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>  RRBLUP <span class="ot">&lt;-</span> <span class="fu">mixed.solve</span>(<span class="at">y =</span> pheno[[i]],</span>
<span id="cb16-11"><a href="#cb16-11" aria-hidden="true" tabindex="-1"></a>                        <span class="at">Z =</span> geno <span class="sc">-</span> <span class="dv">1</span>,</span>
<span id="cb16-12"><a href="#cb16-12" aria-hidden="true" tabindex="-1"></a>                        <span class="at">K =</span> <span class="cn">NULL</span>,</span>
<span id="cb16-13"><a href="#cb16-13" aria-hidden="true" tabindex="-1"></a>                        <span class="at">X =</span> <span class="cn">NULL</span>)</span>
<span id="cb16-14"><a href="#cb16-14" aria-hidden="true" tabindex="-1"></a>  GEBV <span class="ot">&lt;-</span> geno <span class="sc">%*%</span> RRBLUP<span class="sc">$</span>u</span>
<span id="cb16-15"><a href="#cb16-15" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID_Clone =</span> <span class="fu">rownames</span>(GEBV),</span>
<span id="cb16-16"><a href="#cb16-16" aria-hidden="true" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb16-17"><a href="#cb16-17" aria-hidden="true" tabindex="-1"></a>  result[, i] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(GEBV)</span>
<span id="cb16-18"><a href="#cb16-18" aria-hidden="true" tabindex="-1"></a>  result_RR_BLUP <span class="ot">&lt;-</span></span>
<span id="cb16-19"><a href="#cb16-19" aria-hidden="true" tabindex="-1"></a>    <span class="fu">merge</span>(result_RR_BLUP, result, <span class="at">by =</span> <span class="st">&quot;ID_Clone&quot;</span>, <span class="at">all.x =</span> T)</span>
<span id="cb16-20"><a href="#cb16-20" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<div id="gebv-x-blup" class="section level3">
<h3>GEBV x BLUP</h3>
<p>The scatter plot above represents the additive genetic value of each
marker. Once we have acquired these values, we may calculate the Genomic
Estimated Breeding Values (GEBV) of the genotypes. Aqui, iremos
adicionar os valores médios dos valores fenotípicos aos GEBVs obtidos
pelo RR-BLUP e aos BLUPs para uma melhor comparação. These are the
product of the SNPs matrix with the vector of the markers’ genetic
values:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>media_pheno <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;output/media_pheno.csv&quot;</span>)</span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a><span class="fu">colnames</span>(media_pheno) <span class="ot">&lt;-</span></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">c</span>(<span class="fu">colnames</span>(result_RR_BLUP[<span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(result_RR_BLUP)]))</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> traits) {</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  result_RR_BLUP[i] <span class="ot">&lt;-</span> result_RR_BLUP[i] <span class="sc">+</span> media_pheno[, i]</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a>GEBVS_RR_BLUP <span class="ot">&lt;-</span> result_RR_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(NR.P<span class="sc">:</span>Staygreen,</span>
<span id="cb17-12"><a href="#cb17-12" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;Variable&quot;</span>,</span>
<span id="cb17-13"><a href="#cb17-13" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;GEBVS&quot;</span>)</span>
<span id="cb17-14"><a href="#cb17-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-15"><a href="#cb17-15" aria-hidden="true" tabindex="-1"></a>BLUPS_MED <span class="ot">&lt;-</span> pheno</span>
<span id="cb17-16"><a href="#cb17-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-17"><a href="#cb17-17" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> traits) {</span>
<span id="cb17-18"><a href="#cb17-18" aria-hidden="true" tabindex="-1"></a>  BLUPS_MED[i] <span class="ot">&lt;-</span> BLUPS_MED[i] <span class="sc">+</span> media_pheno[, i]</span>
<span id="cb17-19"><a href="#cb17-19" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb17-20"><a href="#cb17-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-21"><a href="#cb17-21" aria-hidden="true" tabindex="-1"></a>BLUPS_MED <span class="ot">&lt;-</span> BLUPS_MED <span class="sc">%&gt;%</span></span>
<span id="cb17-22"><a href="#cb17-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(NR.P<span class="sc">:</span>Staygreen,</span>
<span id="cb17-23"><a href="#cb17-23" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;Variable&quot;</span>,</span>
<span id="cb17-24"><a href="#cb17-24" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;BLUPS&quot;</span>)</span>
<span id="cb17-25"><a href="#cb17-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-26"><a href="#cb17-26" aria-hidden="true" tabindex="-1"></a>data_gws_RR_BLUP <span class="ot">&lt;-</span> GEBVS_RR_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb17-27"><a href="#cb17-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(BLUPS_MED)</span></code></pre></div>
<pre><code>Joining, by = c(&quot;ID_Clone&quot;, &quot;Clone&quot;, &quot;Variable&quot;)</code></pre>
<p>Agora vamos produzir os gráficos para cada variável comparando a
correlação entre os GEBVs e BLUPs.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a>data_gws_RR_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> BLUPS, <span class="at">y =</span> GEBVS)) <span class="sc">+</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> GEBVS), <span class="at">show.legend =</span> F) <span class="sc">+</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> lm, <span class="at">se =</span> F) <span class="sc">+</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_poly_eq</span>(<span class="at">formula =</span> y <span class="sc">~</span> x, <span class="at">label.y =</span> <span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&#39;#c80000&#39;</span>, <span class="at">high =</span> <span class="st">&#39;#FFFF00&#39;</span>) <span class="sc">+</span></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(. <span class="sc">~</span> Variable, <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">x =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;RR-BLUP&quot;</span>)</span></code></pre></div>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/GWS.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Agora vamos comparar os modelos por meio de um boxplot.</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a>data_gws_RR_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">names_to =</span> <span class="st">&quot;Method&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Values&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> , <span class="at">y =</span> Values, <span class="at">fill =</span> Method)) <span class="sc">+</span></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(. <span class="sc">~</span> Variable, <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)  <span class="sc">+</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">x =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;RR-BLUP&quot;</span>)</span></code></pre></div>
<p><img src="figure/GWS.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p><br></p>
</div>
</div>
<div id="gblup" class="section level2">
<h2>GBLUP</h2>
<p>In the GBLUP, we will use the G matrix instead of the SNPs matrix.
Thus, we will obtain the GEBV directly. Note that we will need to build
the G matrix again, since some genotypes were dropped after our
filtering. The rrBLUP package has a function called “A.mat” that build
the Additive Genomic Matrix from a SNP matrix with “-1,0,1”
codification:</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>result_G_BLUP <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">Clone =</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>Clone),</span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ID_Clone =</span> <span class="fu">unique</span>(pheno<span class="sc">$</span>ID_Clone),</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">stringsAsFactors =</span> F</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> traits) {</span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>  GBLUP <span class="ot">&lt;-</span> <span class="fu">mixed.solve</span>(pheno[[i]], <span class="at">K =</span> <span class="fu">A.mat</span>(geno <span class="sc">-</span> <span class="dv">1</span>))</span>
<span id="cb22-9"><a href="#cb22-9" aria-hidden="true" tabindex="-1"></a>  result <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">ID_Clone =</span> <span class="fu">rownames</span>(GBLUP<span class="sc">$</span>u),</span>
<span id="cb22-10"><a href="#cb22-10" aria-hidden="true" tabindex="-1"></a>                       <span class="at">stringsAsFactors =</span> F)</span>
<span id="cb22-11"><a href="#cb22-11" aria-hidden="true" tabindex="-1"></a>  result[, i] <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(GBLUP<span class="sc">$</span>u)</span>
<span id="cb22-12"><a href="#cb22-12" aria-hidden="true" tabindex="-1"></a>  result_G_BLUP <span class="ot">&lt;-</span></span>
<span id="cb22-13"><a href="#cb22-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">merge</span>(result_G_BLUP, result, <span class="at">by =</span> <span class="st">&quot;ID_Clone&quot;</span>, <span class="at">all.x =</span> T)</span>
<span id="cb22-14"><a href="#cb22-14" aria-hidden="true" tabindex="-1"></a>}</span></code></pre></div>
<p>Aqui, iremos adicionar novamente os valores médios dos valores
fenotípicos aos GEBVs obtidos pelo G-BLUP.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(i <span class="cf">in</span> traits) {</span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  result_G_BLUP[i] <span class="ot">&lt;-</span> result_G_BLUP[i] <span class="sc">+</span> media_pheno[, i]</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>GEBVS_G_BLUP <span class="ot">&lt;-</span> result_G_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(NR.P<span class="sc">:</span>Staygreen,</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_to =</span> <span class="st">&quot;Variable&quot;</span>,</span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>               <span class="at">values_to =</span> <span class="st">&quot;GEBVS&quot;</span>)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>data_gws_G_BLUP <span class="ot">&lt;-</span> GEBVS_G_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(BLUPS_MED)</span></code></pre></div>
<pre><code>Joining, by = c(&quot;ID_Clone&quot;, &quot;Clone&quot;, &quot;Variable&quot;)</code></pre>
<p>Agora vamos produzir os gráficos para cada variável comparando a
correlação entre os GEBVs e BLUPs.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>data_gws_G_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> BLUPS, <span class="at">y =</span> GEBVS)) <span class="sc">+</span></span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">color =</span> GEBVS), <span class="at">show.legend =</span> F) <span class="sc">+</span></span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> lm, <span class="at">se =</span> F) <span class="sc">+</span></span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_poly_eq</span>(<span class="at">formula =</span> y <span class="sc">~</span> x, <span class="at">label.y =</span> <span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_gradient</span>(<span class="at">low =</span> <span class="st">&#39;#c80000&#39;</span>, <span class="at">high =</span> <span class="st">&#39;#FFFF00&#39;</span>) <span class="sc">+</span></span>
<span id="cb25-7"><a href="#cb25-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(. <span class="sc">~</span> Variable, <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb25-8"><a href="#cb25-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">x =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb25-9"><a href="#cb25-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;G-BLUP&quot;</span>)</span></code></pre></div>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/GWS.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Agora vamos comparar os modelos por meio de um boxplot.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>data_gws_G_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">4</span><span class="sc">:</span><span class="dv">5</span>, <span class="at">names_to =</span> <span class="st">&quot;Method&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Values&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> , <span class="at">y =</span> Values, <span class="at">fill =</span> Method)) <span class="sc">+</span></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(. <span class="sc">~</span> Variable, <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>)  <span class="sc">+</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">x =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;G-BLUP&quot;</span>)</span></code></pre></div>
<p><img src="figure/GWS.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Agora vamos produzir os gráficos para cada variável comparando a
correlação entre os GEBVs do RR-BLUP e GEBVs do G-BLUP.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a>GEBVS_RR_BLUP <span class="ot">&lt;-</span> GEBVS_RR_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">GEBVS_RR_BLUP =</span> GEBVS)</span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>data_gws <span class="ot">&lt;-</span> GEBVS_RR_BLUP <span class="sc">%&gt;%</span></span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">full_join</span>(GEBVS_G_BLUP)</span></code></pre></div>
<pre><code>Joining, by = c(&quot;ID_Clone&quot;, &quot;Clone&quot;, &quot;Variable&quot;)</code></pre>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a>data_gws <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> GEBVS_RR_BLUP, <span class="at">y =</span> GEBVS)) <span class="sc">+</span></span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_smooth</span>(<span class="at">method =</span> lm, <span class="at">se =</span> F) <span class="sc">+</span></span>
<span id="cb30-5"><a href="#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_poly_eq</span>(<span class="at">formula =</span> y <span class="sc">~</span> x, <span class="at">label.y =</span> <span class="st">&quot;bottom&quot;</span>) <span class="sc">+</span></span>
<span id="cb30-6"><a href="#cb30-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(. <span class="sc">~</span> Variable, <span class="at">ncol =</span> <span class="dv">6</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb30-7"><a href="#cb30-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">expand_limits</span>(<span class="at">y =</span> <span class="dv">0</span>, <span class="at">x =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb30-8"><a href="#cb30-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggtitle</span>(<span class="st">&quot;G-BLUP&quot;</span>)</span></code></pre></div>
<pre><code>`geom_smooth()` using formula &#39;y ~ x&#39;</code></pre>
<p><img src="figure/GWS.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="cross-validation" class="section level2">
<h2>Cross-validation</h2>
<p>To prove that the prediction is accurate, we should perform a
cross-validation (CV) scheme. For this purpose, we divide the data into
a training set and a validation set.</p>
<p>First we separate the data into <em>k</em> folds. Then, we attribute
<em>NA</em> for one fold and try to predict the data from this fold
based on the others. When selecting the number of folds, one must
prioritize the balance between the number of observations in each
fold.</p>
<p>In addition, this process should be repeated for further validation.
The step-by-step below will guide the CV in the data we are
analysing.</p>
<div id="determine-the-number-of-folds-and-repetitions"
class="section level3">
<h3>1. Determine the number of folds and repetitions</h3>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>nfolds <span class="ot">=</span> <span class="dv">5</span></span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>nrept <span class="ot">=</span> <span class="dv">5</span></span></code></pre></div>
<p>Since we defined 5 folds, our data will be divided into 5 parts with
83 observations each.</p>
</div>
<div id="match-the-order-of-the-data-and-the-rows-of-the-snp-matrix"
class="section level3">
<h3>2. Match the order of the data and the rows of the SNP matrix</h3>
<p>The order is decreasing or increasing (numeric or alphabetical)
regarding the name of the genotypes.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno[<span class="fu">order</span>(pheno<span class="sc">$</span>ID_Clone, <span class="at">decreasing =</span> F),]</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>geno <span class="ot">&lt;-</span> geno[<span class="fu">order</span>(<span class="fu">row.names</span>(geno)),]</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(<span class="fu">rownames</span>(geno) <span class="sc">==</span> pheno<span class="sc">$</span>ID_Clone)</span></code></pre></div>
<pre><code>[1] TRUE</code></pre>
</div>
<div id="add-a-column-indicating-a-number-for-each-observation"
class="section level3">
<h3>3. Add a column indicating a number for each observation</h3>
<p>This will be useful to assign each observation for a fold, which will
be the next step.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>pheno<span class="sc">$</span>ID <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(pheno))</span></code></pre></div>
</div>
<div id="folds-assignment" class="section level3">
<h3>4. Folds assignment</h3>
<p>In this step, we will assign each observation to a fold. Bear in mind
that for each repetition, the folds will comprise different
observations. The purpose of the repetition is to make sure of the
randomness of the assignment step.</p>
<p>In this step, we will use the <a
href="https://cran.r-project.org/web/packages/cvTools/cvTools.pdf">cvTools</a>
package <span class="citation">[@cvTools]</span></p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">100</span>)</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>sort <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (a <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nrept) {</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (j <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nfolds) {</span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>    folds <span class="ot">&lt;-</span> <span class="fu">cvFolds</span>(<span class="fu">nlevels</span>(pheno<span class="sc">$</span>ID),</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a>                     <span class="at">type =</span> <span class="st">&quot;random&quot;</span>,</span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a>                     <span class="at">K =</span> <span class="dv">5</span>,</span>
<span id="cb36-10"><a href="#cb36-10" aria-hidden="true" tabindex="-1"></a>                     <span class="at">R =</span> <span class="dv">1</span>)</span>
<span id="cb36-11"><a href="#cb36-11" aria-hidden="true" tabindex="-1"></a>    Sample <span class="ot">&lt;-</span> <span class="fu">cbind</span>(folds<span class="sc">$</span>which, folds<span class="sc">$</span>subsets)</span>
<span id="cb36-12"><a href="#cb36-12" aria-hidden="true" tabindex="-1"></a>    cv <span class="ot">&lt;-</span> <span class="fu">split</span>(Sample[, <span class="dv">2</span>], <span class="at">f =</span> Sample[, <span class="dv">1</span>])</span>
<span id="cb36-13"><a href="#cb36-13" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb36-14"><a href="#cb36-14" aria-hidden="true" tabindex="-1"></a>  sort[[a]] <span class="ot">&lt;-</span> cv</span>
<span id="cb36-15"><a href="#cb36-15" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb36-16"><a href="#cb36-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-17"><a href="#cb36-17" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(a, folds, j, cv, Sample)</span></code></pre></div>
</div>
<div id="cross-validation-1" class="section level3">
<h3>5. Cross-validation</h3>
<p>The next step is the very CV. Here, we will define the linear
predictor and the lists that will be useful in the loop. The first list,
here called “fold.list”, contains the folds assignation that we built in
the previous step. The second (“results_cv”) is empty and will store the
outputs of each iteration of the loop.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>fold.list <span class="ot">&lt;-</span> sort</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>results <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>results_cv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>()</span></code></pre></div>
<p>Then, we will construct the loop. Each iteration will assign
<em>NA</em> for a different fold, and we will use the other folds to
predict the missing values. Note that the folds vary for each
repetition.</p>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span>(j <span class="cf">in</span> traits) {</span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span> (z <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(fold.list)) {</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>    <span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span>nfolds) {</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Training set</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>      train_data <span class="ot">&lt;-</span> pheno</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Validation set</span></span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a>      train_data[train_data<span class="sc">$</span>ID <span class="sc">%in%</span> fold.list[[z]][[i]], j] <span class="ot">&lt;-</span> <span class="cn">NA</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Fitting model</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a>      GBLUP <span class="ot">&lt;-</span> <span class="fu">mixed.solve</span>(train_data[[j]], <span class="at">K =</span> <span class="fu">A.mat</span>(geno <span class="sc">-</span> <span class="dv">1</span>))</span>
<span id="cb38-12"><a href="#cb38-12" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-13"><a href="#cb38-13" aria-hidden="true" tabindex="-1"></a>      <span class="co"># GEBV</span></span>
<span id="cb38-14"><a href="#cb38-14" aria-hidden="true" tabindex="-1"></a>      Pred <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">Yhat =</span> GBLUP<span class="sc">$</span>u, <span class="at">G =</span> pheno<span class="sc">$</span>ID)</span>
<span id="cb38-15"><a href="#cb38-15" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-16"><a href="#cb38-16" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rownames</span>(Pred) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(geno)</span>
<span id="cb38-17"><a href="#cb38-17" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-18"><a href="#cb38-18" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Predicted GEBV</span></span>
<span id="cb38-19"><a href="#cb38-19" aria-hidden="true" tabindex="-1"></a>      results[[i]] <span class="ot">&lt;-</span> Pred[Pred[, <span class="st">&quot;G&quot;</span>] <span class="sc">%in%</span> fold.list[[z]][[i]], ]</span>
<span id="cb38-20"><a href="#cb38-20" aria-hidden="true" tabindex="-1"></a>      </span>
<span id="cb38-21"><a href="#cb38-21" aria-hidden="true" tabindex="-1"></a>      <span class="co"># Remove</span></span>
<span id="cb38-22"><a href="#cb38-22" aria-hidden="true" tabindex="-1"></a>      <span class="co">#rm(GBLUP, Pred, train_data)</span></span>
<span id="cb38-23"><a href="#cb38-23" aria-hidden="true" tabindex="-1"></a>    }</span>
<span id="cb38-24"><a href="#cb38-24" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-25"><a href="#cb38-25" aria-hidden="true" tabindex="-1"></a>    GEBV <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, results)</span>
<span id="cb38-26"><a href="#cb38-26" aria-hidden="true" tabindex="-1"></a>    GEBV <span class="ot">&lt;-</span> GEBV[<span class="fu">order</span>(GEBV<span class="sc">$</span>G), ]</span>
<span id="cb38-27"><a href="#cb38-27" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-28"><a href="#cb38-28" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Log</span></span>
<span id="cb38-29"><a href="#cb38-29" aria-hidden="true" tabindex="-1"></a>    log <span class="ot">&lt;-</span> <span class="fu">all</span>(GEBV<span class="sc">$</span>G <span class="sc">==</span> pheno<span class="sc">$</span>ID)</span>
<span id="cb38-30"><a href="#cb38-30" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-31"><a href="#cb38-31" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Results</span></span>
<span id="cb38-32"><a href="#cb38-32" aria-hidden="true" tabindex="-1"></a>    result_cv <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(</span>
<span id="cb38-33"><a href="#cb38-33" aria-hidden="true" tabindex="-1"></a>      <span class="at">Trait =</span> j,</span>
<span id="cb38-34"><a href="#cb38-34" aria-hidden="true" tabindex="-1"></a>      <span class="at">Rep =</span> z,</span>
<span id="cb38-35"><a href="#cb38-35" aria-hidden="true" tabindex="-1"></a>      <span class="at">Log =</span> log,</span>
<span id="cb38-36"><a href="#cb38-36" aria-hidden="true" tabindex="-1"></a>      <span class="at">Ac =</span> <span class="fu">round</span>(<span class="fu">cor</span>(GEBV<span class="sc">$</span>Yhat, pheno<span class="sc">$</span>MS, <span class="at">use =</span> <span class="st">&quot;na.or.complete&quot;</span>), <span class="dv">3</span>),</span>
<span id="cb38-37"><a href="#cb38-37" aria-hidden="true" tabindex="-1"></a>      <span class="at">MSPE =</span> <span class="fu">round</span>(<span class="fu">mean</span>(((GEBV<span class="sc">$</span>Yhat <span class="sc">-</span> pheno<span class="sc">$</span>MS) <span class="sc">^</span> <span class="dv">2</span></span>
<span id="cb38-38"><a href="#cb38-38" aria-hidden="true" tabindex="-1"></a>      ), <span class="at">na.rm =</span> T), <span class="dv">3</span>)</span>
<span id="cb38-39"><a href="#cb38-39" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb38-40"><a href="#cb38-40" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-41"><a href="#cb38-41" aria-hidden="true" tabindex="-1"></a>    results_cv <span class="ot">&lt;-</span></span>
<span id="cb38-42"><a href="#cb38-42" aria-hidden="true" tabindex="-1"></a>      <span class="fu">rbind</span>(results_cv, result_cv)</span>
<span id="cb38-43"><a href="#cb38-43" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb38-44"><a href="#cb38-44" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb38-45"><a href="#cb38-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-46"><a href="#cb38-46" aria-hidden="true" tabindex="-1"></a>results_cv <span class="sc">%&gt;%</span></span>
<span id="cb38-47"><a href="#cb38-47" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb38-48"><a href="#cb38-48" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb38-49"><a href="#cb38-49" aria-hidden="true" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb38-50"><a href="#cb38-50" aria-hidden="true" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb38-51"><a href="#cb38-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb38-52"><a href="#cb38-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb38-53"><a href="#cb38-53" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb38-54"><a href="#cb38-54" aria-hidden="true" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Trait
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Rep
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Log
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ac
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
MSPE
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
NR.P
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.103
</td>
<td style="text-align:center;">
10.639
</td>
</tr>
<tr>
<td style="text-align:center;">
NR.P
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.127
</td>
<td style="text-align:center;">
10.585
</td>
</tr>
<tr>
<td style="text-align:center;">
NR.P
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.122
</td>
<td style="text-align:center;">
10.596
</td>
</tr>
<tr>
<td style="text-align:center;">
NR.P
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.133
</td>
<td style="text-align:center;">
10.573
</td>
</tr>
<tr>
<td style="text-align:center;">
NR.P
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.131
</td>
<td style="text-align:center;">
10.574
</td>
</tr>
<tr>
<td style="text-align:center;">
PTR
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.142
</td>
<td style="text-align:center;">
10.678
</td>
</tr>
<tr>
<td style="text-align:center;">
PTR
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.163
</td>
<td style="text-align:center;">
10.565
</td>
</tr>
<tr>
<td style="text-align:center;">
PTR
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.173
</td>
<td style="text-align:center;">
10.517
</td>
</tr>
<tr>
<td style="text-align:center;">
PTR
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.180
</td>
<td style="text-align:center;">
10.451
</td>
</tr>
<tr>
<td style="text-align:center;">
PTR
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.146
</td>
<td style="text-align:center;">
10.652
</td>
</tr>
<tr>
<td style="text-align:center;">
PPA
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.112
</td>
<td style="text-align:center;">
13.087
</td>
</tr>
<tr>
<td style="text-align:center;">
PPA
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.161
</td>
<td style="text-align:center;">
12.618
</td>
</tr>
<tr>
<td style="text-align:center;">
PPA
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.168
</td>
<td style="text-align:center;">
12.267
</td>
</tr>
<tr>
<td style="text-align:center;">
PPA
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.169
</td>
<td style="text-align:center;">
12.151
</td>
</tr>
<tr>
<td style="text-align:center;">
PPA
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.136
</td>
<td style="text-align:center;">
12.541
</td>
</tr>
<tr>
<td style="text-align:center;">
MS
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.238
</td>
<td style="text-align:center;">
10.168
</td>
</tr>
<tr>
<td style="text-align:center;">
MS
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.246
</td>
<td style="text-align:center;">
10.111
</td>
</tr>
<tr>
<td style="text-align:center;">
MS
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.262
</td>
<td style="text-align:center;">
10.034
</td>
</tr>
<tr>
<td style="text-align:center;">
MS
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.300
</td>
<td style="text-align:center;">
9.784
</td>
</tr>
<tr>
<td style="text-align:center;">
MS
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.204
</td>
<td style="text-align:center;">
10.416
</td>
</tr>
<tr>
<td style="text-align:center;">
PROD.AMD
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.133
</td>
<td style="text-align:center;">
10.604
</td>
</tr>
<tr>
<td style="text-align:center;">
PROD.AMD
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.171
</td>
<td style="text-align:center;">
10.552
</td>
</tr>
<tr>
<td style="text-align:center;">
PROD.AMD
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.180
</td>
<td style="text-align:center;">
10.542
</td>
</tr>
<tr>
<td style="text-align:center;">
PROD.AMD
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.197
</td>
<td style="text-align:center;">
10.518
</td>
</tr>
<tr>
<td style="text-align:center;">
PROD.AMD
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.149
</td>
<td style="text-align:center;">
10.582
</td>
</tr>
<tr>
<td style="text-align:center;">
AP
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.055
</td>
<td style="text-align:center;">
10.742
</td>
</tr>
<tr>
<td style="text-align:center;">
AP
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.040
</td>
<td style="text-align:center;">
10.745
</td>
</tr>
<tr>
<td style="text-align:center;">
AP
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.055
</td>
<td style="text-align:center;">
10.742
</td>
</tr>
<tr>
<td style="text-align:center;">
AP
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.090
</td>
<td style="text-align:center;">
10.732
</td>
</tr>
<tr>
<td style="text-align:center;">
AP
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.074
</td>
<td style="text-align:center;">
10.736
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.122
</td>
<td style="text-align:center;">
13.563
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.116
</td>
<td style="text-align:center;">
13.747
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.130
</td>
<td style="text-align:center;">
13.678
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.138
</td>
<td style="text-align:center;">
13.020
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.125
</td>
<td style="text-align:center;">
13.677
</td>
</tr>
<tr>
<td style="text-align:center;">
AMD
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.238
</td>
<td style="text-align:center;">
10.170
</td>
</tr>
<tr>
<td style="text-align:center;">
AMD
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.246
</td>
<td style="text-align:center;">
10.107
</td>
</tr>
<tr>
<td style="text-align:center;">
AMD
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.263
</td>
<td style="text-align:center;">
10.028
</td>
</tr>
<tr>
<td style="text-align:center;">
AMD
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.301
</td>
<td style="text-align:center;">
9.783
</td>
</tr>
<tr>
<td style="text-align:center;">
AMD
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.205
</td>
<td style="text-align:center;">
10.410
</td>
</tr>
<tr>
<td style="text-align:center;">
Porte
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.074
</td>
<td style="text-align:center;">
10.873
</td>
</tr>
<tr>
<td style="text-align:center;">
Porte
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.107
</td>
<td style="text-align:center;">
10.914
</td>
</tr>
<tr>
<td style="text-align:center;">
Porte
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.097
</td>
<td style="text-align:center;">
10.910
</td>
</tr>
<tr>
<td style="text-align:center;">
Porte
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.082
</td>
<td style="text-align:center;">
10.881
</td>
</tr>
<tr>
<td style="text-align:center;">
Porte
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.103
</td>
<td style="text-align:center;">
10.915
</td>
</tr>
<tr>
<td style="text-align:center;">
RF
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.153
</td>
<td style="text-align:center;">
10.703
</td>
</tr>
<tr>
<td style="text-align:center;">
RF
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.147
</td>
<td style="text-align:center;">
10.705
</td>
</tr>
<tr>
<td style="text-align:center;">
RF
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.145
</td>
<td style="text-align:center;">
10.703
</td>
</tr>
<tr>
<td style="text-align:center;">
RF
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.147
</td>
<td style="text-align:center;">
10.706
</td>
</tr>
<tr>
<td style="text-align:center;">
RF
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.152
</td>
<td style="text-align:center;">
10.705
</td>
</tr>
<tr>
<td style="text-align:center;">
CR
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.115
</td>
<td style="text-align:center;">
10.777
</td>
</tr>
<tr>
<td style="text-align:center;">
CR
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.123
</td>
<td style="text-align:center;">
10.755
</td>
</tr>
<tr>
<td style="text-align:center;">
CR
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.142
</td>
<td style="text-align:center;">
10.656
</td>
</tr>
<tr>
<td style="text-align:center;">
CR
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.144
</td>
<td style="text-align:center;">
10.629
</td>
</tr>
<tr>
<td style="text-align:center;">
CR
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.146
</td>
<td style="text-align:center;">
10.631
</td>
</tr>
<tr>
<td style="text-align:center;">
DR
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.125
</td>
<td style="text-align:center;">
10.849
</td>
</tr>
<tr>
<td style="text-align:center;">
DR
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.114
</td>
<td style="text-align:center;">
10.931
</td>
</tr>
<tr>
<td style="text-align:center;">
DR
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.116
</td>
<td style="text-align:center;">
10.973
</td>
</tr>
<tr>
<td style="text-align:center;">
DR
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.138
</td>
<td style="text-align:center;">
10.795
</td>
</tr>
<tr>
<td style="text-align:center;">
DR
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.108
</td>
<td style="text-align:center;">
11.022
</td>
</tr>
<tr>
<td style="text-align:center;">
DCaule
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.115
</td>
<td style="text-align:center;">
10.711
</td>
</tr>
<tr>
<td style="text-align:center;">
DCaule
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.097
</td>
<td style="text-align:center;">
10.719
</td>
</tr>
<tr>
<td style="text-align:center;">
DCaule
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.150
</td>
<td style="text-align:center;">
10.696
</td>
</tr>
<tr>
<td style="text-align:center;">
DCaule
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.132
</td>
<td style="text-align:center;">
10.703
</td>
</tr>
<tr>
<td style="text-align:center;">
DCaule
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.115
</td>
<td style="text-align:center;">
10.712
</td>
</tr>
<tr>
<td style="text-align:center;">
Ácaro
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.221
</td>
<td style="text-align:center;">
10.875
</td>
</tr>
<tr>
<td style="text-align:center;">
Ácaro
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.189
</td>
<td style="text-align:center;">
10.863
</td>
</tr>
<tr>
<td style="text-align:center;">
Ácaro
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.204
</td>
<td style="text-align:center;">
10.867
</td>
</tr>
<tr>
<td style="text-align:center;">
Ácaro
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.239
</td>
<td style="text-align:center;">
10.883
</td>
</tr>
<tr>
<td style="text-align:center;">
Ácaro
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.224
</td>
<td style="text-align:center;">
10.877
</td>
</tr>
<tr>
<td style="text-align:center;">
Nº Hastes
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.028
</td>
<td style="text-align:center;">
10.745
</td>
</tr>
<tr>
<td style="text-align:center;">
Nº Hastes
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.066
</td>
<td style="text-align:center;">
10.717
</td>
</tr>
<tr>
<td style="text-align:center;">
Nº Hastes
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.040
</td>
<td style="text-align:center;">
10.736
</td>
</tr>
<tr>
<td style="text-align:center;">
Nº Hastes
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.061
</td>
<td style="text-align:center;">
10.721
</td>
</tr>
<tr>
<td style="text-align:center;">
Nº Hastes
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.059
</td>
<td style="text-align:center;">
10.722
</td>
</tr>
<tr>
<td style="text-align:center;">
Stand_Final
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.047
</td>
<td style="text-align:center;">
10.733
</td>
</tr>
<tr>
<td style="text-align:center;">
Stand_Final
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.076
</td>
<td style="text-align:center;">
10.715
</td>
</tr>
<tr>
<td style="text-align:center;">
Stand_Final
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.056
</td>
<td style="text-align:center;">
10.728
</td>
</tr>
<tr>
<td style="text-align:center;">
Stand_Final
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.028
</td>
<td style="text-align:center;">
10.744
</td>
</tr>
<tr>
<td style="text-align:center;">
Stand_Final
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.019
</td>
<td style="text-align:center;">
10.750
</td>
</tr>
<tr>
<td style="text-align:center;">
Branching_Level
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.055
</td>
<td style="text-align:center;">
10.837
</td>
</tr>
<tr>
<td style="text-align:center;">
Branching_Level
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.073
</td>
<td style="text-align:center;">
10.853
</td>
</tr>
<tr>
<td style="text-align:center;">
Branching_Level
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.049
</td>
<td style="text-align:center;">
10.830
</td>
</tr>
<tr>
<td style="text-align:center;">
Branching_Level
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.099
</td>
<td style="text-align:center;">
10.880
</td>
</tr>
<tr>
<td style="text-align:center;">
Branching_Level
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
-0.057
</td>
<td style="text-align:center;">
10.838
</td>
</tr>
<tr>
<td style="text-align:center;">
Staygreen
</td>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.134
</td>
<td style="text-align:center;">
10.750
</td>
</tr>
<tr>
<td style="text-align:center;">
Staygreen
</td>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.085
</td>
<td style="text-align:center;">
10.751
</td>
</tr>
<tr>
<td style="text-align:center;">
Staygreen
</td>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.086
</td>
<td style="text-align:center;">
10.751
</td>
</tr>
<tr>
<td style="text-align:center;">
Staygreen
</td>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.111
</td>
<td style="text-align:center;">
10.750
</td>
</tr>
<tr>
<td style="text-align:center;">
Staygreen
</td>
<td style="text-align:center;">
5
</td>
<td style="text-align:center;">
TRUE
</td>
<td style="text-align:center;">
0.087
</td>
<td style="text-align:center;">
10.751
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(results_cv, <span class="st">&quot;output/results_cv.csv&quot;</span>)</span></code></pre></div>
<p>The object “result_cv” is divided by repetition. In the “result_cv”
objects for each repetition, “Rep” is the number of the repetition,
“Log” is a diagnostic indicating if the order of the predicted breeding
values matches the order of the adjusted means, “Ac” is the prediction
accuracy (correlation between the GEBV and adjusted means), and “MSPE”
is the mean square prediction error (the lower, the better).</p>
<p>Agora vamos plotar os resultados de acurácias para cada
característica</p>
<div class="sourceCode" id="cb40"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>results_cv <span class="sc">%&gt;%</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Trait) <span class="sc">%&gt;%</span></span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean.Ac =</span> <span class="fu">mean</span>(Ac),</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">sd.Ac =</span> <span class="fu">sd</span>(Ac),</span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">mean.MSPE =</span> <span class="fu">mean</span>(MSPE),</span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">sd.MSPE =</span> <span class="fu">sd</span>(MSPE)</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Trait, <span class="at">y =</span> mean.Ac , <span class="at">fill =</span> Trait)) <span class="sc">+</span></span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>,</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>           <span class="at">width =</span> <span class="fl">0.85</span>,</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>           <span class="at">show.legend =</span> F) <span class="sc">+</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">ymin =</span> mean.Ac <span class="sc">-</span> sd.Ac, <span class="at">ymax =</span> mean.Ac <span class="sc">+</span> sd.Ac),</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> .<span class="dv">2</span>,</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_dodge</span>(.<span class="dv">9</span>)</span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ylim</span>(<span class="sc">-</span><span class="fl">0.3</span>,<span class="fl">0.3</span>)<span class="sc">+</span></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">&quot;Accuracy&quot;</span>)</span></code></pre></div>
<p><img src="figure/GWS.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb41"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.1.3 (2022-03-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19042)

Matrix products: default

locale:
[1] LC_COLLATE=Portuguese_Brazil.1252  LC_CTYPE=Portuguese_Brazil.1252   
[3] LC_MONETARY=Portuguese_Brazil.1252 LC_NUMERIC=C                      
[5] LC_TIME=Portuguese_Brazil.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] cvTools_0.3.2            robustbase_0.95-0        lattice_0.20-45         
 [4] ggpmisc_0.5.1            ggpp_0.4.5               rrBLUP_4.6.1            
 [7] ComplexHeatmap_2.10.0    AGHmatrix_2.0.4          genomicMateSelectR_0.2.0
[10] janitor_2.1.0            kableExtra_1.3.4         forcats_0.5.2           
[13] stringr_1.4.1            dplyr_1.0.10             purrr_0.3.4             
[16] readr_2.1.2              tidyr_1.2.1              tibble_3.1.8            
[19] ggplot2_3.3.6            tidyverse_1.3.2          readxl_1.4.1            

loaded via a namespace (and not attached):
  [1] googledrive_2.0.0   colorspace_2.0-3    rjson_0.2.21       
  [4] ellipsis_0.3.2      rprojroot_2.0.3     circlize_0.4.15    
  [7] snakecase_0.11.0    GlobalOptions_0.1.2 fs_1.5.2           
 [10] clue_0.3-61         rstudioapi_0.14     farver_2.1.1       
 [13] MatrixModels_0.5-1  fansi_1.0.3         lubridate_1.8.0    
 [16] xml2_1.3.3          splines_4.1.3       codetools_0.2-18   
 [19] doParallel_1.0.17   cachem_1.0.6        confintr_0.2.0     
 [22] knitr_1.40          polynom_1.4-1       jsonlite_1.8.0     
 [25] workflowr_1.7.0     broom_1.0.1         cluster_2.1.2      
 [28] dbplyr_2.2.1        png_0.1-7           compiler_4.1.3     
 [31] httr_1.4.4          backports_1.4.1     assertthat_0.2.1   
 [34] Matrix_1.5-1        fastmap_1.1.0       gargle_1.2.1       
 [37] cli_3.3.0           later_1.3.0         htmltools_0.5.3    
 [40] quantreg_5.94       tools_4.1.3         gtable_0.3.1       
 [43] glue_1.6.2          Rcpp_1.0.9          cellranger_1.1.0   
 [46] jquerylib_0.1.4     vctrs_0.4.1         nlme_3.1-159       
 [49] svglite_2.1.0       iterators_1.0.14    xfun_0.32          
 [52] rvest_1.0.3         lifecycle_1.0.3     googlesheets4_1.0.1
 [55] DEoptimR_1.0-11     MASS_7.3-58.1       zoo_1.8-11         
 [58] scales_1.2.1        hms_1.1.2           promises_1.2.0.1   
 [61] parallel_4.1.3      SparseM_1.81        RColorBrewer_1.1-3 
 [64] yaml_2.3.5          sass_0.4.2          stringi_1.7.6      
 [67] highr_0.9           S4Vectors_0.32.4    foreach_1.5.2      
 [70] BiocGenerics_0.40.0 boot_1.3-28         shape_1.4.6        
 [73] rlang_1.0.6         pkgconfig_2.0.3     systemfonts_1.0.4  
 [76] matrixStats_0.62.0  evaluate_0.17       labeling_0.4.2     
 [79] tidyselect_1.2.0    magrittr_2.0.3      R6_2.5.1           
 [82] magick_2.7.3        IRanges_2.28.0      generics_0.1.3     
 [85] DBI_1.1.3           mgcv_1.8-40         pillar_1.8.1       
 [88] haven_2.5.1         withr_2.5.0         survival_3.4-0     
 [91] modelr_0.1.9        crayon_1.5.2        utf8_1.2.2         
 [94] tzdb_0.3.0          rmarkdown_2.17      GetoptLong_1.0.5   
 [97] git2r_0.30.1        reprex_2.0.2        digest_0.6.29      
[100] webshot_0.5.4       httpuv_1.6.5        stats4_4.1.3       
[103] munsell_0.5.0       viridisLite_0.4.1   bslib_0.4.0        </code></pre>
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Weverton Gomes da Costa, Pós-Doutorando, Embrapa
Mandioca e Fruticultura, <a href="mailto:wevertonufv@gmail.com"
class="email">wevertonufv@gmail.com</a><a href="#fnref1"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
