<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Weverton Gomes da Costa" />

<meta name="date" content="2025-01-13" />

<title>Genomic Selection for Drought Tolerance</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomic Selection for Drought Tolerance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="phenotype.html">Phenotype</a>
</li>
<li>
  <a href="mixed_models.html">GxE Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genomic Selection
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="GWS.html">Overview</a>
    </li>
    <li>
      <a href="GWS_RR-BLUP.html">RR-BLUP</a>
    </li>
    <li>
      <a href="GWS_G-BLUP.html">G-BLUP (additive and additive-dominant)</a>
    </li>
    <li>
      <a href="GWS_BayesA.html">Bayes A</a>
    </li>
    <li>
      <a href="GWS_BayesB.html">Bayes B</a>
    </li>
    <li>
      <a href="GWS_RKHS.html">RKHS</a>
    </li>
    <li>
      <a href="GWS_RF.html">Random Forest</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Genomic Selection for Drought
Tolerance</h1>
<h4 class="author">Weverton Gomes da Costa<a href="#fn1"
class="footnote-ref" id="fnref1"><sup>1</sup></a></h4>
<h4 class="date">2025-01-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221020code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221020)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221020code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221020)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree0d2a1ede387c973268e882a7379670a600206618targetblank0d2a1eda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/0d2a1ede387c973268e882a7379670a600206618" target="_blank">0d2a1ed</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree0d2a1ede387c973268e882a7379670a600206618targetblank0d2a1eda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/0d2a1ede387c973268e882a7379670a600206618" target="_blank">0d2a1ed</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Artigo GS tolerancia a seca_revised.rar
    Ignored:    Seleção Genômica deficit hidrico.pptx
    Ignored:    data/allchrAR08.txt

Untracked files:
    Untracked:  Artigo GS tolerancia a seca_revised/
    Untracked:  ETA_1_ScaleBayesA.dat
    Untracked:  ETA_1_parBayesB.dat
    Untracked:  data/Article.docx
    Untracked:  data/Article2.docx
    Untracked:  data/Articles/
    Untracked:  data/Costa et al_FPS_2024 manuscript eng.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript eng_v1.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript eng_v2.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript eng_v3.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript.docx
    Untracked:  data/Costa et al_FPS_2024 supplement.docx
    Untracked:  data/Frontiers_Template.docx
    Untracked:  data/Phenotyping2.xlsx
    Untracked:  data/Supplementary_Material.docx
    Untracked:  data/Tabela auxiliar.xlsx
    Untracked:  data/dados__temp_umi.csv
    Untracked:  data/data.rar
    Untracked:  mu.dat
    Untracked:  output/BLUPS2.csv
    Untracked:  output/BLUPS_GEBV_GETG_all clones.csv
    Untracked:  output/BLUPS_density_med.png
    Untracked:  output/BLUPS_density_med_row_col.png
    Untracked:  output/BLUPS_drgBLUPs__boxplot_med.png
    Untracked:  output/BLUPS_drgBLUPs__boxplot_med_row_col.png
    Untracked:  output/BLUPS_drgBLUPs_density_med.png
    Untracked:  output/BLUPS_drgBLUPs_med.png
    Untracked:  output/BLUPS_par_mmer.Rdata
    Untracked:  output/BLUPS_row_col.csv
    Untracked:  output/BLUPS_row_col_random.csv
    Untracked:  output/BLUPS_x_BLUPS_row_col_boxplot_med.png
    Untracked:  output/Density_residual_row_col.tiff
    Untracked:  output/Figuras_article.rar
    Untracked:  output/GEBVS_BayesA.RDS
    Untracked:  output/GEBVS_BayesB.RDS
    Untracked:  output/GEBVS_DOM.RDS
    Untracked:  output/GEBVS_G_BLUP.RDS
    Untracked:  output/GEBVS_G_BLUP_row_col.RDS
    Untracked:  output/GEBVS_G_BLUP_row_col_random.RDS
    Untracked:  output/GEBVS_RF.RDS
    Untracked:  output/GEBVS_RKHS.RDS
    Untracked:  output/GEBVS_RR_BLUP.RDS
    Untracked:  output/GEBV_BLUP.csv
    Untracked:  output/GEBV_DOMxBLUPS.tiff
    Untracked:  output/GEBV_GBLUPxBLUPS.tiff
    Untracked:  output/GEBVxGETGV.tiff
    Untracked:  output/GETGVxBLUPS.tiff
    Untracked:  output/G_matrix.rds
    Untracked:  output/H2.csv
    Untracked:  output/H2_row_col.csv
    Untracked:  output/H2_row_col_random.csv
    Untracked:  output/Heatmap.tiff
    Untracked:  output/MSPE_all_methods.tiff
    Untracked:  output/Residuals_vs_fitted_row_col.tiff
    Untracked:  output/accuracy_all_methods.tiff
    Untracked:  output/accuracy_all_methods_points.tiff
    Untracked:  output/cor_GEBVxBLUPS.tiff
    Untracked:  output/cor_GEBVxBLUPS_all_methods.tiff
    Untracked:  output/correlation_blups.tiff
    Untracked:  output/density_blups.tiff
    Untracked:  output/dierencial_selecao.csv
    Untracked:  output/dierencial_selecao.xlsx
    Untracked:  output/drgBLUP.csv
    Untracked:  output/fitted_residual_data_row_col_random.csv
    Untracked:  output/fitted_values_row_col_random.csv
    Untracked:  output/indice_selection.tiff
    Untracked:  output/indice_selection3.tiff
    Untracked:  output/indice_selection_GEBV.tiff
    Untracked:  output/indice_selection_GEBV_GETGV.tiff
    Untracked:  output/indice_selection_GETGV.tiff
    Untracked:  output/kappa.tiff
    Untracked:  output/kappa2.tiff
    Untracked:  output/mean_pheno.csv
    Untracked:  output/medias_semestre.xlsx
    Untracked:  output/pesos_BLUPS_all clones.csv
    Untracked:  output/pesos_GEBVS_all clones.csv
    Untracked:  output/pesos_GETGVS_all clones.csv
    Untracked:  output/pheno_mean_sd.csv
    Untracked:  output/residuos_row_col_random.csv
    Untracked:  output/result_sommer.RData
    Untracked:  output/result_sommer_row_col.RData
    Untracked:  output/result_sommer_row_col_random.RDS
    Untracked:  output/result_sommer_row_col_random_optimized.RDS
    Untracked:  output/results_MSPE.csv
    Untracked:  output/results_accuracy.csv
    Untracked:  output/results_cv_BayesA.RDS
    Untracked:  output/results_cv_BayesB.RDS
    Untracked:  output/results_cv_DOM.RDS
    Untracked:  output/results_cv_G_BLUP.RDS
    Untracked:  output/results_cv_G_BLUP_row_col.RDS
    Untracked:  output/results_cv_G_BLUP_row_col_random.RDS
    Untracked:  output/results_cv_RF.RDS
    Untracked:  output/results_cv_RKHS.RDS
    Untracked:  output/results_cv_RR_BLUP.RDS
    Untracked:  output/results_h2_GBLUP (1).rds
    Untracked:  output/results_h2_GBLUP.csv
    Untracked:  output/results_h2_GBLUP.rds
    Untracked:  output/results_h2_GBLUP_D.rds
    Untracked:  output/results_kappa.csv
    Untracked:  output/results_kappa2.csv
    Untracked:  output/results_values-selection.csv
    Untracked:  output/results_values_kappa.csv
    Untracked:  output/results_values_rescale_selection.csv
    Untracked:  output/results_values_selection.csv
    Untracked:  output/rstudio-export.zip
    Untracked:  output/semester_means.xlsx
    Untracked:  output/teste_LRT.csv
    Untracked:  output/teste_LRT_row_col.csv
    Untracked:  output/teste_LRT_row_col_random.csv
    Untracked:  output/varcomp.tiff
    Untracked:  output/varcomp2.csv
    Untracked:  output/varcomp_row_col.csv
    Untracked:  output/varcomp_row_col.tiff
    Untracked:  output/varcomp_row_col_random.csv
    Untracked:  output/varcomp_row_col_random.tiff
    Untracked:  varE.dat

Unstaged changes:
    Modified:   Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava.Rproj
    Modified:   analysis/GWS.Rmd
    Modified:   analysis/index.Rmd
    Modified:   data/SNPs.rds
    Modified:   data/pheno_clean.csv
    Modified:   output/BLUPS.csv
    Modified:   output/BLUPS_par.Rdata
    Modified:   output/media_pheno.csv
    Modified:   output/varcomp.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/index.Rmd</code>) and HTML
(<code>docs/index.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/21def5758a5e07c1c5a402bb7820b36cee3ff92a/docs/index.html" target="_blank">21def57</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2025-01-13
</td>
<td>
update index and phenotype.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/analysis/index.Rmd" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
<td>
Update Index and phenotype
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/54c6e2809526414fcf21dca7ea89db78aa100956/analysis/index.Rmd" target="_blank">54c6e28</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-07
</td>
<td>
update site, index.rmd, license.rmd, phenotype.rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/02143bc96edd0f7c8de094f8f8f8b8e9356042e3/docs/index.html" target="_blank">02143bc</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
add about, index and license html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/286b4922972c00dd9062e8c6a1848a867118378a/analysis/index.Rmd" target="_blank">286b492</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
Update Scripts and README
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/90dc11276f860bdf4c29fc2cbf7a7d66d049d277/docs/index.html" target="_blank">90dc112</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/d930880aef03c824bb532ffc196a8ad2f9105d51/docs/index.html" target="_blank">d930880</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/5988c27353e95ed989eeda03e0ca0091cccc9bf0/analysis/index.Rmd" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/5988c27353e95ed989eeda03e0ca0091cccc9bf0/docs/index.html" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/analysis/index.Rmd" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/index.html" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/b78c842a0b90fe20a5a7c904c9efb5deb23fdcb5/analysis/index.Rmd" target="_blank">b78c842</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-10-20
</td>
<td>
Start workflowr project.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="genomic-selection-for-drought-tolerance"
class="section level1">
<h1>Genomic Selection for Drought Tolerance</h1>
<p>Welcome to the Genomic Selection for Drought Tolerance project
website! This initiative by EMBRAPA Mandioca focuses on utilizing Genome
Wide GBS and/or DART in Cassava to achieve drought tolerance through
genomic selection.</p>
<div id="about-the-project" class="section level2">
<h2>About the Project</h2>
<p>This project aims to analyze the phenotypic data of Brazilian drought
trials to evaluate the performance of various genotypes under drought
conditions. The goal is to identify the genotypes that exhibit superior
performance and resilience. The analysis follows a structured workflow,
including data import and manipulation, exploratory data analysis, and
genotype-environment analysis using mixed-effect models. This project
details the estimation of Best Linear Unbiased Predictions (BLUPs) using
mixed models, along with extensive Exploratory Data Analysis (EDA) and
necessary manipulations to accurately estimate the BLUPs.</p>
</div>
<div id="project-structure" class="section level2">
<h2>Project Structure</h2>
<div id="exploratory-data-analysis-eda" class="section level3">
<h3>1. <a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/phenotype.html">Exploratory
Data Analysis (EDA)</a></h3>
<p>We used packages such as DataExplorer, metan, and data.table to
perform extensive exploratory data analysis on our dataset. The goal was
to understand the data structure, identify missing values, and detect
patterns that might influence the subsequent analyses.</p>
</div>
<div id="genotype-x-environment-interaction-analysis-gxe"
class="section level3">
<h3>2. <a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/mixed_models.html">Genotype
x Environment Interaction Analysis (GxE)</a></h3>
<p>The GxE analysis aims to evaluate the performance of genotypes across
different environments to identify those with stable performance and
high resilience. We used mixed-effect models to estimate the variance
components and calculate BLUPs for each genotype.</p>
</div>
<div id="genomic-wide-selection-gws" class="section level3">
<h3>3. <a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS.html">Genomic
Wide Selection (GWS)</a></h3>
<p>For GWS, we employed the original marker matrix, followed by
necessary data manipulations to organize and prepare the matrix for
model input. We used various models, including:</p>
<ul>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RR-BLUP.html"><strong>RR-BLUP</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_G-BLUP.html"><strong>G-BLUP
(additive and additive-dominant)</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_BayesA.html"><strong>Bayes
A</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_BayesB.html"><strong>Bayes
B</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RKHS.html"><strong>RKHS</strong></a></li>
<li><a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RF.html"><strong>Random
Forest</strong></a></li>
</ul>
<p>(Note: Some models might be computationally intensive.)</p>
</div>
</div>
<div id="study-highlights" class="section level2">
<h2>Study Highlights</h2>
<ul>
<li><strong>Manuscript Access</strong>: <a
href="https://doi.org/10.3389/fpls.2024.1483340">Genomic Selection for
Drought Tolerance Using Genome Wide SNPs in Cassava</a></li>
<li><strong>Authors</strong>:
<ul>
<li>Weverton Gomes da Costa (Pós-Doutorando, Embrapa Mandioca e
Fruticultura) - <a href="mailto:weverton.costa@ufv.br"
class="email">weverton.costa@ufv.br</a></li>
</ul></li>
</ul>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=Portuguese_Brazil.utf8  LC_CTYPE=Portuguese_Brazil.utf8   
[3] LC_MONETARY=Portuguese_Brazil.utf8 LC_NUMERIC=C                      
[5] LC_TIME=Portuguese_Brazil.utf8    

time zone: America/Sao_Paulo
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] vctrs_0.6.3       cli_3.6.1         knitr_1.43        rlang_1.1.1      
 [5] xfun_0.40         stringi_1.7.12    promises_1.2.1    jsonlite_1.8.7   
 [9] workflowr_1.7.1   glue_1.6.2        rprojroot_2.0.3   git2r_0.32.0     
[13] htmltools_0.5.6   httpuv_1.6.11     sass_0.4.7        fansi_1.0.4      
[17] rmarkdown_2.24    tibble_3.2.1      evaluate_0.22     jquerylib_0.1.4  
[21] fastmap_1.1.1     yaml_2.3.7        lifecycle_1.0.3   whisker_0.4.1    
[25] stringr_1.5.0     compiler_4.3.1    fs_1.6.3          pkgconfig_2.0.3  
[29] Rcpp_1.0.11       rstudioapi_0.15.0 later_1.3.1       digest_0.6.33    
[33] R6_2.5.1          utf8_1.2.3        pillar_1.9.0      magrittr_2.0.3   
[37] bslib_0.5.1       tools_4.3.1       cachem_1.0.8     </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Pós-Doutorando, Embrapa Mandioca e Fruticultura, <a
href="mailto:weverton.costa@ufv.br"
class="email">weverton.costa@ufv.br</a><a href="#fnref1"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
