<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Costa, W. G." />

<meta name="date" content="2023-10-27" />

<title>Phenotype Analysis of Brazilian Drought Trials</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomic Selection for Drought Tolerance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="phenotype.html">Phenotype</a>
</li>
<li>
  <a href="GWS.html">Genomic Selection</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Phenotype Analysis of Brazilian Drought
Trials</h1>
<h4 class="author">Costa, W. G.<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a></h4>
<h4 class="date">2023-10-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-10-27
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong>
<code>Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221020code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221020)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221020code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221020)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree02143bc96edd0f7c8de094f8f8f8b8e9356042e3targetblank02143bca">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/02143bc96edd0f7c8de094f8f8f8b8e9356042e3" target="_blank">02143bc</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree02143bc96edd0f7c8de094f8f8f8b8e9356042e3targetblank02143bca"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/02143bc96edd0f7c8de094f8f8f8b8e9356042e3" target="_blank">02143bc</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/allchrAR08.txt

Untracked files:
    Untracked:  GETGV.RDS
    Untracked:  data/Article.docx
    Untracked:  data/Article2.docx
    Untracked:  data/Articles/
    Untracked:  output/BLUPS2.csv
    Untracked:  output/BLUPS3.csv
    Untracked:  output/BLUPS_drgBLUPs__boxplot_med.png
    Untracked:  output/BLUPS_drgBLUPs_density_med.png
    Untracked:  output/BLUPS_drgBLUPs_med.png
    Untracked:  output/BLUPS_par_mmer.Rdata
    Untracked:  output/Figuras_article.rar
    Untracked:  output/GD_matrix.csv
    Untracked:  output/GEBVS_G_BLUP_BGLR.RDS
    Untracked:  output/GEBVS_RR_BLUP.RDS
    Untracked:  output/GEBVxBLUPS.tiff
    Untracked:  output/GEBVxGETV.tiff
    Untracked:  output/GEBVxGETV2.tiff
    Untracked:  output/G_matrix.csv
    Untracked:  output/G_matrix.rds
    Untracked:  output/H2.csv
    Untracked:  output/Tabelas.xlsx
    Untracked:  output/accuracy_all_methods.tiff
    Untracked:  output/correlation_blups.tiff
    Untracked:  output/density_blups.tiff
    Untracked:  output/drgBLUP.csv
    Untracked:  output/indice_selection3.tiff
    Untracked:  output/kappa.tiff
    Untracked:  output/mean_pheno.csv
    Untracked:  output/result_mixed_model.RData
    Untracked:  output/result_mixed_model2.RData
    Untracked:  output/result_mixed_model3.RData
    Untracked:  output/result_sommer.RData
    Untracked:  output/results_cv_BayesA.csv
    Untracked:  output/results_cv_BayesB.csv
    Untracked:  output/results_cv_DOM.csv
    Untracked:  output/results_cv_GETGV.csv
    Untracked:  output/results_cv_G_BLUP.csv
    Untracked:  output/results_cv_G_BLUP2.csv
    Untracked:  output/results_cv_G_BLUP_BGRL.csv
    Untracked:  output/results_cv_G_BLUP_BLUPS2.csv
    Untracked:  output/results_cv_G_BLUP_BLUPS3.csv
    Untracked:  output/results_cv_G_BLUP_sommer.csv
    Untracked:  output/results_cv_RF.csv
    Untracked:  output/results_cv_RKHS.csv
    Untracked:  output/results_cv_RR_BLUP.csv
    Untracked:  output/results_kappa.csv
    Untracked:  output/results_kappa2.csv
    Untracked:  output/varcomp.tiff
    Untracked:  output/varcomp2.csv

Unstaged changes:
    Deleted:    data/Genomic Selection for Drought Tolerance Using Genome-Wide SNPs in Maize.pdf
    Deleted:    data/convet_haplo_diplo.txt
    Deleted:    output/BLUPS.RDS
    Modified:   output/BLUPS.csv
    Deleted:    output/BLUPS_Multi.csv
    Modified:   output/BLUPS_par.Rdata
    Deleted:    output/BLUPS_par_ind.Rdata
    Deleted:    output/H2_Multi.csv
    Deleted:    output/accuracy.png
    Deleted:    output/accuracy2.png
    Deleted:    output/accuracy_multi.png
    Deleted:    output/blups_med.png
    Deleted:    output/herdabilidade.csv
    Modified:   output/media_pheno.csv
    Deleted:    output/media_pheno_Multi.csv
    Deleted:    output/resultMM.Rdata
    Deleted:    output/result_G_BLUP.csv
    Modified:   output/results_cv.csv
    Modified:   output/varcomp.csv
    Deleted:    output/varcomp_multi.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/phenotype.Rmd</code>) and HTML
(<code>docs/phenotype.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/286b4922972c00dd9062e8c6a1848a867118378a/analysis/phenotype.Rmd" target="_blank">286b492</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
Update Scripts and README
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/90dc11276f860bdf4c29fc2cbf7a7d66d049d277/analysis/phenotype.Rmd" target="_blank">90dc112</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/6cc4d2326a79681446b1ea653122061a959c2ffe/analysis/phenotype.Rmd" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/6cc4d2326a79681446b1ea653122061a959c2ffe/docs/phenotype.html" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/d930880aef03c824bb532ffc196a8ad2f9105d51/docs/phenotype.html" target="_blank">d930880</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/5988c27353e95ed989eeda03e0ca0091cccc9bf0/analysis/phenotype.Rmd" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/5988c27353e95ed989eeda03e0ca0091cccc9bf0/docs/phenotype.html" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/analysis/phenotype.Rmd" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/phenotype.html" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-and-libraries" class="section level2">
<h2>Data and libraries</h2>
<p>Load Libraries</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">require</span>(ComplexHeatmap)</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">library</span>(metan)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(DataExplorer)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="fu">library</span>(furrr)</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(future)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">plan</span>(multisession)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span></code></pre></div>
<div id="data-import-and-manipulation" class="section level3">
<h3>Data import and manipulation</h3>
<p>Let’s import the phenotypic dataset, excluding the traits without
information and the traits Local (redundant with Year) and Treatment
(only one observation).</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Phenotyping.xlsx&quot;</span>, <span class="at">na =</span> <span class="st">&quot;NA&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="fu">select_if</span>( <span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span>  <span class="co"># Deleting traits without information </span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;Local&quot;</span>, <span class="st">&quot;Tratamento&quot;</span>))</span></code></pre></div>
<p>We will perform some manipulations to adjust our database and to
facilitate the visualization of the exploratory analysis.</p>
<p>First, let’s convert the traits that are character into factors. Then
we will convert the traits that refer to the grades to integers and then
into factors. After that, let’s create the trait ANo.Bloco for nesting
in the model to obtain the BLUPs.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Clone =</span> <span class="fu">as.factor</span>(Clone),</span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a>         <span class="at">Ano =</span> <span class="fu">as.factor</span>(Ano),</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a>         <span class="at">Bloco =</span> <span class="fu">as.factor</span>(Bloco))   <span class="co"># Convert Clone, Ano,Bloco in factors</span></span></code></pre></div>
</div>
</div>
<div id="exploratory-data-analysis" class="section level2">
<h2>Exploratory Data Analysis</h2>
<p>Introductory analysis of the entire dataset</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">introduce</span>(pheno) <span class="sc">%&gt;%</span> </span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb4-8"><a href="#cb4-8" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb4-9"><a href="#cb4-9" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
rows
</td>
<td style="text-align:center;">
2336
</td>
</tr>
<tr>
<td style="text-align:left;">
columns
</td>
<td style="text-align:center;">
27
</td>
</tr>
<tr>
<td style="text-align:left;">
discrete_columns
</td>
<td style="text-align:center;">
3
</td>
</tr>
<tr>
<td style="text-align:left;">
continuous_columns
</td>
<td style="text-align:center;">
24
</td>
</tr>
<tr>
<td style="text-align:left;">
all_missing_columns
</td>
<td style="text-align:center;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
total_missing_values
</td>
<td style="text-align:center;">
16875
</td>
</tr>
<tr>
<td style="text-align:left;">
complete_rows
</td>
<td style="text-align:center;">
440
</td>
</tr>
<tr>
<td style="text-align:left;">
total_observations
</td>
<td style="text-align:center;">
63072
</td>
</tr>
<tr>
<td style="text-align:left;">
memory_usage
</td>
<td style="text-align:center;">
516592
</td>
</tr>
</tbody>
</table>
<p>We don’t have any columns that have all of the missing observations,
but we do have a lot of missing values in every dataset. Some
manipulations should be performed to improve the quality of the
data.</p>
<div id="year-analysis" class="section level3">
<h3>Year Analysis</h3>
<p>Let’s produce a heatmap to check the clone amount each year. I’m
going to create another dataset with the Year and Clone count. Then I
will create the objects corresponding to the clones and years array.
Finally, I created the matrix that represents the presence and absence
of the clone in the year.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>pheno2 <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>genmat <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Clone, <span class="at">data =</span> pheno2)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>envmat <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Ano, <span class="at">data =</span> pheno2)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>genenvmat <span class="ot">&lt;-</span> <span class="fu">t</span>(envmat) <span class="sc">%*%</span> genmat</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>genenvmat_ch <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(genenvmat <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Abscent&quot;</span>)</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>  genenvmat_ch,</span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;tomato&quot;</span>),</span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>  <span class="at">show_column_names =</span> F,</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>  <span class="at">column_title =</span> <span class="st">&quot;Genotypes&quot;</span>,</span>
<span id="cb5-15"><a href="#cb5-15" tabindex="-1"></a>  <span class="at">row_title =</span> <span class="st">&quot;Environments&quot;</span></span>
<span id="cb5-16"><a href="#cb5-16" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-5-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">rm</span>(pheno2, genmat, envmat, genenvmat, genenvmat_ch)</span></code></pre></div>
<p>From the heatmap, it is clear that the year 2016 has very few
observations. So, we must eliminate it.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> </span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">filter</span>(Ano <span class="sc">!=</span> <span class="dv">2016</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>  <span class="fu">droplevels</span>()</span></code></pre></div>
<p>Just for reference, let’s re-view the clone heatmap by year.</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>pheno2<span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> </span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone)</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  </span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>genmat <span class="ot">=</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Clone, <span class="at">data =</span> pheno2)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>envmat <span class="ot">=</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Ano, <span class="at">data =</span> pheno2)</span>
<span id="cb8-6"><a href="#cb8-6" tabindex="-1"></a>genenvmat <span class="ot">=</span> <span class="fu">t</span>(envmat) <span class="sc">%*%</span> genmat</span>
<span id="cb8-7"><a href="#cb8-7" tabindex="-1"></a>genenvmat_ch <span class="ot">=</span> <span class="fu">ifelse</span>(genenvmat <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Abscent&quot;</span>)</span>
<span id="cb8-8"><a href="#cb8-8" tabindex="-1"></a></span>
<span id="cb8-9"><a href="#cb8-9" tabindex="-1"></a><span class="fu">Heatmap</span>(</span>
<span id="cb8-10"><a href="#cb8-10" tabindex="-1"></a>  genenvmat_ch,</span>
<span id="cb8-11"><a href="#cb8-11" tabindex="-1"></a>  <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;tomato&quot;</span>),</span>
<span id="cb8-12"><a href="#cb8-12" tabindex="-1"></a>  <span class="at">show_column_names =</span> F,</span>
<span id="cb8-13"><a href="#cb8-13" tabindex="-1"></a>  <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb8-14"><a href="#cb8-14" tabindex="-1"></a>  <span class="at">column_title =</span> <span class="st">&quot;Genotypes&quot;</span>,</span>
<span id="cb8-15"><a href="#cb8-15" tabindex="-1"></a>  <span class="at">row_title =</span> <span class="st">&quot;Environments&quot;</span></span>
<span id="cb8-16"><a href="#cb8-16" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-7-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a><span class="fu">rm</span>(pheno2, genmat, envmat, genenvmat, genenvmat_ch)</span></code></pre></div>
<p>We can check how many clones we have in common between the years and
also note that the years differ in the number of clones evaluated:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>pheno2<span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> </span>
<span id="cb10-2"><a href="#cb10-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone)</span>
<span id="cb10-3"><a href="#cb10-3" tabindex="-1"></a>  </span>
<span id="cb10-4"><a href="#cb10-4" tabindex="-1"></a>genmat <span class="ot">=</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Clone, <span class="at">data =</span> pheno2)</span>
<span id="cb10-5"><a href="#cb10-5" tabindex="-1"></a>envmat <span class="ot">=</span> <span class="fu">model.matrix</span>( <span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Ano, <span class="at">data =</span> pheno2)</span>
<span id="cb10-6"><a href="#cb10-6" tabindex="-1"></a>genenvmat <span class="ot">=</span> <span class="fu">t</span>(envmat) <span class="sc">%*%</span> genmat</span>
<span id="cb10-7"><a href="#cb10-7" tabindex="-1"></a></span>
<span id="cb10-8"><a href="#cb10-8" tabindex="-1"></a>genenvmat <span class="sc">%*%</span> <span class="fu">t</span>(genenvmat) <span class="sc">%&gt;%</span></span>
<span id="cb10-9"><a href="#cb10-9" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F,</span>
<span id="cb10-10"><a href="#cb10-10" tabindex="-1"></a>      <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb10-11"><a href="#cb10-11" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb10-12"><a href="#cb10-12" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb10-13"><a href="#cb10-13" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb10-14"><a href="#cb10-14" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb10-15"><a href="#cb10-15" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb10-16"><a href="#cb10-16" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2017
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2018
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2019
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2020
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Ano2017
</td>
<td style="text-align:center;">
165
</td>
<td style="text-align:center;">
42
</td>
<td style="text-align:center;">
22
</td>
<td style="text-align:center;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano2018
</td>
<td style="text-align:center;">
42
</td>
<td style="text-align:center;">
138
</td>
<td style="text-align:center;">
39
</td>
<td style="text-align:center;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano2019
</td>
<td style="text-align:center;">
22
</td>
<td style="text-align:center;">
39
</td>
<td style="text-align:center;">
133
</td>
<td style="text-align:center;">
29
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano2020
</td>
<td style="text-align:center;">
14
</td>
<td style="text-align:center;">
16
</td>
<td style="text-align:center;">
29
</td>
<td style="text-align:center;">
138
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a><span class="fu">rm</span>(pheno2, genmat, envmat, genenvmat)</span></code></pre></div>
<p>The year 2020 has a lower number of clones in common, however, we
will keep it for the analysis.</p>
<p>Here, it is possible to observe that our dataset has clones that were
evaluated in just one year. Let’s visualize this, to see how many clones
were evaluated according to the number of years.</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>pheno <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone) <span class="sc">%&gt;%</span> </span>
<span id="cb12-3"><a href="#cb12-3" tabindex="-1"></a>  <span class="fu">count</span>(Clone) <span class="sc">%&gt;%</span></span>
<span id="cb12-4"><a href="#cb12-4" tabindex="-1"></a>  <span class="fu">count</span>(n) <span class="sc">%&gt;%</span></span>
<span id="cb12-5"><a href="#cb12-5" tabindex="-1"></a>  <span class="fu">kbl</span>(</span>
<span id="cb12-6"><a href="#cb12-6" tabindex="-1"></a>    <span class="at">escape =</span> F,</span>
<span id="cb12-7"><a href="#cb12-7" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">&#39;c&#39;</span>,</span>
<span id="cb12-8"><a href="#cb12-8" tabindex="-1"></a>    <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;N of Environment&quot;</span>, <span class="st">&quot;N of genotypes&quot;</span>)</span>
<span id="cb12-9"><a href="#cb12-9" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb12-10"><a href="#cb12-10" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb12-11"><a href="#cb12-11" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb12-12"><a href="#cb12-12" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb12-13"><a href="#cb12-13" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb12-14"><a href="#cb12-14" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb12-15"><a href="#cb12-15" tabindex="-1"></a>  )</span></code></pre></div>
<pre><code>Storing counts in `nn`, as `n` already present in input
ℹ Use `name = &quot;new_name&quot;` to pick a new name.</code></pre>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
N of Environment
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
N of genotypes
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
350
</td>
</tr>
<tr>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
72
</td>
</tr>
<tr>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
20
</td>
</tr>
<tr>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
5
</td>
</tr>
</tbody>
</table>
<p>Only 5 clones were evaluated in all years, this will possibly
decrease our model accuracy.</p>
<p>Another factor that reduces the accuracy, and therefore adopting
mixed models via REML in the analysis is the most suitable for obtaining
BLUPs.</p>
</div>
<div id="analysis-of-traits" class="section level3">
<h3>Analysis of traits</h3>
<p>Now, we will analyze the frequency for each discrete feature.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">plot_bar</span>(pheno, <span class="at">ncol =</span> <span class="dv">4</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>Mite Incidence and Flowering have little information for some levels
and many NA’s, we will also exclude these traits.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno  <span class="sc">%&gt;%</span> </span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Incidence_Mites, Flowering))</span></code></pre></div>
<p>Let’s just look at the missing values now, to check the
proportions.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a><span class="fu">plot_missing</span>(pheno)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-12-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We have a high missing value ratio for Vigor, Leaf_Lenght,
Canopy_Width and Canopy_Lenght, I’ll exclude those too.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> </span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Vigor, Leaf_Lenght, Canopy_Width, Canopy_Lenght))</span></code></pre></div>
<p>Let’s check the distribution of traits by year now and let’s look at
the histograms of the quantitative traits:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a><span class="fu">plot_histogram</span>(pheno, <span class="at">ncol =</span> <span class="dv">5</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-14-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>For Branching, Leaf.Ret, Mite, PltArc, Stand6MAP and Staygreen don’t
have normal distribution. To get the BLUPs we will have to remove that
traits from the database.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> </span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Branching, Leaf.Ret, Mite, PltArc, Stand6MAP, Staygreen))</span></code></pre></div>
</div>
<div id="analisys-of-clone" class="section level3">
<h3>Analisys of Clone</h3>
<p>First, let’s check the amount of missing values for each clone by
year. We are filtering the Clones with the average bigger than 2 missing
values by year.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a>pheno2 <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span> </span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Bloco) <span class="sc">%&gt;%</span> </span>
<span id="cb20-3"><a href="#cb20-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Clone, Ano) <span class="sc">%&gt;%</span> </span>
<span id="cb20-4"><a href="#cb20-4" tabindex="-1"></a>  <span class="fu">summarise_all</span>(<span class="at">.funs =</span> <span class="fu">list</span>(<span class="sc">~</span><span class="fu">sum</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span> </span>
<span id="cb20-5"><a href="#cb20-5" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb20-6"><a href="#cb20-6" tabindex="-1"></a>  <span class="fu">select_numeric_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-7"><a href="#cb20-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean =</span> <span class="fu">rowMeans</span>(.),</span>
<span id="cb20-8"><a href="#cb20-8" tabindex="-1"></a>         <span class="at">Clone.Ano =</span> <span class="fu">unique</span>(<span class="fu">interaction</span>(pheno<span class="sc">$</span>Clone, pheno<span class="sc">$</span>Ano))) <span class="sc">%&gt;%</span></span>
<span id="cb20-9"><a href="#cb20-9" tabindex="-1"></a>  <span class="fu">filter</span>(mean <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb20-10"><a href="#cb20-10" tabindex="-1"></a>  <span class="fu">droplevels</span>()</span>
<span id="cb20-11"><a href="#cb20-11" tabindex="-1"></a></span>
<span id="cb20-12"><a href="#cb20-12" tabindex="-1"></a><span class="fu">nlevels</span>(pheno2<span class="sc">$</span>Clone.Ano) <span class="sc">%&gt;%</span> </span>
<span id="cb20-13"><a href="#cb20-13" tabindex="-1"></a>  <span class="fu">kbl</span>(</span>
<span id="cb20-14"><a href="#cb20-14" tabindex="-1"></a>    <span class="at">escape =</span> F,</span>
<span id="cb20-15"><a href="#cb20-15" tabindex="-1"></a>    <span class="at">align =</span> <span class="st">&#39;c&#39;</span>,</span>
<span id="cb20-16"><a href="#cb20-16" tabindex="-1"></a>    <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;N of genotypes&quot;</span>)</span>
<span id="cb20-17"><a href="#cb20-17" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb20-18"><a href="#cb20-18" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb20-19"><a href="#cb20-19" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb20-20"><a href="#cb20-20" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb20-21"><a href="#cb20-21" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb20-22"><a href="#cb20-22" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb20-23"><a href="#cb20-23" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
N of genotypes
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
54
</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="fu">rm</span>(pheno2)</span></code></pre></div>
<p>54 clones presented many missing values, i.e., they were evaluated in
less than two blocks by year. Therefore, they should be excluded from
future analyses, according to the year.</p>
<p>Let’s evaluate the descriptive statistics of the combination between
clone and year for the traits.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">ge_details</span>(pheno, Ano, Clone, <span class="at">resp =</span> <span class="fu">everything</span>()) <span class="sc">%&gt;%</span> </span>
<span id="cb22-2"><a href="#cb22-2" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb22-3"><a href="#cb22-3" tabindex="-1"></a><span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="#cb22-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb22-5"><a href="#cb22-5" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb22-6"><a href="#cb22-6" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb22-7"><a href="#cb22-7" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb22-8"><a href="#cb22-8" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb22-9"><a href="#cb22-9" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Parameters
</td>
<td style="text-align:center;">
Mean
</td>
<td style="text-align:center;">
SE
</td>
<td style="text-align:center;">
SD
</td>
<td style="text-align:center;">
CV
</td>
<td style="text-align:center;">
Min
</td>
<td style="text-align:center;">
Max
</td>
<td style="text-align:center;">
MinENV
</td>
<td style="text-align:center;">
MaxENV
</td>
<td style="text-align:center;">
MinGEN
</td>
<td style="text-align:center;">
MaxGEN
</td>
</tr>
<tr>
<td style="text-align:left;">
N_Roots
</td>
<td style="text-align:center;">
4.29
</td>
<td style="text-align:center;">
0.06
</td>
<td style="text-align:center;">
2.51
</td>
<td style="text-align:center;">
58.56
</td>
<td style="text-align:center;">
0.12 (BGM-1031 in 2017)
</td>
<td style="text-align:center;">
15.67 (2012-107-002 in 2019)
</td>
<td style="text-align:center;">
2018 (1.62)
</td>
<td style="text-align:center;">
2019 (5.76)
</td>
<td style="text-align:center;">
BGM-0411 (0.33)
</td>
<td style="text-align:center;">
2012-107-002 (11.33)
</td>
</tr>
<tr>
<td style="text-align:left;">
FRY
</td>
<td style="text-align:center;">
4.95
</td>
<td style="text-align:center;">
0.09
</td>
<td style="text-align:center;">
4.04
</td>
<td style="text-align:center;">
81.79
</td>
<td style="text-align:center;">
0.12 (BGM-0886 in 2017)
</td>
<td style="text-align:center;">
22.2 (BGM-1267 in 2018)
</td>
<td style="text-align:center;">
2017 (2.75)
</td>
<td style="text-align:center;">
2020 (6.52)
</td>
<td style="text-align:center;">
BGM-1488 (0.34)
</td>
<td style="text-align:center;">
IAC-14 (14.07)
</td>
</tr>
<tr>
<td style="text-align:left;">
ShY
</td>
<td style="text-align:center;">
14.23
</td>
<td style="text-align:center;">
0.22
</td>
<td style="text-align:center;">
10.16
</td>
<td style="text-align:center;">
71.45
</td>
<td style="text-align:center;">
0.69 (BGM-0996 in 2017)
</td>
<td style="text-align:center;">
61.17 (BGM-2124 in 2020)
</td>
<td style="text-align:center;">
2017 (8.47)
</td>
<td style="text-align:center;">
2020 (25.87)
</td>
<td style="text-align:center;">
BGM-0048 (1.52)
</td>
<td style="text-align:center;">
BGM-2124 (54.33)
</td>
</tr>
<tr>
<td style="text-align:left;">
DMC
</td>
<td style="text-align:center;">
29.06
</td>
<td style="text-align:center;">
0.17
</td>
<td style="text-align:center;">
6.1
</td>
<td style="text-align:center;">
21
</td>
<td style="text-align:center;">
11.98 (BGM-0626 in 2020)
</td>
<td style="text-align:center;">
48.34 (BGM-1015 in 2020)
</td>
<td style="text-align:center;">
2020 (26.21)
</td>
<td style="text-align:center;">
2018 (35.38)
</td>
<td style="text-align:center;">
BGM-0626 (14.98)
</td>
<td style="text-align:center;">
BGM-1015 (45.02)
</td>
</tr>
<tr>
<td style="text-align:left;">
StY
</td>
<td style="text-align:center;">
1.52
</td>
<td style="text-align:center;">
0.04
</td>
<td style="text-align:center;">
1.27
</td>
<td style="text-align:center;">
84.01
</td>
<td style="text-align:center;">
0.02 (BGM-0340 in 2019)
</td>
<td style="text-align:center;">
8.87 (BGM-0396 in 2018)
</td>
<td style="text-align:center;">
2019 (1.32)
</td>
<td style="text-align:center;">
2018 (1.84)
</td>
<td style="text-align:center;">
BGM-0089 (0.06)
</td>
<td style="text-align:center;">
BGM-1023 (4.76)
</td>
</tr>
<tr>
<td style="text-align:left;">
Plant.Height
</td>
<td style="text-align:center;">
1.19
</td>
<td style="text-align:center;">
0.01
</td>
<td style="text-align:center;">
0.33
</td>
<td style="text-align:center;">
27.43
</td>
<td style="text-align:center;">
0.36 (BGM-0426 in 2020)
</td>
<td style="text-align:center;">
3.03 (BR-11-24-156 in 2020)
</td>
<td style="text-align:center;">
2017 (1)
</td>
<td style="text-align:center;">
2019 (1.48)
</td>
<td style="text-align:center;">
Jatobá (0.58)
</td>
<td style="text-align:center;">
BGM-1200 (1.91)
</td>
</tr>
<tr>
<td style="text-align:left;">
HI
</td>
<td style="text-align:center;">
24.56
</td>
<td style="text-align:center;">
0.27
</td>
<td style="text-align:center;">
11.89
</td>
<td style="text-align:center;">
48.42
</td>
<td style="text-align:center;">
1.57 (BGM-1159 in 2017)
</td>
<td style="text-align:center;">
71.97 (BGM-1315 in 2018)
</td>
<td style="text-align:center;">
2020 (18.61)
</td>
<td style="text-align:center;">
2018 (32.36)
</td>
<td style="text-align:center;">
BGM-0961 (2.5)
</td>
<td style="text-align:center;">
Mata_Fome_Branca (52.78)
</td>
</tr>
<tr>
<td style="text-align:left;">
StC
</td>
<td style="text-align:center;">
24.42
</td>
<td style="text-align:center;">
0.17
</td>
<td style="text-align:center;">
6.11
</td>
<td style="text-align:center;">
25.05
</td>
<td style="text-align:center;">
7.33 (BGM-0626 in 2020)
</td>
<td style="text-align:center;">
43.69 (BGM-1015 in 2020)
</td>
<td style="text-align:center;">
2020 (21.56)
</td>
<td style="text-align:center;">
2018 (30.78)
</td>
<td style="text-align:center;">
BGM-0626 (10.33)
</td>
<td style="text-align:center;">
BGM-1015 (40.37)
</td>
</tr>
<tr>
<td style="text-align:left;">
Root.Le
</td>
<td style="text-align:center;">
23.21
</td>
<td style="text-align:center;">
0.13
</td>
<td style="text-align:center;">
5.8
</td>
<td style="text-align:center;">
24.99
</td>
<td style="text-align:center;">
7 (BGM-1574 in 2017)
</td>
<td style="text-align:center;">
47.33 (BGM-0396 in 2018)
</td>
<td style="text-align:center;">
2017 (19.72)
</td>
<td style="text-align:center;">
2019 (27.14)
</td>
<td style="text-align:center;">
Jatobá (8.5)
</td>
<td style="text-align:center;">
BGM-1956 (35.5)
</td>
</tr>
<tr>
<td style="text-align:left;">
Root.Di
</td>
<td style="text-align:center;">
28.88
</td>
<td style="text-align:center;">
0.18
</td>
<td style="text-align:center;">
7.9
</td>
<td style="text-align:center;">
27.35
</td>
<td style="text-align:center;">
6.12 (BGM-2142 in 2019)
</td>
<td style="text-align:center;">
63.3 (BRS Mulatinha in 2018)
</td>
<td style="text-align:center;">
2017 (24.49)
</td>
<td style="text-align:center;">
2018 (34.74)
</td>
<td style="text-align:center;">
BGM-0089 (12.14)
</td>
<td style="text-align:center;">
BGM-1956 (48.91)
</td>
</tr>
<tr>
<td style="text-align:left;">
Stem.D
</td>
<td style="text-align:center;">
2.11
</td>
<td style="text-align:center;">
0.01
</td>
<td style="text-align:center;">
0.38
</td>
<td style="text-align:center;">
17.9
</td>
<td style="text-align:center;">
1.01 (BGM-0592 in 2018)
</td>
<td style="text-align:center;">
4.37 (BRS Tapioqueira in 2020)
</td>
<td style="text-align:center;">
2018 (2.02)
</td>
<td style="text-align:center;">
2017 (2.16)
</td>
<td style="text-align:center;">
BGM-0048 (1.25)
</td>
<td style="text-align:center;">
BGM-1523 (2.93)
</td>
</tr>
<tr>
<td style="text-align:left;">
Nstem.Plant
</td>
<td style="text-align:center;">
2.13
</td>
<td style="text-align:center;">
0.03
</td>
<td style="text-align:center;">
0.95
</td>
<td style="text-align:center;">
44.53
</td>
<td style="text-align:center;">
1 (BGM-0036 in 2018)
</td>
<td style="text-align:center;">
6.67 (BGM-0714 in 2019)
</td>
<td style="text-align:center;">
2018 (1.44)
</td>
<td style="text-align:center;">
2019 (2.71)
</td>
<td style="text-align:center;">
BGM-0066 (1)
</td>
<td style="text-align:center;">
BGM-0451 (4.44)
</td>
</tr>
</tbody>
</table>
<p>Apparently we no longer have a genotype that could harm our analysis.
Now we must evaluate the clone-only descriptive statistics for the
traits. Some traits presented hight cv, StY and FRY, but we will go
continue.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" tabindex="-1"></a><span class="fu">desc_stat</span>(pheno, <span class="at">by =</span> Ano) <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="#cb23-2" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb23-3"><a href="#cb23-3" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb23-4"><a href="#cb23-4" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb23-5"><a href="#cb23-5" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb23-6"><a href="#cb23-6" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb23-7"><a href="#cb23-7" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb23-8"><a href="#cb23-8" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
variable
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
cv
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
max
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
mean
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
median
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
min
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
sd.amo
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
se
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
ci.t
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
n.valid
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-Inf
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
Inf
</td>
<td style="text-align:center;">
0.0000
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
0
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
66.8379
</td>
<td style="text-align:center;">
9.6700
</td>
<td style="text-align:center;">
2.7517
</td>
<td style="text-align:center;">
2.4310
</td>
<td style="text-align:center;">
0.1160
</td>
<td style="text-align:center;">
1.8392
</td>
<td style="text-align:center;">
0.0743
</td>
<td style="text-align:center;">
0.1459
</td>
<td style="text-align:center;">
613
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
48.2942
</td>
<td style="text-align:center;">
66.6827
</td>
<td style="text-align:center;">
23.5691
</td>
<td style="text-align:center;">
22.7255
</td>
<td style="text-align:center;">
1.5744
</td>
<td style="text-align:center;">
11.3825
</td>
<td style="text-align:center;">
0.4612
</td>
<td style="text-align:center;">
0.9058
</td>
<td style="text-align:center;">
609
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
48.8708
</td>
<td style="text-align:center;">
12.0000
</td>
<td style="text-align:center;">
4.3530
</td>
<td style="text-align:center;">
4.3330
</td>
<td style="text-align:center;">
0.1250
</td>
<td style="text-align:center;">
2.1273
</td>
<td style="text-align:center;">
0.0860
</td>
<td style="text-align:center;">
0.1689
</td>
<td style="text-align:center;">
612
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-Inf
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
Inf
</td>
<td style="text-align:center;">
0.0000
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
0
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
21.0398
</td>
<td style="text-align:center;">
1.6767
</td>
<td style="text-align:center;">
1.0023
</td>
<td style="text-align:center;">
1.0000
</td>
<td style="text-align:center;">
0.4500
</td>
<td style="text-align:center;">
0.2109
</td>
<td style="text-align:center;">
0.0084
</td>
<td style="text-align:center;">
0.0165
</td>
<td style="text-align:center;">
632
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
21.0039
</td>
<td style="text-align:center;">
39.6900
</td>
<td style="text-align:center;">
24.4889
</td>
<td style="text-align:center;">
24.4733
</td>
<td style="text-align:center;">
8.9867
</td>
<td style="text-align:center;">
5.1436
</td>
<td style="text-align:center;">
0.2077
</td>
<td style="text-align:center;">
0.4080
</td>
<td style="text-align:center;">
613
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
21.5823
</td>
<td style="text-align:center;">
31.6667
</td>
<td style="text-align:center;">
19.7236
</td>
<td style="text-align:center;">
19.6667
</td>
<td style="text-align:center;">
7.0000
</td>
<td style="text-align:center;">
4.2568
</td>
<td style="text-align:center;">
0.1719
</td>
<td style="text-align:center;">
0.3376
</td>
<td style="text-align:center;">
613
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
45.4973
</td>
<td style="text-align:center;">
22.2220
</td>
<td style="text-align:center;">
8.4677
</td>
<td style="text-align:center;">
8.0750
</td>
<td style="text-align:center;">
0.6940
</td>
<td style="text-align:center;">
3.8526
</td>
<td style="text-align:center;">
0.1542
</td>
<td style="text-align:center;">
0.3029
</td>
<td style="text-align:center;">
624
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-Inf
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
Inf
</td>
<td style="text-align:center;">
0.0000
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
0
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
-Inf
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
Inf
</td>
<td style="text-align:center;">
0.0000
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NaN
</td>
<td style="text-align:center;">
0
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
16.7774
</td>
<td style="text-align:center;">
3.3333
</td>
<td style="text-align:center;">
2.1581
</td>
<td style="text-align:center;">
2.1667
</td>
<td style="text-align:center;">
1.2000
</td>
<td style="text-align:center;">
0.3621
</td>
<td style="text-align:center;">
0.0144
</td>
<td style="text-align:center;">
0.0284
</td>
<td style="text-align:center;">
629
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
13.6086
</td>
<td style="text-align:center;">
46.0784
</td>
<td style="text-align:center;">
35.3819
</td>
<td style="text-align:center;">
36.0000
</td>
<td style="text-align:center;">
18.1818
</td>
<td style="text-align:center;">
4.8150
</td>
<td style="text-align:center;">
0.2785
</td>
<td style="text-align:center;">
0.5480
</td>
<td style="text-align:center;">
299
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
86.0073
</td>
<td style="text-align:center;">
22.2000
</td>
<td style="text-align:center;">
5.6640
</td>
<td style="text-align:center;">
4.2500
</td>
<td style="text-align:center;">
0.3330
</td>
<td style="text-align:center;">
4.8714
</td>
<td style="text-align:center;">
0.2776
</td>
<td style="text-align:center;">
0.5462
</td>
<td style="text-align:center;">
308
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
41.5442
</td>
<td style="text-align:center;">
71.9673
</td>
<td style="text-align:center;">
32.3554
</td>
<td style="text-align:center;">
32.6094
</td>
<td style="text-align:center;">
5.8772
</td>
<td style="text-align:center;">
13.4418
</td>
<td style="text-align:center;">
0.7672
</td>
<td style="text-align:center;">
1.5096
</td>
<td style="text-align:center;">
307
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
63.2913
</td>
<td style="text-align:center;">
6.2500
</td>
<td style="text-align:center;">
1.6249
</td>
<td style="text-align:center;">
1.4085
</td>
<td style="text-align:center;">
0.1900
</td>
<td style="text-align:center;">
1.0284
</td>
<td style="text-align:center;">
0.0588
</td>
<td style="text-align:center;">
0.1157
</td>
<td style="text-align:center;">
306
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
33.3849
</td>
<td style="text-align:center;">
3.3333
</td>
<td style="text-align:center;">
1.4397
</td>
<td style="text-align:center;">
1.3333
</td>
<td style="text-align:center;">
1.0000
</td>
<td style="text-align:center;">
0.4806
</td>
<td style="text-align:center;">
0.0248
</td>
<td style="text-align:center;">
0.0487
</td>
<td style="text-align:center;">
376
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
28.0409
</td>
<td style="text-align:center;">
1.9967
</td>
<td style="text-align:center;">
1.0953
</td>
<td style="text-align:center;">
1.0600
</td>
<td style="text-align:center;">
0.4867
</td>
<td style="text-align:center;">
0.3071
</td>
<td style="text-align:center;">
0.0159
</td>
<td style="text-align:center;">
0.0313
</td>
<td style="text-align:center;">
373
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
23.3943
</td>
<td style="text-align:center;">
63.3033
</td>
<td style="text-align:center;">
34.7356
</td>
<td style="text-align:center;">
34.4400
</td>
<td style="text-align:center;">
14.1300
</td>
<td style="text-align:center;">
8.1262
</td>
<td style="text-align:center;">
0.4707
</td>
<td style="text-align:center;">
0.9264
</td>
<td style="text-align:center;">
298
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
27.1625
</td>
<td style="text-align:center;">
47.3333
</td>
<td style="text-align:center;">
23.6102
</td>
<td style="text-align:center;">
23.0000
</td>
<td style="text-align:center;">
9.5000
</td>
<td style="text-align:center;">
6.4131
</td>
<td style="text-align:center;">
0.3715
</td>
<td style="text-align:center;">
0.7311
</td>
<td style="text-align:center;">
298
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
69.0625
</td>
<td style="text-align:center;">
31.6000
</td>
<td style="text-align:center;">
8.7610
</td>
<td style="text-align:center;">
7.2000
</td>
<td style="text-align:center;">
1.0000
</td>
<td style="text-align:center;">
6.0506
</td>
<td style="text-align:center;">
0.3022
</td>
<td style="text-align:center;">
0.5940
</td>
<td style="text-align:center;">
401
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
15.6937
</td>
<td style="text-align:center;">
41.4284
</td>
<td style="text-align:center;">
30.7820
</td>
<td style="text-align:center;">
31.3500
</td>
<td style="text-align:center;">
13.5318
</td>
<td style="text-align:center;">
4.8309
</td>
<td style="text-align:center;">
0.2794
</td>
<td style="text-align:center;">
0.5498
</td>
<td style="text-align:center;">
299
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
91.8742
</td>
<td style="text-align:center;">
8.8669
</td>
<td style="text-align:center;">
1.8356
</td>
<td style="text-align:center;">
1.3539
</td>
<td style="text-align:center;">
0.0839
</td>
<td style="text-align:center;">
1.6864
</td>
<td style="text-align:center;">
0.0984
</td>
<td style="text-align:center;">
0.1936
</td>
<td style="text-align:center;">
294
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
22.3941
</td>
<td style="text-align:center;">
3.7317
</td>
<td style="text-align:center;">
2.0158
</td>
<td style="text-align:center;">
1.9785
</td>
<td style="text-align:center;">
1.0133
</td>
<td style="text-align:center;">
0.4514
</td>
<td style="text-align:center;">
0.0234
</td>
<td style="text-align:center;">
0.0460
</td>
<td style="text-align:center;">
372
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
13.8212
</td>
<td style="text-align:center;">
40.5941
</td>
<td style="text-align:center;">
28.2826
</td>
<td style="text-align:center;">
28.2828
</td>
<td style="text-align:center;">
15.3846
</td>
<td style="text-align:center;">
3.9090
</td>
<td style="text-align:center;">
0.1745
</td>
<td style="text-align:center;">
0.3428
</td>
<td style="text-align:center;">
502
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
70.0326
</td>
<td style="text-align:center;">
22.2000
</td>
<td style="text-align:center;">
5.5039
</td>
<td style="text-align:center;">
4.6670
</td>
<td style="text-align:center;">
0.1430
</td>
<td style="text-align:center;">
3.8545
</td>
<td style="text-align:center;">
0.1712
</td>
<td style="text-align:center;">
0.3363
</td>
<td style="text-align:center;">
507
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
41.8572
</td>
<td style="text-align:center;">
60.1626
</td>
<td style="text-align:center;">
27.2545
</td>
<td style="text-align:center;">
27.3973
</td>
<td style="text-align:center;">
1.7258
</td>
<td style="text-align:center;">
11.4080
</td>
<td style="text-align:center;">
0.5066
</td>
<td style="text-align:center;">
0.9954
</td>
<td style="text-align:center;">
507
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
47.2357
</td>
<td style="text-align:center;">
15.6670
</td>
<td style="text-align:center;">
5.7620
</td>
<td style="text-align:center;">
5.6670
</td>
<td style="text-align:center;">
0.3330
</td>
<td style="text-align:center;">
2.7217
</td>
<td style="text-align:center;">
0.1187
</td>
<td style="text-align:center;">
0.2331
</td>
<td style="text-align:center;">
526
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
37.3740
</td>
<td style="text-align:center;">
6.6667
</td>
<td style="text-align:center;">
2.7113
</td>
<td style="text-align:center;">
2.6667
</td>
<td style="text-align:center;">
1.0000
</td>
<td style="text-align:center;">
1.0133
</td>
<td style="text-align:center;">
0.0441
</td>
<td style="text-align:center;">
0.0867
</td>
<td style="text-align:center;">
527
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
19.1125
</td>
<td style="text-align:center;">
2.4233
</td>
<td style="text-align:center;">
1.4843
</td>
<td style="text-align:center;">
1.4633
</td>
<td style="text-align:center;">
0.7400
</td>
<td style="text-align:center;">
0.2837
</td>
<td style="text-align:center;">
0.0123
</td>
<td style="text-align:center;">
0.0243
</td>
<td style="text-align:center;">
528
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
24.3155
</td>
<td style="text-align:center;">
58.9267
</td>
<td style="text-align:center;">
33.4940
</td>
<td style="text-align:center;">
33.6500
</td>
<td style="text-align:center;">
6.1200
</td>
<td style="text-align:center;">
8.1442
</td>
<td style="text-align:center;">
0.3596
</td>
<td style="text-align:center;">
0.7064
</td>
<td style="text-align:center;">
513
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
21.1089
</td>
<td style="text-align:center;">
45.6667
</td>
<td style="text-align:center;">
27.1361
</td>
<td style="text-align:center;">
27.0000
</td>
<td style="text-align:center;">
11.0000
</td>
<td style="text-align:center;">
5.7281
</td>
<td style="text-align:center;">
0.2529
</td>
<td style="text-align:center;">
0.4969
</td>
<td style="text-align:center;">
513
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
47.4901
</td>
<td style="text-align:center;">
47.5710
</td>
<td style="text-align:center;">
13.4607
</td>
<td style="text-align:center;">
12.7735
</td>
<td style="text-align:center;">
1.2860
</td>
<td style="text-align:center;">
6.3925
</td>
<td style="text-align:center;">
0.2782
</td>
<td style="text-align:center;">
0.5465
</td>
<td style="text-align:center;">
528
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
16.5407
</td>
<td style="text-align:center;">
35.9441
</td>
<td style="text-align:center;">
23.6326
</td>
<td style="text-align:center;">
23.6328
</td>
<td style="text-align:center;">
10.7346
</td>
<td style="text-align:center;">
3.9090
</td>
<td style="text-align:center;">
0.1745
</td>
<td style="text-align:center;">
0.3428
</td>
<td style="text-align:center;">
502
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
72.3243
</td>
<td style="text-align:center;">
5.4021
</td>
<td style="text-align:center;">
1.3238
</td>
<td style="text-align:center;">
1.1095
</td>
<td style="text-align:center;">
0.0154
</td>
<td style="text-align:center;">
0.9574
</td>
<td style="text-align:center;">
0.0428
</td>
<td style="text-align:center;">
0.0841
</td>
<td style="text-align:center;">
500
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
15.7037
</td>
<td style="text-align:center;">
3.3203
</td>
<td style="text-align:center;">
2.1061
</td>
<td style="text-align:center;">
2.0963
</td>
<td style="text-align:center;">
1.1197
</td>
<td style="text-align:center;">
0.3307
</td>
<td style="text-align:center;">
0.0144
</td>
<td style="text-align:center;">
0.0283
</td>
<td style="text-align:center;">
528
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
22.5059
</td>
<td style="text-align:center;">
48.3360
</td>
<td style="text-align:center;">
26.2094
</td>
<td style="text-align:center;">
26.0400
</td>
<td style="text-align:center;">
11.9760
</td>
<td style="text-align:center;">
5.8986
</td>
<td style="text-align:center;">
0.2569
</td>
<td style="text-align:center;">
0.5048
</td>
<td style="text-align:center;">
527
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
68.4455
</td>
<td style="text-align:center;">
20.5670
</td>
<td style="text-align:center;">
6.5246
</td>
<td style="text-align:center;">
5.9000
</td>
<td style="text-align:center;">
0.4000
</td>
<td style="text-align:center;">
4.4658
</td>
<td style="text-align:center;">
0.1934
</td>
<td style="text-align:center;">
0.3800
</td>
<td style="text-align:center;">
533
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
43.3204
</td>
<td style="text-align:center;">
45.9340
</td>
<td style="text-align:center;">
18.6129
</td>
<td style="text-align:center;">
18.0486
</td>
<td style="text-align:center;">
1.9608
</td>
<td style="text-align:center;">
8.0632
</td>
<td style="text-align:center;">
0.3496
</td>
<td style="text-align:center;">
0.6867
</td>
<td style="text-align:center;">
532
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
47.0839
</td>
<td style="text-align:center;">
10.3330
</td>
<td style="text-align:center;">
4.3060
</td>
<td style="text-align:center;">
4.3330
</td>
<td style="text-align:center;">
0.3330
</td>
<td style="text-align:center;">
2.0274
</td>
<td style="text-align:center;">
0.0881
</td>
<td style="text-align:center;">
0.1730
</td>
<td style="text-align:center;">
530
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
37.1507
</td>
<td style="text-align:center;">
4.6667
</td>
<td style="text-align:center;">
2.0463
</td>
<td style="text-align:center;">
2.0000
</td>
<td style="text-align:center;">
1.0000
</td>
<td style="text-align:center;">
0.7602
</td>
<td style="text-align:center;">
0.0326
</td>
<td style="text-align:center;">
0.0641
</td>
<td style="text-align:center;">
543
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
24.0927
</td>
<td style="text-align:center;">
3.0333
</td>
<td style="text-align:center;">
1.1945
</td>
<td style="text-align:center;">
1.1600
</td>
<td style="text-align:center;">
0.3600
</td>
<td style="text-align:center;">
0.2878
</td>
<td style="text-align:center;">
0.0124
</td>
<td style="text-align:center;">
0.0243
</td>
<td style="text-align:center;">
543
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
20.5858
</td>
<td style="text-align:center;">
43.2733
</td>
<td style="text-align:center;">
26.2101
</td>
<td style="text-align:center;">
26.1800
</td>
<td style="text-align:center;">
11.8533
</td>
<td style="text-align:center;">
5.3956
</td>
<td style="text-align:center;">
0.2337
</td>
<td style="text-align:center;">
0.4591
</td>
<td style="text-align:center;">
533
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
18.9730
</td>
<td style="text-align:center;">
36.6667
</td>
<td style="text-align:center;">
23.2345
</td>
<td style="text-align:center;">
23.3333
</td>
<td style="text-align:center;">
9.0000
</td>
<td style="text-align:center;">
4.4083
</td>
<td style="text-align:center;">
0.1909
</td>
<td style="text-align:center;">
0.3751
</td>
<td style="text-align:center;">
533
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
42.7475
</td>
<td style="text-align:center;">
61.1670
</td>
<td style="text-align:center;">
25.8653
</td>
<td style="text-align:center;">
25.9165
</td>
<td style="text-align:center;">
3.3330
</td>
<td style="text-align:center;">
11.0568
</td>
<td style="text-align:center;">
0.4794
</td>
<td style="text-align:center;">
0.9417
</td>
<td style="text-align:center;">
532
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
27.3600
</td>
<td style="text-align:center;">
43.6860
</td>
<td style="text-align:center;">
21.5594
</td>
<td style="text-align:center;">
21.3900
</td>
<td style="text-align:center;">
7.3260
</td>
<td style="text-align:center;">
5.8986
</td>
<td style="text-align:center;">
0.2569
</td>
<td style="text-align:center;">
0.5048
</td>
<td style="text-align:center;">
527
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
81.2220
</td>
<td style="text-align:center;">
6.0724
</td>
<td style="text-align:center;">
1.5211
</td>
<td style="text-align:center;">
1.2326
</td>
<td style="text-align:center;">
0.0483
</td>
<td style="text-align:center;">
1.2354
</td>
<td style="text-align:center;">
0.0540
</td>
<td style="text-align:center;">
0.1060
</td>
<td style="text-align:center;">
524
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
17.5042
</td>
<td style="text-align:center;">
4.3733
</td>
<td style="text-align:center;">
2.1321
</td>
<td style="text-align:center;">
2.1207
</td>
<td style="text-align:center;">
1.0547
</td>
<td style="text-align:center;">
0.3732
</td>
<td style="text-align:center;">
0.0160
</td>
<td style="text-align:center;">
0.0315
</td>
<td style="text-align:center;">
543
</td>
</tr>
</tbody>
</table>
<p>Again, some traits were not computed for the year 2017, so we have to
eliminate that year when performing the analysis for these traits.</p>
<p>What draws attention in this table are the high cv for some traits,
especially: HI, Nstem.Plant, N_Roots, ShY, StY and FRY.</p>
<p>This may be due to the presence of outliers, let’s inspect the entire
dataset to assess whether there are outliers:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" tabindex="-1"></a><span class="fu">inspect</span>(pheno <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="#cb24-2" tabindex="-1"></a>          <span class="fu">select_if</span>( <span class="sc">~</span> <span class="sc">!</span><span class="fu">is.factor</span>(.)), <span class="at">verbose =</span> <span class="cn">FALSE</span>, <span class="at">plot =</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="#cb24-3" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb24-4"><a href="#cb24-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb24-5"><a href="#cb24-5" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb24-6"><a href="#cb24-6" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb24-7"><a href="#cb24-7" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb24-8"><a href="#cb24-8" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb24-9"><a href="#cb24-9" tabindex="-1"></a>  )</span></code></pre></div>
<img src="figure/phenotype.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" />
<table class=" lightable-classic lightable-hover" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Variable
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Class
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Missing
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Levels
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Valid_n
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Min
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Median
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Max
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Outlier
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Text
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1974
</td>
<td style="text-align:center;">
0.12
</td>
<td style="text-align:center;">
4.00
</td>
<td style="text-align:center;">
15.67
</td>
<td style="text-align:center;">
16
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1961
</td>
<td style="text-align:center;">
0.12
</td>
<td style="text-align:center;">
3.75
</td>
<td style="text-align:center;">
22.20
</td>
<td style="text-align:center;">
71
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
2085
</td>
<td style="text-align:center;">
0.69
</td>
<td style="text-align:center;">
11.17
</td>
<td style="text-align:center;">
61.17
</td>
<td style="text-align:center;">
89
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1328
</td>
<td style="text-align:center;">
11.98
</td>
<td style="text-align:center;">
28.80
</td>
<td style="text-align:center;">
48.34
</td>
<td style="text-align:center;">
6
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1318
</td>
<td style="text-align:center;">
0.02
</td>
<td style="text-align:center;">
1.21
</td>
<td style="text-align:center;">
8.87
</td>
<td style="text-align:center;">
49
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
2076
</td>
<td style="text-align:center;">
0.36
</td>
<td style="text-align:center;">
1.16
</td>
<td style="text-align:center;">
3.03
</td>
<td style="text-align:center;">
23
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1955
</td>
<td style="text-align:center;">
1.57
</td>
<td style="text-align:center;">
23.35
</td>
<td style="text-align:center;">
71.97
</td>
<td style="text-align:center;">
15
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1328
</td>
<td style="text-align:center;">
7.33
</td>
<td style="text-align:center;">
24.15
</td>
<td style="text-align:center;">
43.69
</td>
<td style="text-align:center;">
6
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1957
</td>
<td style="text-align:center;">
7.00
</td>
<td style="text-align:center;">
23.00
</td>
<td style="text-align:center;">
47.33
</td>
<td style="text-align:center;">
19
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1957
</td>
<td style="text-align:center;">
6.12
</td>
<td style="text-align:center;">
28.17
</td>
<td style="text-align:center;">
63.30
</td>
<td style="text-align:center;">
32
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
2072
</td>
<td style="text-align:center;">
1.01
</td>
<td style="text-align:center;">
2.10
</td>
<td style="text-align:center;">
4.37
</td>
<td style="text-align:center;">
22
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1446
</td>
<td style="text-align:center;">
1.00
</td>
<td style="text-align:center;">
2.00
</td>
<td style="text-align:center;">
6.67
</td>
<td style="text-align:center;">
30
</td>
<td style="text-align:center;">
NA
</td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p>Confirming what was described before, most traits with high cv have
many outliers and therefore we will exclude them in the loop to obtain
the blups.</p>
<div id="general-inspection" class="section level4">
<h4>General Inspection</h4>
<p>Now let’s just perform a general inspection of the data to finish the
manipulations.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" tabindex="-1"></a><span class="fu">corr_plot</span>(pheno, <span class="at">col.by =</span> Ano)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-20-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>StC with DMC and StY with FRY show high correlation.</p>
<p>Furthermore most of the traits apparently show normal distribution of
phenotypic data. So let’s save the clean data and move on to getting the
blups.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" tabindex="-1"></a><span class="fu">write.csv</span>(pheno, <span class="st">&quot;data/pheno_clean.csv&quot;</span>, <span class="at">row.names =</span> F, <span class="at">quote =</span> F)</span>
<span id="cb26-2"><a href="#cb26-2" tabindex="-1"></a></span>
<span id="cb26-3"><a href="#cb26-3" tabindex="-1"></a>pheno<span class="ot">&lt;-</span><span class="fu">read.csv</span>(<span class="st">&quot;data/pheno_clean.csv&quot;</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb26-4"><a href="#cb26-4" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Clone =</span> <span class="fu">as.factor</span>(Clone),</span>
<span id="cb26-5"><a href="#cb26-5" tabindex="-1"></a>         <span class="at">Ano =</span> <span class="fu">as.factor</span>(Ano),</span>
<span id="cb26-6"><a href="#cb26-6" tabindex="-1"></a>         <span class="at">Bloco =</span> <span class="fu">as.factor</span>(Bloco))</span></code></pre></div>
</div>
</div>
</div>
<div id="genotype-environment-analysis-by-mixed-effect-models"
class="section level2">
<h2>Genotype-environment analysis by mixed-effect models</h2>
<p>First, I’m going to create a function to get the blups and some
parameters from our model.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" tabindex="-1"></a>BLUPS_par_mmer <span class="ot">&lt;-</span> <span class="cf">function</span>(model, trait) {</span>
<span id="cb27-2"><a href="#cb27-2" tabindex="-1"></a>  <span class="co"># log likelihood of the model, AIC, convergence T/F</span></span>
<span id="cb27-3"><a href="#cb27-3" tabindex="-1"></a>  modelfit <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>logo</span>
<span id="cb27-4"><a href="#cb27-4" tabindex="-1"></a>  <span class="co"># number of groups for factors, could be used to compute DF</span></span>
<span id="cb27-5"><a href="#cb27-5" tabindex="-1"></a>  groups <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>groups</span>
<span id="cb27-6"><a href="#cb27-6" tabindex="-1"></a>  <span class="co"># variance components</span></span>
<span id="cb27-7"><a href="#cb27-7" tabindex="-1"></a>  varcomp <span class="ot">&lt;-</span> <span class="fu">summary</span>(model)<span class="sc">$</span>varcomp</span>
<span id="cb27-8"><a href="#cb27-8" tabindex="-1"></a>  </span>
<span id="cb27-9"><a href="#cb27-9" tabindex="-1"></a>  <span class="co"># variance components</span></span>
<span id="cb27-10"><a href="#cb27-10" tabindex="-1"></a>  Vg <span class="ot">&lt;-</span> model<span class="sc">$</span>sigma<span class="sc">$</span>Clone <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb27-11"><a href="#cb27-11" tabindex="-1"></a>  Ve <span class="ot">&lt;-</span> model<span class="sc">$</span>sigma<span class="sc">$</span>units <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb27-12"><a href="#cb27-12" tabindex="-1"></a>  </span>
<span id="cb27-13"><a href="#cb27-13" tabindex="-1"></a>  <span class="co"># Narrow-sense heritability</span></span>
<span id="cb27-14"><a href="#cb27-14" tabindex="-1"></a>  H2_narrow <span class="ot">&lt;-</span> Vg <span class="sc">/</span> (Vg <span class="sc">+</span> Ve)</span>
<span id="cb27-15"><a href="#cb27-15" tabindex="-1"></a>  </span>
<span id="cb27-16"><a href="#cb27-16" tabindex="-1"></a>  <span class="co"># Broad-sense heritability</span></span>
<span id="cb27-17"><a href="#cb27-17" tabindex="-1"></a>  Va <span class="ot">&lt;-</span> model<span class="sc">$</span>sigma<span class="sc">$</span><span class="st">`</span><span class="at">Ano:Clone</span><span class="st">`</span> <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb27-18"><a href="#cb27-18" tabindex="-1"></a>  Ve <span class="ot">&lt;-</span> model<span class="sc">$</span>sigma<span class="sc">$</span>units <span class="sc">%&gt;%</span> <span class="fu">as.vector</span>()</span>
<span id="cb27-19"><a href="#cb27-19" tabindex="-1"></a>  n.ano <span class="ot">&lt;-</span> <span class="fu">length</span>(model<span class="sc">$</span>U<span class="sc">$</span>Ano[[trait]])</span>
<span id="cb27-20"><a href="#cb27-20" tabindex="-1"></a>  H2_Broad<span class="ot">&lt;-</span> (Vg) <span class="sc">/</span> (Vg <span class="sc">+</span> (Va<span class="sc">/</span>n.ano) <span class="sc">+</span> (Ve<span class="sc">/</span>(<span class="dv">2</span><span class="sc">*</span>n.ano)))  </span>
<span id="cb27-21"><a href="#cb27-21" tabindex="-1"></a>  </span>
<span id="cb27-22"><a href="#cb27-22" tabindex="-1"></a>  <span class="co"># Extract the BLUPs and PEVs, compute Reliability (REL),</span></span>
<span id="cb27-23"><a href="#cb27-23" tabindex="-1"></a>  <span class="co"># de-regressed BLUPs and weights for downstream analysis</span></span>
<span id="cb27-24"><a href="#cb27-24" tabindex="-1"></a>  blups <span class="ot">&lt;-</span> model<span class="sc">$</span>U<span class="sc">$</span>Clone <span class="sc">%&gt;%</span></span>
<span id="cb27-25"><a href="#cb27-25" tabindex="-1"></a>    <span class="fu">unlist</span>() <span class="sc">%&gt;%</span></span>
<span id="cb27-26"><a href="#cb27-26" tabindex="-1"></a>    <span class="fu">tibble</span>(<span class="at">Clone =</span> <span class="fu">names</span>(.), <span class="at">BLUP =</span> .) <span class="sc">%&gt;%</span></span>
<span id="cb27-27"><a href="#cb27-27" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb27-28"><a href="#cb27-28" tabindex="-1"></a>      <span class="at">Clone =</span> <span class="fu">gsub</span>(<span class="st">&quot;Clone&quot;</span>, <span class="st">&quot;&quot;</span>, Clone),</span>
<span id="cb27-29"><a href="#cb27-29" tabindex="-1"></a>      <span class="at">PEV =</span> <span class="fu">diag</span>(<span class="fu">as.matrix</span>(model<span class="sc">$</span>PevU<span class="sc">$</span>Clone[[<span class="dv">1</span>]])),</span>
<span id="cb27-30"><a href="#cb27-30" tabindex="-1"></a>      <span class="co"># prediction error variance</span></span>
<span id="cb27-31"><a href="#cb27-31" tabindex="-1"></a>      <span class="at">REL =</span> <span class="dv">1</span> <span class="sc">-</span> PEV <span class="sc">/</span> Vg,</span>
<span id="cb27-32"><a href="#cb27-32" tabindex="-1"></a>      <span class="co"># Reliability</span></span>
<span id="cb27-33"><a href="#cb27-33" tabindex="-1"></a>      <span class="at">drgBLUP =</span> BLUP <span class="sc">/</span> REL,</span>
<span id="cb27-34"><a href="#cb27-34" tabindex="-1"></a>      <span class="co"># De-regressed BLUP</span></span>
<span id="cb27-35"><a href="#cb27-35" tabindex="-1"></a>      <span class="at">WT =</span> (<span class="dv">1</span> <span class="sc">-</span> H2_narrow) <span class="sc">/</span> ((<span class="fl">0.1</span> <span class="sc">+</span> (<span class="dv">1</span> <span class="sc">-</span> REL) <span class="sc">/</span> REL) <span class="sc">*</span> H2_narrow) <span class="co"># weight for downstream</span></span>
<span id="cb27-36"><a href="#cb27-36" tabindex="-1"></a>    )</span>
<span id="cb27-37"><a href="#cb27-37" tabindex="-1"></a>  </span>
<span id="cb27-38"><a href="#cb27-38" tabindex="-1"></a>  <span class="co"># Combine all outputs into one object the function can return()</span></span>
<span id="cb27-39"><a href="#cb27-39" tabindex="-1"></a>  out <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb27-40"><a href="#cb27-40" tabindex="-1"></a>    <span class="at">Trait =</span> trait,</span>
<span id="cb27-41"><a href="#cb27-41" tabindex="-1"></a>    <span class="at">H2_Broad =</span> H2_Broad,</span>
<span id="cb27-42"><a href="#cb27-42" tabindex="-1"></a>    <span class="at">H2_narrow=</span>H2_narrow,</span>
<span id="cb27-43"><a href="#cb27-43" tabindex="-1"></a>    <span class="at">modelfit =</span> <span class="fu">list</span>(modelfit),</span>
<span id="cb27-44"><a href="#cb27-44" tabindex="-1"></a>    <span class="at">groups =</span> <span class="fu">list</span>(groups),</span>
<span id="cb27-45"><a href="#cb27-45" tabindex="-1"></a>    <span class="at">blups =</span> <span class="fu">list</span>(blups),</span>
<span id="cb27-46"><a href="#cb27-46" tabindex="-1"></a>    <span class="at">varcomp =</span> <span class="fu">list</span>(varcomp)</span>
<span id="cb27-47"><a href="#cb27-47" tabindex="-1"></a>  )</span>
<span id="cb27-48"><a href="#cb27-48" tabindex="-1"></a>  </span>
<span id="cb27-49"><a href="#cb27-49" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb27-50"><a href="#cb27-50" tabindex="-1"></a>}</span>
<span id="cb27-51"><a href="#cb27-51" tabindex="-1"></a></span>
<span id="cb27-52"><a href="#cb27-52" tabindex="-1"></a><span class="fu">save</span>(BLUPS_par_mmer, <span class="at">file =</span> <span class="st">&quot;output/BLUPS_par_mmer.Rdata&quot;</span>)</span></code></pre></div>
<div id="the-blup-model" class="section level3">
<h3>The BLUP model</h3>
<p>Here we have to remember that we have outliers for some traits and
also that we must exclude the year 2017 for some.</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" tabindex="-1"></a>traits <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pheno)[<span class="dv">4</span><span class="sc">:</span><span class="fu">ncol</span>(pheno)]</span>
<span id="cb28-2"><a href="#cb28-2" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb28-3"><a href="#cb28-3" tabindex="-1"></a>  <span class="fu">mutate_at</span>(traits, as.numeric)</span></code></pre></div>
<p>Now let’s perform the mixed model analysis to get the BLUPs.</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" tabindex="-1"></a><span class="fu">load</span>(<span class="st">&quot;output/BLUPS_par_mmer.Rdata&quot;</span>)</span>
<span id="cb29-2"><a href="#cb29-2" tabindex="-1"></a></span>
<span id="cb29-3"><a href="#cb29-3" tabindex="-1"></a>resultMM <span class="ot">&lt;-</span> <span class="fu">future_map</span>(traits, <span class="cf">function</span>(i) {</span>
<span id="cb29-4"><a href="#cb29-4" tabindex="-1"></a>      data <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb29-5"><a href="#cb29-5" tabindex="-1"></a>        <span class="fu">select</span>(<span class="dv">1</span><span class="sc">:</span><span class="dv">3</span>,<span class="fu">all_of</span>(i)) <span class="sc">%&gt;%</span> </span>
<span id="cb29-6"><a href="#cb29-6" tabindex="-1"></a>        <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb29-7"><a href="#cb29-7" tabindex="-1"></a>        <span class="fu">droplevels</span>()</span>
<span id="cb29-8"><a href="#cb29-8" tabindex="-1"></a>    </span>
<span id="cb29-9"><a href="#cb29-9" tabindex="-1"></a>    model <span class="ot">&lt;-</span> sommer<span class="sc">::</span><span class="fu">mmer</span>(<span class="at">fixed =</span> <span class="fu">as.formula</span>(<span class="fu">paste</span>(i, <span class="st">&quot;~&quot;</span>, <span class="st">&quot;Ano:Bloco&quot;</span>)),</span>
<span id="cb29-10"><a href="#cb29-10" tabindex="-1"></a>      <span class="at">random =</span> <span class="sc">~</span> Clone <span class="sc">+</span> Ano <span class="sc">+</span> Ano<span class="sc">:</span>Clone,</span>
<span id="cb29-11"><a href="#cb29-11" tabindex="-1"></a>      <span class="at">rcov=</span> <span class="sc">~</span> units,</span>
<span id="cb29-12"><a href="#cb29-12" tabindex="-1"></a>      <span class="at">data =</span> data,</span>
<span id="cb29-13"><a href="#cb29-13" tabindex="-1"></a>      <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="co"># Clone:Ano and Ano.Bloco are random and Clone and Ano is fixed</span></span>
<span id="cb29-14"><a href="#cb29-14" tabindex="-1"></a>    </span>
<span id="cb29-15"><a href="#cb29-15" tabindex="-1"></a>    result <span class="ot">&lt;-</span> <span class="fu">BLUPS_par_mmer</span>(model, i)</span>
<span id="cb29-16"><a href="#cb29-16" tabindex="-1"></a>  }</span>
<span id="cb29-17"><a href="#cb29-17" tabindex="-1"></a>)</span>
<span id="cb29-18"><a href="#cb29-18" tabindex="-1"></a></span>
<span id="cb29-19"><a href="#cb29-19" tabindex="-1"></a><span class="fu">names</span>(resultMM) <span class="ot">&lt;-</span> traits</span>
<span id="cb29-20"><a href="#cb29-20" tabindex="-1"></a><span class="fu">save</span>(resultMM, <span class="at">file =</span> <span class="st">&quot;output/result_sommer.RData&quot;</span>)</span></code></pre></div>
</div>
<div id="blups-for-clone" class="section level3">
<h3>BLUPS for <code>Clone</code></h3>
<p>First, I will add the average of the traits with the BLUPs for better
interpretation.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" tabindex="-1"></a>mean_pheno <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(pheno <span class="sc">%&gt;%</span></span>
<span id="cb30-2"><a href="#cb30-2" tabindex="-1"></a>                               <span class="fu">summarise_if</span>(is.numeric, mean, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb30-3"><a href="#cb30-3" tabindex="-1"></a></span>
<span id="cb30-4"><a href="#cb30-4" tabindex="-1"></a><span class="fu">write.csv</span>(mean_pheno, <span class="st">&quot;output/mean_pheno.csv&quot;</span>, <span class="at">row.names =</span> F)</span></code></pre></div>
<p>As I used “future_map” to run each stage 1 analysis in parallel, each
trait is in a separate element of a list. We need to process the
resultMM object into a data.frame or matrix for further analysis.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" tabindex="-1"></a>BLUPS <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">lapply</span>(traits, <span class="cf">function</span>(i) {</span>
<span id="cb31-2"><a href="#cb31-2" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb31-3"><a href="#cb31-3" tabindex="-1"></a>    <span class="at">trait =</span> i,</span>
<span id="cb31-4"><a href="#cb31-4" tabindex="-1"></a>    <span class="at">Clone =</span> <span class="fu">str_split_i</span>(resultMM[[i]][[<span class="st">&quot;blups&quot;</span>]][[<span class="dv">1</span>]][[<span class="st">&quot;Clone&quot;</span>]], <span class="at">pattern =</span> <span class="st">&quot;[.]&quot;</span>, <span class="sc">-</span><span class="dv">1</span>),</span>
<span id="cb31-5"><a href="#cb31-5" tabindex="-1"></a>    resultMM[[i]][[<span class="st">&quot;blups&quot;</span>]][[<span class="dv">1</span>]][<span class="sc">-</span><span class="dv">1</span>],</span>
<span id="cb31-6"><a href="#cb31-6" tabindex="-1"></a>    <span class="at">BLUPS_mean =</span> resultMM[[i]][[<span class="st">&quot;blups&quot;</span>]][[<span class="dv">1</span>]][[<span class="st">&quot;BLUP&quot;</span>]] <span class="sc">+</span> mean_pheno[, i],</span>
<span id="cb31-7"><a href="#cb31-7" tabindex="-1"></a>    <span class="at">drgBLUP_mean =</span> resultMM[[i]][[<span class="st">&quot;blups&quot;</span>]][[<span class="dv">1</span>]][[<span class="st">&quot;drgBLUP&quot;</span>]] <span class="sc">+</span> mean_pheno[, i]</span>
<span id="cb31-8"><a href="#cb31-8" tabindex="-1"></a>  )</span>
<span id="cb31-9"><a href="#cb31-9" tabindex="-1"></a>}))</span>
<span id="cb31-10"><a href="#cb31-10" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" tabindex="-1"></a>H2 <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">lapply</span>(traits, <span class="cf">function</span>(i) {</span>
<span id="cb31-12"><a href="#cb31-12" tabindex="-1"></a>  <span class="fu">data.frame</span>(<span class="at">trait =</span> i,</span>
<span id="cb31-13"><a href="#cb31-13" tabindex="-1"></a>             <span class="at">H2_Broad =</span>  resultMM[[i]][[<span class="st">&quot;H2_Broad&quot;</span>]],</span>
<span id="cb31-14"><a href="#cb31-14" tabindex="-1"></a>             <span class="at">H2_narrow =</span>  resultMM[[i]][[<span class="st">&quot;H2_narrow&quot;</span>]])</span>
<span id="cb31-15"><a href="#cb31-15" tabindex="-1"></a>}))</span>
<span id="cb31-16"><a href="#cb31-16" tabindex="-1"></a></span>
<span id="cb31-17"><a href="#cb31-17" tabindex="-1"></a>varcomp <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(<span class="fu">lapply</span>(traits, <span class="cf">function</span>(i) {</span>
<span id="cb31-18"><a href="#cb31-18" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb31-19"><a href="#cb31-19" tabindex="-1"></a>    <span class="at">trait =</span> i,</span>
<span id="cb31-20"><a href="#cb31-20" tabindex="-1"></a>    <span class="at">grp =</span> <span class="fu">c</span>(<span class="st">&quot;Clone&quot;</span>, <span class="st">&quot;Ano&quot;</span>, <span class="st">&quot;Ano:Clone&quot;</span>, <span class="st">&quot;units&quot;</span>),</span>
<span id="cb31-21"><a href="#cb31-21" tabindex="-1"></a>    resultMM[[i]][[<span class="st">&quot;varcomp&quot;</span>]][[<span class="dv">1</span>]]</span>
<span id="cb31-22"><a href="#cb31-22" tabindex="-1"></a>  )</span>
<span id="cb31-23"><a href="#cb31-23" tabindex="-1"></a>}))</span></code></pre></div>
<p>Saving the results of BLUPs and parameters</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" tabindex="-1"></a><span class="fu">write.csv</span>(BLUPS,</span>
<span id="cb32-2"><a href="#cb32-2" tabindex="-1"></a>          <span class="st">&quot;output/BLUPS.csv&quot;</span>,</span>
<span id="cb32-3"><a href="#cb32-3" tabindex="-1"></a>          <span class="at">row.names =</span> F,</span>
<span id="cb32-4"><a href="#cb32-4" tabindex="-1"></a>          <span class="at">quote =</span> F)</span>
<span id="cb32-5"><a href="#cb32-5" tabindex="-1"></a></span>
<span id="cb32-6"><a href="#cb32-6" tabindex="-1"></a><span class="fu">write.csv</span>(H2,</span>
<span id="cb32-7"><a href="#cb32-7" tabindex="-1"></a>          <span class="st">&quot;output/H2.csv&quot;</span>,</span>
<span id="cb32-8"><a href="#cb32-8" tabindex="-1"></a>          <span class="at">row.names =</span> F,</span>
<span id="cb32-9"><a href="#cb32-9" tabindex="-1"></a>          <span class="at">quote =</span> F)</span>
<span id="cb32-10"><a href="#cb32-10" tabindex="-1"></a></span>
<span id="cb32-11"><a href="#cb32-11" tabindex="-1"></a><span class="fu">write.csv</span>(varcomp,</span>
<span id="cb32-12"><a href="#cb32-12" tabindex="-1"></a>          <span class="st">&quot;output/varcomp.csv&quot;</span>,</span>
<span id="cb32-13"><a href="#cb32-13" tabindex="-1"></a>          <span class="at">row.names =</span> F,</span>
<span id="cb32-14"><a href="#cb32-14" tabindex="-1"></a>          <span class="at">quote =</span> F)</span></code></pre></div>
<p>Create a figure about de variance components</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" tabindex="-1"></a>varcomp <span class="sc">|&gt;</span> </span>
<span id="cb33-2"><a href="#cb33-2" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> trait, <span class="at">y =</span>  VarComp, <span class="at">fill =</span> <span class="fu">fct_reorder</span>(grp, VarComp), <span class="at">by =</span> trait)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="#cb33-3" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="st">&quot;fill&quot;</span>)<span class="sc">+</span> </span>
<span id="cb33-4"><a href="#cb33-4" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y=</span><span class="st">&quot;Variance Component (%)&quot;</span>,</span>
<span id="cb33-5"><a href="#cb33-5" tabindex="-1"></a>       <span class="at">x=</span><span class="st">&quot;Traits&quot;</span>,</span>
<span id="cb33-6"><a href="#cb33-6" tabindex="-1"></a>       <span class="at">fill =</span> <span class="st">&quot;Var. Comp.&quot;</span>)<span class="sc">+</span></span>
<span id="cb33-7"><a href="#cb33-7" tabindex="-1"></a>  <span class="fu">scale_fill_gdocs</span>()<span class="sc">+</span></span>
<span id="cb33-8"><a href="#cb33-8" tabindex="-1"></a>  <span class="fu">theme_minimal</span>()<span class="sc">+</span></span>
<span id="cb33-9"><a href="#cb33-9" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">25</span>, <span class="at">face =</span> <span class="st">&quot;bold&quot;</span>),</span>
<span id="cb33-10"><a href="#cb33-10" tabindex="-1"></a>    <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(</span>
<span id="cb33-11"><a href="#cb33-11" tabindex="-1"></a>      <span class="at">size =</span> <span class="dv">20</span>,</span>
<span id="cb33-12"><a href="#cb33-12" tabindex="-1"></a>      <span class="at">angle =</span> <span class="dv">45</span>,</span>
<span id="cb33-13"><a href="#cb33-13" tabindex="-1"></a>      <span class="at">hjust =</span> <span class="dv">1</span>,</span>
<span id="cb33-14"><a href="#cb33-14" tabindex="-1"></a>      <span class="at">vjust =</span> <span class="dv">1</span></span>
<span id="cb33-15"><a href="#cb33-15" tabindex="-1"></a>    )) </span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-28-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-28-1">
Past versions of unnamed-chunk-28-1.png
</button>
</p>
<div id="fig-unnamed-chunk-28-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-28-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;output/varcomp.tiff&quot;</span>, <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">8</span>)</span></code></pre></div>
</div>
<div id="ploting-blups-for-all-traits" class="section level3">
<h3>Ploting BLUPS for all traits</h3>
<p>Let’s plot the density and boxplots figures of the traits.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" tabindex="-1"></a>BLUPS <span class="sc">%&gt;%</span></span>
<span id="cb35-2"><a href="#cb35-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">8</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">names_to =</span> <span class="st">&quot;Method&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Values&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb35-3"><a href="#cb35-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Values, <span class="at">fill =</span> Method)) <span class="sc">+</span></span>
<span id="cb35-4"><a href="#cb35-4" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb35-5"><a href="#cb35-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(trait<span class="sc">~</span>., <span class="at">ncol =</span> <span class="dv">4</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb35-6"><a href="#cb35-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb35-7"><a href="#cb35-7" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb35-8"><a href="#cb35-8" tabindex="-1"></a>  <span class="fu">scale_fill_gdocs</span>() <span class="sc">+</span></span>
<span id="cb35-9"><a href="#cb35-9" tabindex="-1"></a>  <span class="fu">scale_color_gdocs</span>() <span class="sc">+</span></span>
<span id="cb35-10"><a href="#cb35-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-29-1">
Past versions of unnamed-chunk-29-1.png
</button>
</p>
<div id="fig-unnamed-chunk-29-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-29-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb36"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;output/BLUPS_drgBLUPs__boxplot_med.png&quot;</span>, <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span><span class="dv">300</span>)</span></code></pre></div>
<p>Here we will only evaluate the distribution of BLUPs without the
mean.</p>
<div class="sourceCode" id="cb37"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" tabindex="-1"></a>BLUPS <span class="sc">%&gt;%</span></span>
<span id="cb37-2"><a href="#cb37-2" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="dv">8</span><span class="sc">:</span><span class="dv">9</span>, <span class="at">names_to =</span> <span class="st">&quot;Method&quot;</span>, <span class="at">values_to =</span> <span class="st">&quot;Values&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb37-3"><a href="#cb37-3" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Values, <span class="at">fill =</span> Method, <span class="at">color =</span> Method)) <span class="sc">+</span></span>
<span id="cb37-4"><a href="#cb37-4" tabindex="-1"></a>  <span class="fu">geom_density</span>(<span class="at">alpha =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb37-5"><a href="#cb37-5" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(trait<span class="sc">~</span>., <span class="at">ncol =</span> <span class="dv">4</span>, <span class="at">scales =</span> <span class="st">&quot;free&quot;</span>) <span class="sc">+</span></span>
<span id="cb37-6"><a href="#cb37-6" tabindex="-1"></a>  <span class="fu">theme_bw</span>() <span class="sc">+</span> </span>
<span id="cb37-7"><a href="#cb37-7" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> <span class="dv">15</span>)) <span class="sc">+</span></span>
<span id="cb37-8"><a href="#cb37-8" tabindex="-1"></a>  <span class="fu">scale_fill_gdocs</span>() <span class="sc">+</span></span>
<span id="cb37-9"><a href="#cb37-9" tabindex="-1"></a>  <span class="fu">scale_color_gdocs</span>() <span class="sc">+</span></span>
<span id="cb37-10"><a href="#cb37-10" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">&quot;&quot;</span>, <span class="at">y =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-30-1">
Past versions of unnamed-chunk-30-1.png
</button>
</p>
<div id="fig-unnamed-chunk-30-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/6cc4d2326a79681446b1ea653122061a959c2ffe/docs/figure/phenotype.Rmd/unnamed-chunk-30-1.png" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
</tr>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/figure/phenotype.Rmd/unnamed-chunk-30-1.png" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="sourceCode" id="cb38"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="st">&quot;output/BLUPS_drgBLUPs_density_med.png&quot;</span>, <span class="at">width =</span> <span class="dv">16</span>, <span class="at">height =</span> <span class="dv">10</span>, <span class="at">dpi =</span><span class="dv">300</span>)</span></code></pre></div>
<p>Apparently most BLUPs for the traits follow normal distribution and
can be applied to GWS by conventional methods.</p>
<p>Then, now we go to execute the Genomic Wide Selection script: <a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS">GWS.Rmd</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb39"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.2.3 (2023-03-15 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default

locale:
[1] LC_COLLATE=Portuguese_Brazil.utf8  LC_CTYPE=Portuguese_Brazil.utf8   
[3] LC_MONETARY=Portuguese_Brazil.utf8 LC_NUMERIC=C                      
[5] LC_TIME=Portuguese_Brazil.utf8    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] furrr_0.3.1           future_1.33.0         GGally_2.1.2         
 [4] ggthemes_4.2.4        DataExplorer_0.8.2    metan_1.18.0         
 [7] readxl_1.4.3          data.table_1.14.8     ComplexHeatmap_2.14.0
[10] lubridate_1.9.2       forcats_1.0.0         stringr_1.5.0        
[13] dplyr_1.1.2           purrr_1.0.2           readr_2.1.4          
[16] tidyr_1.3.0           tibble_3.2.1          ggplot2_3.4.3        
[19] tidyverse_2.0.0       kableExtra_1.3.4     

loaded via a namespace (and not attached):
  [1] minqa_1.2.6         colorspace_2.1-0    rjson_0.2.21       
  [4] rprojroot_2.0.3     circlize_0.4.15     GlobalOptions_0.1.2
  [7] fs_1.6.3            clue_0.3-65         rstudioapi_0.15.0  
 [10] listenv_0.9.0       farver_2.1.1        ggrepel_0.9.3      
 [13] fansi_1.0.4         mathjaxr_1.6-0      xml2_1.3.5         
 [16] codetools_0.2-19    splines_4.2.3       doParallel_1.0.17  
 [19] cachem_1.0.8        knitr_1.44          polyclip_1.10-4    
 [22] jsonlite_1.8.7      workflowr_1.7.1     nloptr_2.0.3       
 [25] cluster_2.1.4       png_0.1-8           ggforce_0.4.1      
 [28] compiler_4.2.3      httr_1.4.7          Matrix_1.6-1       
 [31] fastmap_1.1.1       cli_3.6.1           later_1.3.1        
 [34] tweenr_2.0.2        htmltools_0.5.6     tools_4.2.3        
 [37] lmerTest_3.1-3      igraph_1.5.1        gtable_0.3.4       
 [40] glue_1.6.2          Rcpp_1.0.11         cellranger_1.1.0   
 [43] jquerylib_0.1.4     vctrs_0.6.3         svglite_2.1.1      
 [46] nlme_3.1-163        iterators_1.0.14    xfun_0.40          
 [49] globals_0.16.2      networkD3_0.4       lme4_1.1-34        
 [52] rvest_1.0.3         timechange_0.2.0    lifecycle_1.0.3    
 [55] MASS_7.3-58.2       scales_1.2.1        ragg_1.2.6         
 [58] hms_1.1.3           promises_1.2.1      parallel_4.2.3     
 [61] RColorBrewer_1.1-3  yaml_2.3.7          gridExtra_2.3      
 [64] sass_0.4.7          reshape_0.8.9       stringi_1.7.12     
 [67] S4Vectors_0.36.2    foreach_1.5.2       BiocGenerics_0.44.0
 [70] boot_1.3-28.1       shape_1.4.6         rlang_1.1.1        
 [73] pkgconfig_2.0.3     systemfonts_1.0.4   matrixStats_1.0.0  
 [76] evaluate_0.22       lattice_0.21-8      labeling_0.4.3     
 [79] patchwork_1.1.3     htmlwidgets_1.6.2   tidyselect_1.2.0   
 [82] parallelly_1.36.0   plyr_1.8.8          magrittr_2.0.3     
 [85] R6_2.5.1            IRanges_2.32.0      generics_0.1.3     
 [88] mgcv_1.9-0          pillar_1.9.0        whisker_0.4.1      
 [91] withr_2.5.1         crayon_1.5.2        utf8_1.2.3         
 [94] tzdb_0.4.0          rmarkdown_2.25      GetoptLong_1.0.5   
 [97] git2r_0.32.0        digest_0.6.33       webshot_0.5.5      
[100] httpuv_1.6.11       numDeriv_2016.8-1.1 textshaping_0.3.7  
[103] stats4_4.2.3        munsell_0.5.0       viridisLite_0.4.2  
[106] bslib_0.5.1        </code></pre>
</div>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Weverton Gomes da Costa, Pós-Doutorando, Embrapa
Mandioca e Fruticultura, <a href="mailto:wevertonufv@gmail.com"
class="email">wevertonufv@gmail.com</a><a href="#fnref1"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
