<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Costa, W. G." />

<meta name="date" content="2025-01-13" />

<title>Phenotype Analysis of Brazilian Drought Trials</title>

<script src="site_libs/header-attrs-2.24/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; } /* Alert */
code span.an { color: #008000; } /* Annotation */
code span.at { } /* Attribute */
code span.bu { } /* BuiltIn */
code span.cf { color: #0000ff; } /* ControlFlow */
code span.ch { color: #008080; } /* Char */
code span.cn { } /* Constant */
code span.co { color: #008000; } /* Comment */
code span.cv { color: #008000; } /* CommentVar */
code span.do { color: #008000; } /* Documentation */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.im { } /* Import */
code span.in { color: #008000; } /* Information */
code span.kw { color: #0000ff; } /* Keyword */
code span.op { } /* Operator */
code span.ot { color: #ff4000; } /* Other */
code span.pp { color: #ff4000; } /* Preprocessor */
code span.sc { color: #008080; } /* SpecialChar */
code span.ss { color: #008080; } /* SpecialString */
code span.st { color: #008080; } /* String */
code span.va { } /* Variable */
code span.vs { color: #008080; } /* VerbatimString */
code span.wa { color: #008000; font-weight: bold; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Genomic Selection for Drought Tolerance</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="phenotype.html">Phenotype</a>
</li>
<li>
  <a href="mixed_models.html">GxE Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Genomic Selection
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="GWS.html">Overview</a>
    </li>
    <li>
      <a href="GWS_RR-BLUP.html">RR-BLUP</a>
    </li>
    <li>
      <a href="GWS_G-BLUP.html">G-BLUP (additive and additive-dominant)</a>
    </li>
    <li>
      <a href="GWS_BayesA.html">Bayes A</a>
    </li>
    <li>
      <a href="GWS_BayesB.html">Bayes B</a>
    </li>
    <li>
      <a href="GWS_RKHS.html">RKHS</a>
    </li>
    <li>
      <a href="GWS_RF.html">Random Forest</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Phenotype Analysis of Brazilian Drought
Trials</h1>
<h4 class="author">Costa, W. G.<a href="#fn1" class="footnote-ref"
id="fnref1"><sup>1</sup></a></h4>
<h4 class="date">2025-01-13</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-01-13
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong>
<code>Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/</code>
<span class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20221020code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20221020)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20221020code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20221020)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree77e21a61ed405f0bea68834f5bc83b3594dc035atargetblank77e21a6a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/77e21a61ed405f0bea68834f5bc83b3594dc035a" target="_blank">77e21a6</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomWevertonGomesCostaGenomicSelectionforDroughtToleranceUsingGenomeWideSNPsinCasavatree77e21a61ed405f0bea68834f5bc83b3594dc035atargetblank77e21a6a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/tree/77e21a61ed405f0bea68834f5bc83b3594dc035a" target="_blank">77e21a6</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    Artigo GS tolerancia a seca_revised.rar
    Ignored:    Seleção Genômica deficit hidrico.pptx
    Ignored:    data/allchrAR08.txt

Untracked files:
    Untracked:  Artigo GS tolerancia a seca_revised/
    Untracked:  ETA_1_ScaleBayesA.dat
    Untracked:  ETA_1_parBayesB.dat
    Untracked:  analysis/figure/
    Untracked:  analysis/mixed_models.Rmd
    Untracked:  data/Article.docx
    Untracked:  data/Article2.docx
    Untracked:  data/Articles/
    Untracked:  data/Costa et al_FPS_2024 manuscript eng.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript eng_v1.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript eng_v2.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript eng_v3.docx
    Untracked:  data/Costa et al_FPS_2024 manuscript.docx
    Untracked:  data/Costa et al_FPS_2024 supplement.docx
    Untracked:  data/Frontiers_Template.docx
    Untracked:  data/Phenotyping2.xlsx
    Untracked:  data/Supplementary_Material.docx
    Untracked:  data/Tabela auxiliar.xlsx
    Untracked:  data/dados__temp_umi.csv
    Untracked:  data/data.rar
    Untracked:  mu.dat
    Untracked:  output/BLUPS2.csv
    Untracked:  output/BLUPS_GEBV_GETG_all clones.csv
    Untracked:  output/BLUPS_density_med.png
    Untracked:  output/BLUPS_density_med_row_col.png
    Untracked:  output/BLUPS_drgBLUPs__boxplot_med.png
    Untracked:  output/BLUPS_drgBLUPs__boxplot_med_row_col.png
    Untracked:  output/BLUPS_drgBLUPs_density_med.png
    Untracked:  output/BLUPS_drgBLUPs_med.png
    Untracked:  output/BLUPS_par_mmer.Rdata
    Untracked:  output/BLUPS_row_col.csv
    Untracked:  output/BLUPS_row_col_random.csv
    Untracked:  output/BLUPS_x_BLUPS_row_col_boxplot_med.png
    Untracked:  output/Density_residual_row_col.tiff
    Untracked:  output/Figuras_article.rar
    Untracked:  output/GEBVS_BayesA.RDS
    Untracked:  output/GEBVS_BayesB.RDS
    Untracked:  output/GEBVS_DOM.RDS
    Untracked:  output/GEBVS_G_BLUP.RDS
    Untracked:  output/GEBVS_G_BLUP_row_col.RDS
    Untracked:  output/GEBVS_G_BLUP_row_col_random.RDS
    Untracked:  output/GEBVS_RF.RDS
    Untracked:  output/GEBVS_RKHS.RDS
    Untracked:  output/GEBVS_RR_BLUP.RDS
    Untracked:  output/GEBV_BLUP.csv
    Untracked:  output/GEBV_DOMxBLUPS.tiff
    Untracked:  output/GEBV_GBLUPxBLUPS.tiff
    Untracked:  output/GEBVxGETGV.tiff
    Untracked:  output/GETGVxBLUPS.tiff
    Untracked:  output/G_matrix.rds
    Untracked:  output/H2.csv
    Untracked:  output/H2_row_col.csv
    Untracked:  output/H2_row_col_random.csv
    Untracked:  output/Heatmap.tiff
    Untracked:  output/MSPE_all_methods.tiff
    Untracked:  output/Residuals_vs_fitted_row_col.tiff
    Untracked:  output/accuracy_all_methods.tiff
    Untracked:  output/accuracy_all_methods_points.tiff
    Untracked:  output/cor_GEBVxBLUPS.tiff
    Untracked:  output/cor_GEBVxBLUPS_all_methods.tiff
    Untracked:  output/correlation_blups.tiff
    Untracked:  output/density_blups.tiff
    Untracked:  output/dierencial_selecao.csv
    Untracked:  output/dierencial_selecao.xlsx
    Untracked:  output/drgBLUP.csv
    Untracked:  output/fitted_residual_data_row_col_random.csv
    Untracked:  output/fitted_values_row_col_random.csv
    Untracked:  output/indice_selection.tiff
    Untracked:  output/indice_selection3.tiff
    Untracked:  output/indice_selection_GEBV.tiff
    Untracked:  output/indice_selection_GEBV_GETGV.tiff
    Untracked:  output/indice_selection_GETGV.tiff
    Untracked:  output/kappa.tiff
    Untracked:  output/kappa2.tiff
    Untracked:  output/mean_pheno.csv
    Untracked:  output/medias_semestre.xlsx
    Untracked:  output/pesos_BLUPS_all clones.csv
    Untracked:  output/pesos_GEBVS_all clones.csv
    Untracked:  output/pesos_GETGVS_all clones.csv
    Untracked:  output/pheno_mean_sd.csv
    Untracked:  output/residuos_row_col_random.csv
    Untracked:  output/result_sommer.RData
    Untracked:  output/result_sommer_row_col.RData
    Untracked:  output/result_sommer_row_col_random.RDS
    Untracked:  output/result_sommer_row_col_random_optimized.RDS
    Untracked:  output/results_MSPE.csv
    Untracked:  output/results_accuracy.csv
    Untracked:  output/results_cv_BayesA.RDS
    Untracked:  output/results_cv_BayesB.RDS
    Untracked:  output/results_cv_DOM.RDS
    Untracked:  output/results_cv_G_BLUP.RDS
    Untracked:  output/results_cv_G_BLUP_row_col.RDS
    Untracked:  output/results_cv_G_BLUP_row_col_random.RDS
    Untracked:  output/results_cv_RF.RDS
    Untracked:  output/results_cv_RKHS.RDS
    Untracked:  output/results_cv_RR_BLUP.RDS
    Untracked:  output/results_h2_GBLUP (1).rds
    Untracked:  output/results_h2_GBLUP.csv
    Untracked:  output/results_h2_GBLUP.rds
    Untracked:  output/results_h2_GBLUP_D.rds
    Untracked:  output/results_kappa.csv
    Untracked:  output/results_kappa2.csv
    Untracked:  output/results_values-selection.csv
    Untracked:  output/results_values_kappa.csv
    Untracked:  output/results_values_rescale_selection.csv
    Untracked:  output/results_values_selection.csv
    Untracked:  output/rstudio-export.zip
    Untracked:  output/semester_means.xlsx
    Untracked:  output/teste_LRT.csv
    Untracked:  output/teste_LRT_row_col.csv
    Untracked:  output/teste_LRT_row_col_random.csv
    Untracked:  output/varcomp.tiff
    Untracked:  output/varcomp2.csv
    Untracked:  output/varcomp_row_col.csv
    Untracked:  output/varcomp_row_col.tiff
    Untracked:  output/varcomp_row_col_random.csv
    Untracked:  output/varcomp_row_col_random.tiff
    Untracked:  varE.dat

Unstaged changes:
    Modified:   analysis/GWS.Rmd
    Modified:   analysis/phenotype.Rmd
    Modified:   data/SNPs.rds
    Modified:   data/pheno_clean.csv
    Modified:   output/BLUPS.csv
    Modified:   output/BLUPS_par.Rdata
    Modified:   output/media_pheno.csv
    Modified:   output/varcomp.csv

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/phenotype.Rmd</code>) and HTML
(<code>docs/phenotype.html</code>) files. If you’ve configured a remote
Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/analysis/phenotype.Rmd" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
<td>
Update Index and phenotype
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/97fd0b90ba19fd059832c5afbd7f21c647e21699/docs/phenotype.html" target="_blank">97fd0b9</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-07
</td>
<td>
update about.html, license.html, phenotype.html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/54c6e2809526414fcf21dca7ea89db78aa100956/analysis/phenotype.Rmd" target="_blank">54c6e28</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-07
</td>
<td>
update site, index.rmd, license.rmd, phenotype.rmd
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/c64a99143990138a32d3620f14bb6ef4e77d722b/docs/phenotype.html" target="_blank">c64a991</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
add phenotype html
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/286b4922972c00dd9062e8c6a1848a867118378a/analysis/phenotype.Rmd" target="_blank">286b492</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2023-10-27
</td>
<td>
Update Scripts and README
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/90dc11276f860bdf4c29fc2cbf7a7d66d049d277/analysis/phenotype.Rmd" target="_blank">90dc112</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/6cc4d2326a79681446b1ea653122061a959c2ffe/analysis/phenotype.Rmd" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/6cc4d2326a79681446b1ea653122061a959c2ffe/docs/phenotype.html" target="_blank">6cc4d23</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-17
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/d930880aef03c824bb532ffc196a8ad2f9105d51/docs/phenotype.html" target="_blank">d930880</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/5988c27353e95ed989eeda03e0ca0091cccc9bf0/analysis/phenotype.Rmd" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/5988c27353e95ed989eeda03e0ca0091cccc9bf0/docs/phenotype.html" target="_blank">5988c27</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/bf7b1d33a925b9e607ab46f19b205aece363379f/analysis/phenotype.Rmd" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/bf7b1d33a925b9e607ab46f19b205aece363379f/docs/phenotype.html" target="_blank">bf7b1d3</a>
</td>
<td>
WevertonGomesCosta
</td>
<td>
2022-11-11
</td>
<td>
Update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="libraries" class="section level2">
<h2>Libraries</h2>
<p>Load the necessary libraries:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="co">#if (!require(&quot;BiocManager&quot;, quietly = TRUE))</span></span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a><span class="co">#    install.packages(&quot;BiocManager&quot;)</span></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co">#BiocManager::install(&quot;ComplexHeatmap&quot;)</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a><span class="fu">require</span>(ComplexHeatmap)</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="fu">library</span>(readxl)</span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="fu">library</span>(metan)</span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="fu">library</span>(DataExplorer)</span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="fu">library</span>(ggthemes)</span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="fu">library</span>(GGally)</span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="fu">theme_set</span>(<span class="fu">theme_bw</span>())</span></code></pre></div>
</div>
<div id="data-import-and-manipulation" class="section level2">
<h2>Data import and manipulation</h2>
<p>Import the phenotypic dataset, excluding traits without information
and redundant traits:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(<span class="st">&quot;data/Phenotyping2.xlsx&quot;</span>, <span class="at">na =</span> <span class="st">&quot;NA&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a>  <span class="fu">select_if</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">all</span>(<span class="fu">is.na</span>(.))) <span class="sc">%&gt;%</span>  <span class="co"># Deleting traits without information</span></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="st">&quot;Local&quot;</span>, <span class="st">&quot;Tratamento&quot;</span>))</span></code></pre></div>
<p>Convert character traits to factors, and numeric grades to integer
factors:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(Clone, Ano, row, col, Bloco), as.factor))</span></code></pre></div>
</div>
<div id="exploratory-data-analysis" class="section level2">
<h2>Exploratory Data Analysis</h2>
<p>Provide an introductory analysis of the dataset:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">introduce</span>(pheno) <span class="sc">%&gt;%</span></span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
rows
</td>
<td style="text-align:center;">
2336
</td>
</tr>
<tr>
<td style="text-align:left;">
columns
</td>
<td style="text-align:center;">
29
</td>
</tr>
<tr>
<td style="text-align:left;">
discrete_columns
</td>
<td style="text-align:center;">
6
</td>
</tr>
<tr>
<td style="text-align:left;">
continuous_columns
</td>
<td style="text-align:center;">
23
</td>
</tr>
<tr>
<td style="text-align:left;">
all_missing_columns
</td>
<td style="text-align:center;">
0
</td>
</tr>
<tr>
<td style="text-align:left;">
total_missing_values
</td>
<td style="text-align:center;">
16875
</td>
</tr>
<tr>
<td style="text-align:left;">
complete_rows
</td>
<td style="text-align:center;">
440
</td>
</tr>
<tr>
<td style="text-align:left;">
total_observations
</td>
<td style="text-align:center;">
67744
</td>
</tr>
<tr>
<td style="text-align:left;">
memory_usage
</td>
<td style="text-align:center;">
534240
</td>
</tr>
</tbody>
</table>
<p>We don’t have any columns that have all of the missing observations,
but we do have a lot of missing values in every dataset. Some
manipulations should be performed to improve the quality of the
data.</p>
<div id="environment-analysis" class="section level3">
<h3>Environment Analysis</h3>
<p>Heatmap visualization of clone presence per year:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a>pheno_year_count <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone)</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>genmat <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Clone, <span class="at">data =</span> pheno_year_count)</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>envmat <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Ano, <span class="at">data =</span> pheno_year_count)</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>genenvmat <span class="ot">&lt;-</span> <span class="fu">t</span>(envmat) <span class="sc">%*%</span> genmat</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>genenvmat_ch <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(genenvmat <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Absent&quot;</span>)</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a><span class="fu">Heatmap</span>(genenvmat_ch, </span>
<span id="cb5-10"><a href="#cb5-10" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;tomato&quot;</span>), </span>
<span id="cb5-11"><a href="#cb5-11" tabindex="-1"></a>        <span class="at">show_column_names =</span> F, </span>
<span id="cb5-12"><a href="#cb5-12" tabindex="-1"></a>        <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb5-13"><a href="#cb5-13" tabindex="-1"></a>        <span class="at">column_title =</span> <span class="st">&quot;Genotypes&quot;</span>, </span>
<span id="cb5-14"><a href="#cb5-14" tabindex="-1"></a>        <span class="at">row_title =</span> <span class="st">&quot;Environments&quot;</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/heatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-heatmap-1">
Past versions of heatmap-1.png
</button>
</p>
<div id="fig-heatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/docs/figure/phenotype.Rmd/heatmap-1.png" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Filter out the year 2016 due to insufficient observations:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>  <span class="fu">filter</span>(Ano <span class="sc">!=</span> <span class="dv">2016</span>) <span class="sc">%&gt;%</span></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>  <span class="fu">droplevels</span>()</span></code></pre></div>
<p>Reinspect the updated heatmap:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>pheno_year_count <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone)</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a></span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>genmat <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Clone, <span class="at">data =</span> pheno_year_count)</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>envmat <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="sc">-</span><span class="dv">1</span> <span class="sc">+</span> Ano, <span class="at">data =</span> pheno_year_count)</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>genenvmat <span class="ot">&lt;-</span> <span class="fu">t</span>(envmat) <span class="sc">%*%</span> genmat</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>genenvmat_ch <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(genenvmat <span class="sc">==</span> <span class="dv">1</span>, <span class="st">&quot;Present&quot;</span>, <span class="st">&quot;Absent&quot;</span>)</span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a></span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a><span class="fu">Heatmap</span>(genenvmat_ch, </span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>        <span class="at">col =</span> <span class="fu">c</span>(<span class="st">&quot;white&quot;</span>, <span class="st">&quot;tomato&quot;</span>), </span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>        <span class="at">show_column_names =</span> F, </span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>        <span class="at">heatmap_legend_param =</span> <span class="fu">list</span>(<span class="at">title =</span> <span class="st">&quot;&quot;</span>),</span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a>        <span class="at">column_title =</span> <span class="st">&quot;Genotypes&quot;</span>, </span>
<span id="cb7-14"><a href="#cb7-14" tabindex="-1"></a>        <span class="at">row_title =</span> <span class="st">&quot;Environments&quot;</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/reheatmap-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-reheatmap-1">
Past versions of reheatmap-1.png
</button>
</p>
<div id="fig-reheatmap-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/docs/figure/phenotype.Rmd/reheatmap-1.png" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Visualize the number of environments and genotypes:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>genotype_year_table <span class="ot">&lt;-</span> genenvmat <span class="sc">%*%</span> <span class="fu">t</span>(genenvmat) <span class="sc">%&gt;%</span></span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a><span class="fu">rm</span>(pheno_year_count, genmat, envmat, genenvmat, genenvmat_ch)</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>genotype_year_table</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2017
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2018
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2019
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano2020
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Ano2017
</td>
<td style="text-align:center;">
165
</td>
<td style="text-align:center;">
42
</td>
<td style="text-align:center;">
22
</td>
<td style="text-align:center;">
14
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano2018
</td>
<td style="text-align:center;">
42
</td>
<td style="text-align:center;">
138
</td>
<td style="text-align:center;">
39
</td>
<td style="text-align:center;">
16
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano2019
</td>
<td style="text-align:center;">
22
</td>
<td style="text-align:center;">
39
</td>
<td style="text-align:center;">
133
</td>
<td style="text-align:center;">
29
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano2020
</td>
<td style="text-align:center;">
14
</td>
<td style="text-align:center;">
16
</td>
<td style="text-align:center;">
29
</td>
<td style="text-align:center;">
138
</td>
</tr>
</tbody>
</table>
<p>Observe that some clones were evaluated in only one year. The number
of clones evaluated across different years is summarized below:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>pheno <span class="sc">%&gt;%</span></span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>  <span class="fu">count</span>(Ano, Clone) <span class="sc">%&gt;%</span></span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>  <span class="fu">count</span>(Clone) <span class="sc">%&gt;%</span></span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>  <span class="fu">count</span>(n) <span class="sc">%&gt;%</span></span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>, <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;N of Environment&quot;</span>, <span class="st">&quot;N of Genotypes&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<pre><code>Storing counts in `nn`, as `n` already present in input
ℹ Use `name = &quot;new_name&quot;` to pick a new name.</code></pre>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
N of Environment
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
N of Genotypes
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
1
</td>
<td style="text-align:center;">
350
</td>
</tr>
<tr>
<td style="text-align:center;">
2
</td>
<td style="text-align:center;">
72
</td>
</tr>
<tr>
<td style="text-align:center;">
3
</td>
<td style="text-align:center;">
20
</td>
</tr>
<tr>
<td style="text-align:center;">
4
</td>
<td style="text-align:center;">
5
</td>
</tr>
</tbody>
</table>
<p>Only 5 clones were evaluated in all years, which might affect the
model’s accuracy. Therefore, adopting mixed models via REML in the
analysis is suitable for obtaining BLUPs.</p>
</div>
<div id="analysis-of-traits" class="section level3">
<h3>Analysis of traits</h3>
<p>Provide descriptive statistics for each trait:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" tabindex="-1"></a>summary_table <span class="ot">&lt;-</span> <span class="fu">summary</span>(pheno) <span class="sc">%&gt;%</span></span>
<span id="cb11-2"><a href="#cb11-2" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb11-3"><a href="#cb11-3" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb11-4"><a href="#cb11-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span>
<span id="cb11-5"><a href="#cb11-5" tabindex="-1"></a>summary_table</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Clone
</td>
<td style="text-align:center;">
BGM-0164: 16
</td>
<td style="text-align:center;">
BGM-0170: 16
</td>
<td style="text-align:center;">
BGM-0396: 16
</td>
<td style="text-align:center;">
BGM-1243: 16
</td>
<td style="text-align:center;">
BGM-1267: 16
</td>
<td style="text-align:center;">
BGM-0134: 12
</td>
<td style="text-align:center;">
(Other) :2204
</td>
</tr>
<tr>
<td style="text-align:left;">
Ano
</td>
<td style="text-align:center;">
2017:660
</td>
<td style="text-align:center;">
2018:552
</td>
<td style="text-align:center;">
2019:532
</td>
<td style="text-align:center;">
2020:552
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Bloco
</td>
<td style="text-align:center;">
1:574
</td>
<td style="text-align:center;">
2:574
</td>
<td style="text-align:center;">
3:574
</td>
<td style="text-align:center;">
4:574
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
row
</td>
<td style="text-align:center;">
11 : 121
</td>
<td style="text-align:center;">
3 : 120
</td>
<td style="text-align:center;">
10 : 120
</td>
<td style="text-align:center;">
2 : 119
</td>
<td style="text-align:center;">
6 : 118
</td>
<td style="text-align:center;">
4 : 117
</td>
<td style="text-align:center;">
(Other):1581
</td>
</tr>
<tr>
<td style="text-align:left;">
col
</td>
<td style="text-align:center;">
6 : 131
</td>
<td style="text-align:center;">
4 : 130
</td>
<td style="text-align:center;">
8 : 130
</td>
<td style="text-align:center;">
2 : 128
</td>
<td style="text-align:center;">
5 : 128
</td>
<td style="text-align:center;">
9 : 128
</td>
<td style="text-align:center;">
(Other):1521
</td>
</tr>
<tr>
<td style="text-align:left;">
N_Roots
</td>
<td style="text-align:center;">
Min. : 0.125
</td>
<td style="text-align:center;">
1st Qu.: 2.333
</td>
<td style="text-align:center;">
Median : 4.000
</td>
<td style="text-align:center;">
Mean : 4.293
</td>
<td style="text-align:center;">
3rd Qu.: 6.000
</td>
<td style="text-align:center;">
Max. :15.667
</td>
<td style="text-align:center;">
NA’s :322
</td>
</tr>
<tr>
<td style="text-align:left;">
FRY
</td>
<td style="text-align:center;">
Min. : 0.116
</td>
<td style="text-align:center;">
1st Qu.: 1.857
</td>
<td style="text-align:center;">
Median : 3.750
</td>
<td style="text-align:center;">
Mean : 4.946
</td>
<td style="text-align:center;">
3rd Qu.: 6.857
</td>
<td style="text-align:center;">
Max. :22.200
</td>
<td style="text-align:center;">
NA’s :335
</td>
</tr>
<tr>
<td style="text-align:left;">
ShY
</td>
<td style="text-align:center;">
Min. : 0.694
</td>
<td style="text-align:center;">
1st Qu.: 6.944
</td>
<td style="text-align:center;">
Median :11.167
</td>
<td style="text-align:center;">
Mean :14.228
</td>
<td style="text-align:center;">
3rd Qu.:18.714
</td>
<td style="text-align:center;">
Max. :61.167
</td>
<td style="text-align:center;">
NA’s :211
</td>
</tr>
<tr>
<td style="text-align:left;">
DMC
</td>
<td style="text-align:center;">
Min. :11.98
</td>
<td style="text-align:center;">
1st Qu.:25.00
</td>
<td style="text-align:center;">
Median :28.80
</td>
<td style="text-align:center;">
Mean :29.06
</td>
<td style="text-align:center;">
3rd Qu.:32.78
</td>
<td style="text-align:center;">
Max. :48.34
</td>
<td style="text-align:center;">
NA’s :968
</td>
</tr>
<tr>
<td style="text-align:left;">
StY
</td>
<td style="text-align:center;">
Min. :0.0154
</td>
<td style="text-align:center;">
1st Qu.:0.5494
</td>
<td style="text-align:center;">
Median :1.2068
</td>
<td style="text-align:center;">
Mean :1.5164
</td>
<td style="text-align:center;">
3rd Qu.:2.1092
</td>
<td style="text-align:center;">
Max. :8.8669
</td>
<td style="text-align:center;">
NA’s :978
</td>
</tr>
<tr>
<td style="text-align:left;">
Plant.Height
</td>
<td style="text-align:center;">
Min. :0.3600
</td>
<td style="text-align:center;">
1st Qu.:0.9633
</td>
<td style="text-align:center;">
Median :1.1600
</td>
<td style="text-align:center;">
Mean :1.1919
</td>
<td style="text-align:center;">
3rd Qu.:1.3908
</td>
<td style="text-align:center;">
Max. :3.0333
</td>
<td style="text-align:center;">
NA’s :220
</td>
</tr>
<tr>
<td style="text-align:left;">
HI
</td>
<td style="text-align:center;">
Min. : 1.574
</td>
<td style="text-align:center;">
1st Qu.:15.726
</td>
<td style="text-align:center;">
Median :23.345
</td>
<td style="text-align:center;">
Mean :24.556
</td>
<td style="text-align:center;">
3rd Qu.:31.884
</td>
<td style="text-align:center;">
Max. :71.967
</td>
<td style="text-align:center;">
NA’s :341
</td>
</tr>
<tr>
<td style="text-align:left;">
StC
</td>
<td style="text-align:center;">
Min. : 7.326
</td>
<td style="text-align:center;">
1st Qu.:20.350
</td>
<td style="text-align:center;">
Median :24.147
</td>
<td style="text-align:center;">
Mean :24.419
</td>
<td style="text-align:center;">
3rd Qu.:28.160
</td>
<td style="text-align:center;">
Max. :43.686
</td>
<td style="text-align:center;">
NA’s :968
</td>
</tr>
<tr>
<td style="text-align:left;">
PltArc
</td>
<td style="text-align:center;">
Min. :1.000
</td>
<td style="text-align:center;">
1st Qu.:1.000
</td>
<td style="text-align:center;">
Median :2.000
</td>
<td style="text-align:center;">
Mean :1.996
</td>
<td style="text-align:center;">
3rd Qu.:3.000
</td>
<td style="text-align:center;">
Max. :5.000
</td>
<td style="text-align:center;">
NA’s :678
</td>
</tr>
<tr>
<td style="text-align:left;">
Leaf.Ret
</td>
<td style="text-align:center;">
Min. :0.3333
</td>
<td style="text-align:center;">
1st Qu.:1.0000
</td>
<td style="text-align:center;">
Median :1.3333
</td>
<td style="text-align:center;">
Mean :1.7796
</td>
<td style="text-align:center;">
3rd Qu.:2.0000
</td>
<td style="text-align:center;">
Max. :5.0000
</td>
<td style="text-align:center;">
NA’s :216
</td>
</tr>
<tr>
<td style="text-align:left;">
Root.Le
</td>
<td style="text-align:center;">
Min. : 7.00
</td>
<td style="text-align:center;">
1st Qu.:19.00
</td>
<td style="text-align:center;">
Median :23.00
</td>
<td style="text-align:center;">
Mean :23.21
</td>
<td style="text-align:center;">
3rd Qu.:27.00
</td>
<td style="text-align:center;">
Max. :47.33
</td>
<td style="text-align:center;">
NA’s :339
</td>
</tr>
<tr>
<td style="text-align:left;">
Root.Di
</td>
<td style="text-align:center;">
Min. : 6.12
</td>
<td style="text-align:center;">
1st Qu.:23.36
</td>
<td style="text-align:center;">
Median :28.17
</td>
<td style="text-align:center;">
Mean :28.88
</td>
<td style="text-align:center;">
3rd Qu.:33.50
</td>
<td style="text-align:center;">
Max. :63.30
</td>
<td style="text-align:center;">
NA’s :339
</td>
</tr>
<tr>
<td style="text-align:left;">
Stem.D
</td>
<td style="text-align:center;">
Min. :1.013
</td>
<td style="text-align:center;">
1st Qu.:1.859
</td>
<td style="text-align:center;">
Median :2.101
</td>
<td style="text-align:center;">
Mean :2.112
</td>
<td style="text-align:center;">
3rd Qu.:2.362
</td>
<td style="text-align:center;">
Max. :4.373
</td>
<td style="text-align:center;">
NA’s :224
</td>
</tr>
<tr>
<td style="text-align:left;">
Mite
</td>
<td style="text-align:center;">
Min. :1.000
</td>
<td style="text-align:center;">
1st Qu.:3.000
</td>
<td style="text-align:center;">
Median :3.667
</td>
<td style="text-align:center;">
Mean :3.475
</td>
<td style="text-align:center;">
3rd Qu.:4.000
</td>
<td style="text-align:center;">
Max. :5.000
</td>
<td style="text-align:center;">
NA’s :222
</td>
</tr>
<tr>
<td style="text-align:left;">
Incidence_Mites
</td>
<td style="text-align:center;">
Min. :1
</td>
<td style="text-align:center;">
1st Qu.:1
</td>
<td style="text-align:center;">
Median :1
</td>
<td style="text-align:center;">
Mean :1
</td>
<td style="text-align:center;">
3rd Qu.:1
</td>
<td style="text-align:center;">
Max. :1
</td>
<td style="text-align:center;">
NA’s :674
</td>
</tr>
<tr>
<td style="text-align:left;">
Nstem.Plant
</td>
<td style="text-align:center;">
Min. :1.000
</td>
<td style="text-align:center;">
1st Qu.:1.333
</td>
<td style="text-align:center;">
Median :2.000
</td>
<td style="text-align:center;">
Mean :2.131
</td>
<td style="text-align:center;">
3rd Qu.:2.667
</td>
<td style="text-align:center;">
Max. :6.667
</td>
<td style="text-align:center;">
NA’s :850
</td>
</tr>
<tr>
<td style="text-align:left;">
Stand6MAP
</td>
<td style="text-align:center;">
Min. :1.000
</td>
<td style="text-align:center;">
1st Qu.:4.000
</td>
<td style="text-align:center;">
Median :6.000
</td>
<td style="text-align:center;">
Mean :5.202
</td>
<td style="text-align:center;">
3rd Qu.:7.000
</td>
<td style="text-align:center;">
Max. :8.000
</td>
<td style="text-align:center;">
NA’s :849
</td>
</tr>
<tr>
<td style="text-align:left;">
Branching
</td>
<td style="text-align:center;">
Min. :0.0000
</td>
<td style="text-align:center;">
1st Qu.:0.0000
</td>
<td style="text-align:center;">
Median :0.3333
</td>
<td style="text-align:center;">
Mean :0.5911
</td>
<td style="text-align:center;">
3rd Qu.:1.0000
</td>
<td style="text-align:center;">
Max. :4.0000
</td>
<td style="text-align:center;">
NA’s :673
</td>
</tr>
<tr>
<td style="text-align:left;">
Staygreen
</td>
<td style="text-align:center;">
Min. :1.000
</td>
<td style="text-align:center;">
1st Qu.:1.000
</td>
<td style="text-align:center;">
Median :1.000
</td>
<td style="text-align:center;">
Mean :1.292
</td>
<td style="text-align:center;">
3rd Qu.:2.000
</td>
<td style="text-align:center;">
Max. :3.000
</td>
<td style="text-align:center;">
NA’s :669
</td>
</tr>
<tr>
<td style="text-align:left;">
Vigor
</td>
<td style="text-align:center;">
Mode:logical
</td>
<td style="text-align:center;">
TRUE:1075
</td>
<td style="text-align:center;">
NA’s:1221
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:left;">
Flowering
</td>
<td style="text-align:center;">
Min. :0.0000
</td>
<td style="text-align:center;">
1st Qu.:0.0000
</td>
<td style="text-align:center;">
Median :0.0000
</td>
<td style="text-align:center;">
Mean :0.0074
</td>
<td style="text-align:center;">
3rd Qu.:0.0000
</td>
<td style="text-align:center;">
Max. :1.0000
</td>
<td style="text-align:center;">
NA’s :673
</td>
</tr>
<tr>
<td style="text-align:left;">
Canopy_Lenght
</td>
<td style="text-align:center;">
Min. : 10.00
</td>
<td style="text-align:center;">
1st Qu.: 54.33
</td>
<td style="text-align:center;">
Median : 72.00
</td>
<td style="text-align:center;">
Mean : 73.35
</td>
<td style="text-align:center;">
3rd Qu.: 91.67
</td>
<td style="text-align:center;">
Max. :240.17
</td>
<td style="text-align:center;">
NA’s :1458
</td>
</tr>
<tr>
<td style="text-align:left;">
Canopy_Width
</td>
<td style="text-align:center;">
Min. : 7.50
</td>
<td style="text-align:center;">
1st Qu.: 56.33
</td>
<td style="text-align:center;">
Median : 75.58
</td>
<td style="text-align:center;">
Mean : 76.68
</td>
<td style="text-align:center;">
3rd Qu.: 96.00
</td>
<td style="text-align:center;">
Max. :213.33
</td>
<td style="text-align:center;">
NA’s :1458
</td>
</tr>
<tr>
<td style="text-align:left;">
Leaf_Lenght
</td>
<td style="text-align:center;">
Min. : 3.750
</td>
<td style="text-align:center;">
1st Qu.: 8.417
</td>
<td style="text-align:center;">
Median :11.250
</td>
<td style="text-align:center;">
Mean :11.676
</td>
<td style="text-align:center;">
3rd Qu.:14.200
</td>
<td style="text-align:center;">
Max. :32.667
</td>
<td style="text-align:center;">
NA’s :1459
</td>
</tr>
</tbody>
</table>
<p>Exclude traits with high missing value ratios:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb12-2"><a href="#cb12-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Incidence_Mites, Vigor, Flowering, Leaf_Lenght, Canopy_Width, Canopy_Lenght))</span></code></pre></div>
<p>Ensure that the traits have acceptable missing value ratios:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" tabindex="-1"></a><span class="fu">plot_missing</span>(pheno)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/check-missing-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-check-missing-1">
Past versions of check-missing-1.png
</button>
</p>
<div id="fig-check-missing-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/docs/figure/phenotype.Rmd/check-missing-1.png" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Evaluate the distribution of traits by year with histograms for
quantitative traits:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" tabindex="-1"></a><span class="fu">plot_histogram</span>(pheno, <span class="at">ncol =</span> <span class="dv">6</span>)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/histograms-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-histograms-1">
Past versions of histograms-1.png
</button>
</p>
<div id="fig-histograms-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/docs/figure/phenotype.Rmd/histograms-1.png" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Remove traits that lack normal distribution:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb15-2"><a href="#cb15-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(Branching, Leaf.Ret, Mite, PltArc, Stand6MAP, Staygreen))</span></code></pre></div>
</div>
<div id="analisys-of-clone" class="section level3">
<h3>Analisys of Clone</h3>
<p>Inspect missing values by clone and year:</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" tabindex="-1"></a>pheno_missing_summary <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb16-2"><a href="#cb16-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>Bloco, <span class="sc">-</span>row, <span class="sc">-</span>col) <span class="sc">%&gt;%</span></span>
<span id="cb16-3"><a href="#cb16-3" tabindex="-1"></a>  <span class="fu">group_by</span>(Clone, Ano) <span class="sc">%&gt;%</span></span>
<span id="cb16-4"><a href="#cb16-4" tabindex="-1"></a>  <span class="fu">summarise_all</span>(<span class="at">.funs =</span> <span class="fu">list</span>(<span class="sc">~</span> <span class="fu">sum</span>(<span class="fu">is.na</span>(.)))) <span class="sc">%&gt;%</span></span>
<span id="cb16-5"><a href="#cb16-5" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-6"><a href="#cb16-6" tabindex="-1"></a>  <span class="fu">select_numeric_cols</span>() <span class="sc">%&gt;%</span></span>
<span id="cb16-7"><a href="#cb16-7" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean =</span> <span class="fu">rowMeans</span>(.),</span>
<span id="cb16-8"><a href="#cb16-8" tabindex="-1"></a>         <span class="at">Clone.Ano =</span> <span class="fu">factor</span>(<span class="fu">unique</span>(<span class="fu">interaction</span>(pheno<span class="sc">$</span>Clone, pheno<span class="sc">$</span>Ano)))) <span class="sc">%&gt;%</span></span>
<span id="cb16-9"><a href="#cb16-9" tabindex="-1"></a>  <span class="fu">filter</span>(mean <span class="sc">&gt;</span> <span class="dv">2</span>) <span class="sc">%&gt;%</span></span>
<span id="cb16-10"><a href="#cb16-10" tabindex="-1"></a>  <span class="fu">droplevels</span>()</span>
<span id="cb16-11"><a href="#cb16-11" tabindex="-1"></a></span>
<span id="cb16-12"><a href="#cb16-12" tabindex="-1"></a>missing_genotypes <span class="ot">&lt;-</span> <span class="fu">nlevels</span>(pheno_missing_summary<span class="sc">$</span>Clone.Ano) <span class="sc">%&gt;%</span></span>
<span id="cb16-13"><a href="#cb16-13" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>, <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">&quot;N of genotypes&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb16-14"><a href="#cb16-14" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span>
<span id="cb16-15"><a href="#cb16-15" tabindex="-1"></a>missing_genotypes</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
N of genotypes
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
54
</td>
</tr>
</tbody>
</table>
<p>Evaluate clone and year descriptive statistics:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" tabindex="-1"></a>clone_year_stats <span class="ot">&lt;-</span> <span class="fu">ge_details</span>(pheno, Ano, Clone, <span class="at">resp =</span> <span class="fu">everything</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb17-2"><a href="#cb17-2" tabindex="-1"></a>  <span class="fu">t</span>() <span class="sc">%&gt;%</span></span>
<span id="cb17-3"><a href="#cb17-3" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb17-4"><a href="#cb17-4" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span>
<span id="cb17-5"><a href="#cb17-5" tabindex="-1"></a>clone_year_stats</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<tbody>
<tr>
<td style="text-align:left;">
Parameters
</td>
<td style="text-align:center;">
Mean
</td>
<td style="text-align:center;">
SE
</td>
<td style="text-align:center;">
SD
</td>
<td style="text-align:center;">
CV
</td>
<td style="text-align:center;">
Min
</td>
<td style="text-align:center;">
Max
</td>
<td style="text-align:center;">
MinENV
</td>
<td style="text-align:center;">
MaxENV
</td>
<td style="text-align:center;">
MinGEN
</td>
<td style="text-align:center;">
MaxGEN
</td>
</tr>
<tr>
<td style="text-align:left;">
N_Roots
</td>
<td style="text-align:center;">
4.29
</td>
<td style="text-align:center;">
0.06
</td>
<td style="text-align:center;">
2.51
</td>
<td style="text-align:center;">
58.56
</td>
<td style="text-align:center;">
0.12 (BGM-1031 in 2017)
</td>
<td style="text-align:center;">
15.67 (2012-107-002 in 2019)
</td>
<td style="text-align:center;">
2018 (1.62)
</td>
<td style="text-align:center;">
2019 (5.76)
</td>
<td style="text-align:center;">
BGM-0411 (0.33)
</td>
<td style="text-align:center;">
2012-107-002 (11.33)
</td>
</tr>
<tr>
<td style="text-align:left;">
FRY
</td>
<td style="text-align:center;">
4.95
</td>
<td style="text-align:center;">
0.09
</td>
<td style="text-align:center;">
4.04
</td>
<td style="text-align:center;">
81.79
</td>
<td style="text-align:center;">
0.12 (BGM-0886 in 2017)
</td>
<td style="text-align:center;">
22.2 (BGM-1267 in 2018)
</td>
<td style="text-align:center;">
2017 (2.75)
</td>
<td style="text-align:center;">
2020 (6.52)
</td>
<td style="text-align:center;">
BGM-1488 (0.34)
</td>
<td style="text-align:center;">
IAC-14 (14.07)
</td>
</tr>
<tr>
<td style="text-align:left;">
ShY
</td>
<td style="text-align:center;">
14.23
</td>
<td style="text-align:center;">
0.22
</td>
<td style="text-align:center;">
10.16
</td>
<td style="text-align:center;">
71.45
</td>
<td style="text-align:center;">
0.69 (BGM-0996 in 2017)
</td>
<td style="text-align:center;">
61.17 (BGM-2124 in 2020)
</td>
<td style="text-align:center;">
2017 (8.47)
</td>
<td style="text-align:center;">
2020 (25.87)
</td>
<td style="text-align:center;">
BGM-0048 (1.52)
</td>
<td style="text-align:center;">
BGM-2124 (54.33)
</td>
</tr>
<tr>
<td style="text-align:left;">
DMC
</td>
<td style="text-align:center;">
29.06
</td>
<td style="text-align:center;">
0.17
</td>
<td style="text-align:center;">
6.1
</td>
<td style="text-align:center;">
21
</td>
<td style="text-align:center;">
11.98 (BGM-0626 in 2020)
</td>
<td style="text-align:center;">
48.34 (BGM-1015 in 2020)
</td>
<td style="text-align:center;">
2020 (26.21)
</td>
<td style="text-align:center;">
2018 (35.38)
</td>
<td style="text-align:center;">
BGM-0626 (14.98)
</td>
<td style="text-align:center;">
BGM-1015 (45.02)
</td>
</tr>
<tr>
<td style="text-align:left;">
StY
</td>
<td style="text-align:center;">
1.52
</td>
<td style="text-align:center;">
0.04
</td>
<td style="text-align:center;">
1.27
</td>
<td style="text-align:center;">
84.01
</td>
<td style="text-align:center;">
0.02 (BGM-0340 in 2019)
</td>
<td style="text-align:center;">
8.87 (BGM-0396 in 2018)
</td>
<td style="text-align:center;">
2019 (1.32)
</td>
<td style="text-align:center;">
2018 (1.84)
</td>
<td style="text-align:center;">
BGM-0089 (0.06)
</td>
<td style="text-align:center;">
BGM-1023 (4.76)
</td>
</tr>
<tr>
<td style="text-align:left;">
Plant.Height
</td>
<td style="text-align:center;">
1.19
</td>
<td style="text-align:center;">
0.01
</td>
<td style="text-align:center;">
0.33
</td>
<td style="text-align:center;">
27.43
</td>
<td style="text-align:center;">
0.36 (BGM-0426 in 2020)
</td>
<td style="text-align:center;">
3.03 (BR-11-24-156 in 2020)
</td>
<td style="text-align:center;">
2017 (1)
</td>
<td style="text-align:center;">
2019 (1.48)
</td>
<td style="text-align:center;">
Jatobá (0.58)
</td>
<td style="text-align:center;">
BGM-1200 (1.91)
</td>
</tr>
<tr>
<td style="text-align:left;">
HI
</td>
<td style="text-align:center;">
24.56
</td>
<td style="text-align:center;">
0.27
</td>
<td style="text-align:center;">
11.89
</td>
<td style="text-align:center;">
48.42
</td>
<td style="text-align:center;">
1.57 (BGM-1159 in 2017)
</td>
<td style="text-align:center;">
71.97 (BGM-1315 in 2018)
</td>
<td style="text-align:center;">
2020 (18.61)
</td>
<td style="text-align:center;">
2018 (32.36)
</td>
<td style="text-align:center;">
BGM-0961 (2.5)
</td>
<td style="text-align:center;">
Mata_Fome_Branca (52.78)
</td>
</tr>
<tr>
<td style="text-align:left;">
StC
</td>
<td style="text-align:center;">
24.42
</td>
<td style="text-align:center;">
0.17
</td>
<td style="text-align:center;">
6.11
</td>
<td style="text-align:center;">
25.05
</td>
<td style="text-align:center;">
7.33 (BGM-0626 in 2020)
</td>
<td style="text-align:center;">
43.69 (BGM-1015 in 2020)
</td>
<td style="text-align:center;">
2020 (21.56)
</td>
<td style="text-align:center;">
2018 (30.78)
</td>
<td style="text-align:center;">
BGM-0626 (10.33)
</td>
<td style="text-align:center;">
BGM-1015 (40.37)
</td>
</tr>
<tr>
<td style="text-align:left;">
Root.Le
</td>
<td style="text-align:center;">
23.21
</td>
<td style="text-align:center;">
0.13
</td>
<td style="text-align:center;">
5.8
</td>
<td style="text-align:center;">
24.99
</td>
<td style="text-align:center;">
7 (BGM-1574 in 2017)
</td>
<td style="text-align:center;">
47.33 (BGM-0396 in 2018)
</td>
<td style="text-align:center;">
2017 (19.72)
</td>
<td style="text-align:center;">
2019 (27.14)
</td>
<td style="text-align:center;">
Jatobá (8.5)
</td>
<td style="text-align:center;">
BGM-1956 (35.5)
</td>
</tr>
<tr>
<td style="text-align:left;">
Root.Di
</td>
<td style="text-align:center;">
28.88
</td>
<td style="text-align:center;">
0.18
</td>
<td style="text-align:center;">
7.9
</td>
<td style="text-align:center;">
27.35
</td>
<td style="text-align:center;">
6.12 (BGM-2142 in 2019)
</td>
<td style="text-align:center;">
63.3 (BRS Mulatinha in 2018)
</td>
<td style="text-align:center;">
2017 (24.49)
</td>
<td style="text-align:center;">
2018 (34.74)
</td>
<td style="text-align:center;">
BGM-0089 (12.14)
</td>
<td style="text-align:center;">
BGM-1956 (48.91)
</td>
</tr>
<tr>
<td style="text-align:left;">
Stem.D
</td>
<td style="text-align:center;">
2.11
</td>
<td style="text-align:center;">
0.01
</td>
<td style="text-align:center;">
0.38
</td>
<td style="text-align:center;">
17.9
</td>
<td style="text-align:center;">
1.01 (BGM-0592 in 2018)
</td>
<td style="text-align:center;">
4.37 (BRS Tapioqueira in 2020)
</td>
<td style="text-align:center;">
2018 (2.02)
</td>
<td style="text-align:center;">
2017 (2.16)
</td>
<td style="text-align:center;">
BGM-0048 (1.25)
</td>
<td style="text-align:center;">
BGM-1523 (2.93)
</td>
</tr>
<tr>
<td style="text-align:left;">
Nstem.Plant
</td>
<td style="text-align:center;">
2.13
</td>
<td style="text-align:center;">
0.03
</td>
<td style="text-align:center;">
0.95
</td>
<td style="text-align:center;">
44.53
</td>
<td style="text-align:center;">
1 (BGM-0036 in 2018)
</td>
<td style="text-align:center;">
6.67 (BGM-0714 in 2019)
</td>
<td style="text-align:center;">
2018 (1.44)
</td>
<td style="text-align:center;">
2019 (2.71)
</td>
<td style="text-align:center;">
BGM-0066 (1)
</td>
<td style="text-align:center;">
BGM-0451 (4.44)
</td>
</tr>
</tbody>
</table>
<p>Again, some traits were not computed for the year 2017, so we have to
eliminate that year when performing the analysis for these traits.</p>
<p>Evaluate the clone-only descriptive statistics for the traits.</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" tabindex="-1"></a>cv_stats <span class="ot">&lt;-</span> <span class="fu">desc_stat</span>(pheno, <span class="at">by =</span> Ano, <span class="at">na.rm =</span> <span class="cn">TRUE</span>, <span class="at">stats =</span> <span class="st">&quot;cv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="#cb18-2" tabindex="-1"></a>  <span class="fu">na.omit</span>() <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="#cb18-3" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(cv)) <span class="sc">%&gt;%</span></span>
<span id="cb18-4"><a href="#cb18-4" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">&quot;Ano&quot;</span>, <span class="at">values_from =</span> <span class="st">&quot;cv&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-5"><a href="#cb18-5" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-6"><a href="#cb18-6" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span>
<span id="cb18-7"><a href="#cb18-7" tabindex="-1"></a>cv_stats</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
variable
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
2018
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
2020
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
2019
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
2017
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
91.8742
</td>
<td style="text-align:center;">
81.2220
</td>
<td style="text-align:center;">
72.3243
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
86.0073
</td>
<td style="text-align:center;">
68.4455
</td>
<td style="text-align:center;">
70.0326
</td>
<td style="text-align:center;">
66.8379
</td>
</tr>
<tr>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
69.0625
</td>
<td style="text-align:center;">
42.7475
</td>
<td style="text-align:center;">
47.4901
</td>
<td style="text-align:center;">
45.4973
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
63.2913
</td>
<td style="text-align:center;">
47.0839
</td>
<td style="text-align:center;">
47.2357
</td>
<td style="text-align:center;">
48.8708
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
41.5442
</td>
<td style="text-align:center;">
43.3204
</td>
<td style="text-align:center;">
41.8572
</td>
<td style="text-align:center;">
48.2942
</td>
</tr>
<tr>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
33.3849
</td>
<td style="text-align:center;">
37.1507
</td>
<td style="text-align:center;">
37.3740
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
28.0409
</td>
<td style="text-align:center;">
24.0927
</td>
<td style="text-align:center;">
19.1125
</td>
<td style="text-align:center;">
21.0398
</td>
</tr>
<tr>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
15.6937
</td>
<td style="text-align:center;">
27.3600
</td>
<td style="text-align:center;">
16.5407
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
27.1625
</td>
<td style="text-align:center;">
18.9730
</td>
<td style="text-align:center;">
21.1089
</td>
<td style="text-align:center;">
21.5823
</td>
</tr>
<tr>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
23.3943
</td>
<td style="text-align:center;">
20.5858
</td>
<td style="text-align:center;">
24.3155
</td>
<td style="text-align:center;">
21.0039
</td>
</tr>
<tr>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
13.6086
</td>
<td style="text-align:center;">
22.5059
</td>
<td style="text-align:center;">
13.8212
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
22.3941
</td>
<td style="text-align:center;">
17.5042
</td>
<td style="text-align:center;">
15.7037
</td>
<td style="text-align:center;">
16.7774
</td>
</tr>
</tbody>
</table>
<p>Some traits were not computed for the year 2017, so we have to
eliminate that year when performing the analysis for these traits and
some traits presented hight cv, as StY and FRY.</p>
</div>
</div>
<div id="general-inspection" class="section level2">
<h2>General Inspection</h2>
<p>Identifying outliers in all non-categorical variables:</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" tabindex="-1"></a><span class="fu">inspect</span>(pheno <span class="sc">%&gt;%</span></span>
<span id="cb19-2"><a href="#cb19-2" tabindex="-1"></a>          <span class="fu">select_if</span>(<span class="sc">~</span> <span class="sc">!</span><span class="fu">is.factor</span>(.)), <span class="at">verbose =</span> <span class="cn">FALSE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-3"><a href="#cb19-3" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(Outlier)) <span class="sc">%&gt;%</span></span>
<span id="cb19-4"><a href="#cb19-4" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb19-5"><a href="#cb19-5" tabindex="-1"></a>  <span class="fu">kable_classic</span>(</span>
<span id="cb19-6"><a href="#cb19-6" tabindex="-1"></a>    <span class="st">&quot;hover&quot;</span>,</span>
<span id="cb19-7"><a href="#cb19-7" tabindex="-1"></a>    <span class="at">full_width =</span> F,</span>
<span id="cb19-8"><a href="#cb19-8" tabindex="-1"></a>    <span class="at">position =</span> <span class="st">&quot;center&quot;</span>,</span>
<span id="cb19-9"><a href="#cb19-9" tabindex="-1"></a>    <span class="at">fixed_thead =</span> T</span>
<span id="cb19-10"><a href="#cb19-10" tabindex="-1"></a>  )</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style='font-family: "Arial Narrow", "Source Sans Pro", sans-serif; width: auto !important; margin-left: auto; margin-right: auto;'>
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Variable
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Class
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Missing
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Levels
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Valid_n
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Min
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Median
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Max
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Outlier
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Text
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
ShY
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
2085
</td>
<td style="text-align:center;">
0.69
</td>
<td style="text-align:center;">
11.17
</td>
<td style="text-align:center;">
61.17
</td>
<td style="text-align:center;">
89
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
FRY
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1961
</td>
<td style="text-align:center;">
0.12
</td>
<td style="text-align:center;">
3.75
</td>
<td style="text-align:center;">
22.20
</td>
<td style="text-align:center;">
71
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
StY
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1318
</td>
<td style="text-align:center;">
0.02
</td>
<td style="text-align:center;">
1.21
</td>
<td style="text-align:center;">
8.87
</td>
<td style="text-align:center;">
49
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Root.Di
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1957
</td>
<td style="text-align:center;">
6.12
</td>
<td style="text-align:center;">
28.17
</td>
<td style="text-align:center;">
63.30
</td>
<td style="text-align:center;">
32
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Nstem.Plant
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1446
</td>
<td style="text-align:center;">
1.00
</td>
<td style="text-align:center;">
2.00
</td>
<td style="text-align:center;">
6.67
</td>
<td style="text-align:center;">
30
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Plant.Height
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
2076
</td>
<td style="text-align:center;">
0.36
</td>
<td style="text-align:center;">
1.16
</td>
<td style="text-align:center;">
3.03
</td>
<td style="text-align:center;">
23
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Stem.D
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
2072
</td>
<td style="text-align:center;">
1.01
</td>
<td style="text-align:center;">
2.10
</td>
<td style="text-align:center;">
4.37
</td>
<td style="text-align:center;">
22
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
Root.Le
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1957
</td>
<td style="text-align:center;">
7.00
</td>
<td style="text-align:center;">
23.00
</td>
<td style="text-align:center;">
47.33
</td>
<td style="text-align:center;">
19
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
N_Roots
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1974
</td>
<td style="text-align:center;">
0.12
</td>
<td style="text-align:center;">
4.00
</td>
<td style="text-align:center;">
15.67
</td>
<td style="text-align:center;">
16
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
HI
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1955
</td>
<td style="text-align:center;">
1.57
</td>
<td style="text-align:center;">
23.35
</td>
<td style="text-align:center;">
71.97
</td>
<td style="text-align:center;">
15
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
DMC
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1328
</td>
<td style="text-align:center;">
11.98
</td>
<td style="text-align:center;">
28.80
</td>
<td style="text-align:center;">
48.34
</td>
<td style="text-align:center;">
6
</td>
<td style="text-align:center;">
NA
</td>
</tr>
<tr>
<td style="text-align:center;">
StC
</td>
<td style="text-align:center;">
numeric
</td>
<td style="text-align:center;">
Yes
</td>
<td style="text-align:center;">
<ul>
<li></td>
<td style="text-align:center;">
1328
</td>
<td style="text-align:center;">
7.33
</td>
<td style="text-align:center;">
24.15
</td>
<td style="text-align:center;">
43.69
</td>
<td style="text-align:center;">
6
</td>
<td style="text-align:center;">
NA
</td>
</tr>
</tbody>
</table></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul></li>
</ul>
<p>Confirming what was previously described, most traits with high
coefficients of variation (CV) have many outliers.</p>
<p>Inspect overall data correlations and save clean data:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" tabindex="-1"></a><span class="co"># Plotting correlations</span></span>
<span id="cb20-2"><a href="#cb20-2" tabindex="-1"></a><span class="fu">corr_plot</span>(pheno, <span class="at">col.by =</span> Ano)</span></code></pre></div>
<p><img src="figure/phenotype.Rmd/correlation-cleaning-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-correlation-cleaning-1">
Past versions of correlation-cleaning-1.png
</button>
</p>
<div id="fig-correlation-cleaning-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/WevertonGomesCosta/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/blob/77e21a61ed405f0bea68834f5bc83b3594dc035a/docs/figure/phenotype.Rmd/correlation-cleaning-1.png" target="_blank">77e21a6</a>
</td>
<td>
Weverton Gomes
</td>
<td>
2025-01-13
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="climate-data" class="section level2">
<h2>Climate data</h2>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" tabindex="-1"></a><span class="co"># Load the climate data</span></span>
<span id="cb21-2"><a href="#cb21-2" tabindex="-1"></a>climate_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">&quot;data/dados__temp_umi.csv&quot;</span>, <span class="at">sep =</span> <span class="st">&quot;;&quot;</span>, <span class="at">na =</span> <span class="st">&quot;null&quot;</span>, <span class="at">dec =</span> <span class="st">&quot;,&quot;</span>)</span>
<span id="cb21-3"><a href="#cb21-3" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" tabindex="-1"></a><span class="co"># Convert the date column to date format</span></span>
<span id="cb21-5"><a href="#cb21-5" tabindex="-1"></a>climate_data<span class="sc">$</span>Data.Medicao <span class="ot">&lt;-</span> <span class="fu">dmy</span>(climate_data<span class="sc">$</span>Data.Medicao)</span>
<span id="cb21-6"><a href="#cb21-6" tabindex="-1"></a></span>
<span id="cb21-7"><a href="#cb21-7" tabindex="-1"></a><span class="co"># Extract the year and semester from the date</span></span>
<span id="cb21-8"><a href="#cb21-8" tabindex="-1"></a>climate_data <span class="ot">&lt;-</span> climate_data <span class="sc">%&gt;%</span></span>
<span id="cb21-9"><a href="#cb21-9" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Ano =</span> <span class="fu">year</span>(Data.Medicao), <span class="at">Semestre =</span> <span class="fu">ifelse</span>(<span class="fu">month</span>(Data.Medicao) <span class="sc">&lt;=</span> <span class="dv">6</span>, <span class="st">&quot;1-6&quot;</span>, <span class="st">&quot;7-12&quot;</span>))</span>
<span id="cb21-10"><a href="#cb21-10" tabindex="-1"></a></span>
<span id="cb21-11"><a href="#cb21-11" tabindex="-1"></a><span class="co"># Calculate the means per semester and year for each variable</span></span>
<span id="cb21-12"><a href="#cb21-12" tabindex="-1"></a>semester_means <span class="ot">&lt;-</span> climate_data <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" tabindex="-1"></a>  <span class="fu">group_by</span>(Ano, Semestre) <span class="sc">%&gt;%</span></span>
<span id="cb21-14"><a href="#cb21-14" tabindex="-1"></a>  <span class="fu">summarise_if</span>(is.numeric, <span class="sc">~</span><span class="fu">mean</span>(., <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb21-15"><a href="#cb21-15" tabindex="-1"></a></span>
<span id="cb21-16"><a href="#cb21-16" tabindex="-1"></a><span class="co"># Display the results</span></span>
<span id="cb21-17"><a href="#cb21-17" tabindex="-1"></a>semester_means <span class="sc">%&gt;%</span></span>
<span id="cb21-18"><a href="#cb21-18" tabindex="-1"></a>  <span class="fu">kbl</span>(<span class="at">escape =</span> F, <span class="at">align =</span> <span class="st">&#39;c&#39;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb21-19"><a href="#cb21-19" tabindex="-1"></a>  <span class="fu">kable_classic</span>(<span class="st">&quot;hover&quot;</span>, <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">&quot;center&quot;</span>, <span class="at">fixed_thead =</span> T)</span></code></pre></div>
<table class=" lightable-classic lightable-hover" style="font-family: &quot;Arial Narrow&quot;, &quot;Source Sans Pro&quot;, sans-serif; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Ano
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
Semestre
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
EVAPORACAO.DO.PICHE..DIARIA.mm.
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
INSOLACAO.TOTAL..DIARIO.h.
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
PRECIPITACAO.TOTAL..DIARIO.mm.
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
TEMPERATURA.MAXIMA..DIARIA..C.
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
TEMPERATURA.MEDIA.COMPENSADA..DIARIA..C.
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
TEMPERATURA.MINIMA..DIARIA..C.
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
UMIDADE.RELATIVA.DO.AR..MEDIA.DIARIA…
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
UMIDADE.RELATIVA.DO.AR..MINIMA.DIARIA…
</th>
<th style="text-align:center;position: sticky; top:0; background-color: #FFFFFF;">
VENTO..VELOCIDADE.MEDIA.DIARIA.m.s.
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
2016
</td>
<td style="text-align:center;">
1-6
</td>
<td style="text-align:center;">
10.607182
</td>
<td style="text-align:center;">
8.453297
</td>
<td style="text-align:center;">
1.7829670
</td>
<td style="text-align:center;">
32.94780
</td>
<td style="text-align:center;">
27.85220
</td>
<td style="text-align:center;">
23.30440
</td>
<td style="text-align:center;">
65.63571
</td>
<td style="text-align:center;">
55.33516
</td>
<td style="text-align:center;">
2.703846
</td>
</tr>
<tr>
<td style="text-align:center;">
2016
</td>
<td style="text-align:center;">
7-12
</td>
<td style="text-align:center;">
13.772826
</td>
<td style="text-align:center;">
9.549456
</td>
<td style="text-align:center;">
0.1625000
</td>
<td style="text-align:center;">
33.72609
</td>
<td style="text-align:center;">
28.05217
</td>
<td style="text-align:center;">
22.64511
</td>
<td style="text-align:center;">
58.46141
</td>
<td style="text-align:center;">
47.90217
</td>
<td style="text-align:center;">
3.155978
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
1-6
</td>
<td style="text-align:center;">
12.409945
</td>
<td style="text-align:center;">
8.513260
</td>
<td style="text-align:center;">
0.5801105
</td>
<td style="text-align:center;">
33.51271
</td>
<td style="text-align:center;">
28.48122
</td>
<td style="text-align:center;">
23.77845
</td>
<td style="text-align:center;">
60.43833
</td>
<td style="text-align:center;">
50.20994
</td>
<td style="text-align:center;">
2.971271
</td>
</tr>
<tr>
<td style="text-align:center;">
2017
</td>
<td style="text-align:center;">
7-12
</td>
<td style="text-align:center;">
13.032065
</td>
<td style="text-align:center;">
8.686957
</td>
<td style="text-align:center;">
0.2304348
</td>
<td style="text-align:center;">
32.30109
</td>
<td style="text-align:center;">
26.99239
</td>
<td style="text-align:center;">
22.08533
</td>
<td style="text-align:center;">
55.83770
</td>
<td style="text-align:center;">
44.88587
</td>
<td style="text-align:center;">
3.279891
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
1-6
</td>
<td style="text-align:center;">
9.982873
</td>
<td style="text-align:center;">
7.917680
</td>
<td style="text-align:center;">
1.4602210
</td>
<td style="text-align:center;">
32.39779
</td>
<td style="text-align:center;">
27.39116
</td>
<td style="text-align:center;">
23.11713
</td>
<td style="text-align:center;">
62.37821
</td>
<td style="text-align:center;">
50.77348
</td>
<td style="text-align:center;">
2.782320
</td>
</tr>
<tr>
<td style="text-align:center;">
2018
</td>
<td style="text-align:center;">
7-12
</td>
<td style="text-align:center;">
12.984239
</td>
<td style="text-align:center;">
9.203261
</td>
<td style="text-align:center;">
0.3222826
</td>
<td style="text-align:center;">
33.13152
</td>
<td style="text-align:center;">
27.72283
</td>
<td style="text-align:center;">
22.60054
</td>
<td style="text-align:center;">
54.03859
</td>
<td style="text-align:center;">
43.55978
</td>
<td style="text-align:center;">
3.023913
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
1-6
</td>
<td style="text-align:center;">
10.839326
</td>
<td style="text-align:center;">
8.501695
</td>
<td style="text-align:center;">
1.1387640
</td>
<td style="text-align:center;">
33.57175
</td>
<td style="text-align:center;">
28.31243
</td>
<td style="text-align:center;">
23.88596
</td>
<td style="text-align:center;">
61.06723
</td>
<td style="text-align:center;">
48.66298
</td>
<td style="text-align:center;">
2.758989
</td>
</tr>
<tr>
<td style="text-align:center;">
2019
</td>
<td style="text-align:center;">
7-12
</td>
<td style="text-align:center;">
13.895652
</td>
<td style="text-align:center;">
9.229891
</td>
<td style="text-align:center;">
0.1972826
</td>
<td style="text-align:center;">
33.72391
</td>
<td style="text-align:center;">
27.91413
</td>
<td style="text-align:center;">
22.59565
</td>
<td style="text-align:center;">
53.56667
</td>
<td style="text-align:center;">
43.01087
</td>
<td style="text-align:center;">
3.244565
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
1-6
</td>
<td style="text-align:center;">
8.186517
</td>
<td style="text-align:center;">
6.869101
</td>
<td style="text-align:center;">
1.9252809
</td>
<td style="text-align:center;">
32.20337
</td>
<td style="text-align:center;">
27.24157
</td>
<td style="text-align:center;">
23.48090
</td>
<td style="text-align:center;">
70.01461
</td>
<td style="text-align:center;">
59.37234
</td>
<td style="text-align:center;">
2.555056
</td>
</tr>
<tr>
<td style="text-align:center;">
2020
</td>
<td style="text-align:center;">
7-12
</td>
<td style="text-align:center;">
12.382065
</td>
<td style="text-align:center;">
8.848913
</td>
<td style="text-align:center;">
0.9619565
</td>
<td style="text-align:center;">
32.79946
</td>
<td style="text-align:center;">
27.07017
</td>
<td style="text-align:center;">
21.95217
</td>
<td style="text-align:center;">
59.94088
</td>
<td style="text-align:center;">
47.12500
</td>
<td style="text-align:center;">
3.020109
</td>
</tr>
</tbody>
</table>
<p>Then, now we go to execute the mixed models analisys script: <a
href="https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/mixed_models">mixed_models.Rmd</a></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" tabindex="-1"></a><span class="fu">sessionInfo</span>()</span></code></pre></div>
<pre><code>R version 4.3.1 (2023-06-16 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=Portuguese_Brazil.utf8  LC_CTYPE=Portuguese_Brazil.utf8   
[3] LC_MONETARY=Portuguese_Brazil.utf8 LC_NUMERIC=C                      
[5] LC_TIME=Portuguese_Brazil.utf8    

time zone: America/Sao_Paulo
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] GGally_2.1.2          ggthemes_4.2.4        DataExplorer_0.8.2   
 [4] metan_1.18.0          readxl_1.4.3          data.table_1.14.8    
 [7] ComplexHeatmap_2.16.0 lubridate_1.9.2       forcats_1.0.0        
[10] stringr_1.5.0         dplyr_1.1.2           purrr_1.0.2          
[13] readr_2.1.4           tidyr_1.3.0           tibble_3.2.1         
[16] ggplot2_3.4.3         tidyverse_2.0.0       kableExtra_1.3.4     

loaded via a namespace (and not attached):
  [1] gridExtra_2.3       writexl_1.4.2       rlang_1.1.1        
  [4] magrittr_2.0.3      clue_0.3-64         GetoptLong_1.0.5   
  [7] git2r_0.32.0        matrixStats_1.0.0   compiler_4.3.1     
 [10] png_0.1-8           systemfonts_1.0.4   vctrs_0.6.3        
 [13] rvest_1.0.3         pkgconfig_2.0.3     shape_1.4.6        
 [16] crayon_1.5.2        fastmap_1.1.1       labeling_0.4.2     
 [19] utf8_1.2.3          promises_1.2.1      rmarkdown_2.24     
 [22] tzdb_0.4.0          nloptr_2.0.3        xfun_0.40          
 [25] cachem_1.0.8        jsonlite_1.8.7      highr_0.10         
 [28] later_1.3.1         reshape_0.8.9       tweenr_2.0.2       
 [31] parallel_4.3.1      cluster_2.1.4       R6_2.5.1           
 [34] bslib_0.5.1         stringi_1.7.12      RColorBrewer_1.1-3 
 [37] boot_1.3-28.1       numDeriv_2016.8-1.1 jquerylib_0.1.4    
 [40] cellranger_1.1.0    Rcpp_1.0.11         iterators_1.0.14   
 [43] knitr_1.43          IRanges_2.34.1      igraph_1.5.1       
 [46] Matrix_1.6-1        httpuv_1.6.11       splines_4.3.1      
 [49] timechange_0.2.0    tidyselect_1.2.0    rstudioapi_0.15.0  
 [52] yaml_2.3.7          doParallel_1.0.17   codetools_0.2-19   
 [55] lmerTest_3.1-3      lattice_0.21-8      plyr_1.8.8         
 [58] withr_2.5.2         evaluate_0.22       polyclip_1.10-4    
 [61] xml2_1.3.5          circlize_0.4.15     pillar_1.9.0       
 [64] whisker_0.4.1       foreach_1.5.2       stats4_4.3.1       
 [67] generics_0.1.3      rprojroot_2.0.3     mathjaxr_1.6-0     
 [70] S4Vectors_0.38.1    hms_1.1.3           munsell_0.5.0      
 [73] scales_1.2.1        minqa_1.2.6         glue_1.6.2         
 [76] tools_4.3.1         lme4_1.1-34         webshot_0.5.5      
 [79] fs_1.6.3            colorspace_2.1-0    networkD3_0.4      
 [82] nlme_3.1-163        patchwork_1.1.3     ggforce_0.4.1      
 [85] cli_3.6.1           workflowr_1.7.1     fansi_1.0.4        
 [88] viridisLite_0.4.2   svglite_2.1.1       gtable_0.3.4       
 [91] sass_0.4.7          digest_0.6.33       BiocGenerics_0.46.0
 [94] ggrepel_0.9.3       htmlwidgets_1.6.2   rjson_0.2.21       
 [97] farver_2.1.1        htmltools_0.5.6     lifecycle_1.0.3    
[100] httr_1.4.7          GlobalOptions_0.1.2 MASS_7.3-60        </code></pre>
</div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr />
<ol>
<li id="fn1"><p>Weverton Gomes da Costa, Pós-Doutorando, Embrapa
Mandioca e Fruticultura, <a href="mailto:wevertonufv@gmail.com"
class="email">wevertonufv@gmail.com</a><a href="#fnref1"
class="footnote-back">↩︎</a></p></li>
</ol>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
