---
title: "Genomic Selection - Prepare Data"
author: 
  - Costa, W. G.^[Weverton Gomes da Costa, PÃ³s-Doutorando, Embrapa Mandioca e Fruticultura, wevertonufv@gmail.com]
date: "`r Sys.Date()`"
site: workflowr::wflow_site
url: https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava
output:
  workflowr::wflow_html:
    toc: TRUE
editor_options:
  chunk_output_type: console
---

## Configurations and packages

```{r setup, include=F} 
knitr::opts_chunk$set(echo = T, warning = F)
```

To perform the analyses, we will need the following packages:

```{r libraries, message=FALSE}
library(readxl)
library(tidyverse)
library(kableExtra)
#devtools::install_github("wolfemd/genomicMateSelectR", ref = 'master') 
library(genomicMateSelectR)
library(AGHmatrix)
library(ComplexHeatmap)
```

<br>

## Data

The data set is based in genotypes evalueted in five years (2016 to 2020), each year was considered as environment. 

### Names marker data

First let's get the marker IDs for each clone.

```{r import-data-names}
names <-
  read_excel("data/Phenotyping2.xlsx", sheet = "GBS") |>
  rename(Clone = `Names trials Petrolina`, ID_Clone = `Nome GBS`) |>
  mutate(ID_Clone = str_replace_all(ID_Clone, ":", ".")) |>
  select(Clone, ID_Clone)

names |>
  head() |> 
  kbl(escape = F, align = 'c') |>
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

### Phenotypic data

Now let's group the marker IDs with the clone names.

```{r import-pheno-data}
pheno <- read.csv("output/BLUPS_row_col_random.csv") |> 
  mutate(Clone = str_split_i(Clone, "[.]", -1)) |>
  inner_join(names) |> # Join Phenotypic with Genotypic datas
  mutate(Clone = factor(Clone), ID_Clone = factor(ID_Clone))

pheno |> 
  head() |>
  kbl(escape = F, align = 'c') |>
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)

traits <- levels(factor(pheno$trait))

```

### Genotypic data

Now let's load the genotypic data from the GBS markers and correct the base pair values. Furthermore, we will also divide the alleles column into two columns, for the reference allele and the recessive allele. And we will select the columns with the names of the markers, reference alleles and the columns with the clone IDs according to the BLUPs data.

```{r import-genotypic-data}
geno <- read.table("data/allchrAR08.txt", header = T) |>
  mutate(across(12:3365, ~{
    case_when(
      . == "A" ~ "AA",
      . == "R" ~ "AG",
      . == "W" ~ "AT",
      . == "M" ~ "AC",
      . == "C" ~ "CC",
      . == "S" ~ "CG",
      . == "Y" ~ "CT",
      . == "G" ~ "GG",
      . == "K" ~ "GT",
      . == "T" ~ "TT")
  })) |>
  separate(alleles, c("reference", "recess")) |>
  select(rs, reference, recess, any_of(pheno$ID_Clone))

geno[1:5, 1:20] |>
  kbl(escape = F, align = 'c') |>
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

Now we need to do the base pair conversion for allelic dosage according to the reference allele. I will also add the rs column as the column name. Then I will exclude the reference and recess allele columns.
To convert into the format to perform GWS analyses, we have to transpose the marker matrix.

```{r recode-genotypic-data}
geno <- geno |>
  mutate(across(4:ncol(geno), ~{
      case_when(
        . == paste(reference, reference, sep = "") ~ 2,
        . == paste(recess, recess, sep = "") ~ 0,
        TRUE ~ 1
      )
  })) |>
  select(-c(reference, recess)) |> 
  column_to_rownames(var ="rs") |> 
  t()
  
geno[1:5, 1:5] |>
  kbl(escape = F, align = 'c') |>
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

Let's check how many clones present data genotyped with the markers.

```{r geno-clones}
geno |>
  dim() |>
  t() |>
  kbl(escape = F, align = 'c',
      col.names = c("Number of Clones", "Number of markers")) |>
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

By filtering common genotypes, we have 415 genotypes and 27045 brands.

Now let's filter the SNPS using MAF of 0.01 and check how many markers will remain.

```{r filter-genotypic-data}
geno <- maf_filter(geno, thresh = 0.01)

geno |>
  dim() |>
  t() |>
  kbl(escape = F, align = 'c',
      col.names = c("Number of Clones", "Number of markers")) |>
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

With the MAF filter at 1%, 22779 markers remained. I'm going to save the matrix now so we can load it if necessary.

## Genomic selection

For this purpose, we will use only individuals with BLUps and SNPs available. 

```{r geno-pheno}
pheno <- pheno |> 
  select(ID_Clone, trait, BLUP) %>% 
  pivot_wider(names_from = trait, values_from = BLUP, id_cols = 1) %>% 
  filter(ID_Clone %in% rownames(geno)) |>
  droplevels()

traits <- colnames(pheno)[-1]

pheno <- pheno[order(pheno$ID_Clone, decreasing = F),]
geno <- geno[order(row.names(geno)),]
all(rownames(geno) == pheno$ID_Clone)
```

```{r save-load-data, include=FALSE}
saveRDS(geno, "data/geno.rds")
geno <- readRDS("data/geno.rds")
saveRDS(pheno, "data/pheno.rds")
```

### Building the G matrix

Again, we will use the [AGHmatrix](https://cran.r-project.org/web/packages/AGHmatrix/vignettes/Tutorial_AGHmatrix.html) package [@amadeu_aghmatrix_2016] to build the G matrix:

```{r G-matrix}
G_matrix <- Gmatrix(geno,
                    method = "VanRaden",
                    ploidy = 2,
                    missingValue = NA)
```

```{r save-G-matrix, include=FALSE}
saveRDS(G_matrix, "output/G_matrix.rds")
```

Now we have the whole G matrix (414 x 414), which we can represent using a heatmap:

########### Heatmap of the genomic kinship matrix between clones


```{r heatmap-G-matrix}
Heatmap(
  G_matrix,
  show_row_names = F,
  show_column_names = F,
  heatmap_legend_param = list(title = "Res")
)
```

"Res" in the heatmap legend title is for "Resemblance".

Let's now move on to processing each model. Since this can be time-consuming, I've separated it into parallel processing and each model into each script:

- [**RR-BLUP**](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RR-BLUP.html)
- [**G-BLUP (additive and additive-dominant)**](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_G-BLUP.html)
- [**Bayes A**](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_BayesA.html)
- [**Bayes B**](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_BayesB.html)
- [**RKHS**](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RKHS.html)
- [**Random Forest**](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/GWS_RF.html)
