---
title: "Phenotype Analysis of Brazilian Drought Trials"
author: 
  - Costa, W. G.^[Weverton Gomes da Costa, PÃ³s-Doutorando, Embrapa Mandioca e Fruticultura, wevertonufv@gmail.com]
date: "`r Sys.Date()`"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: TRUE
editor_options:
  chunk_output_type: console
---

```{r setup, include=FALSE} 
knitr::opts_chunk$set(
	echo = TRUE,
	warning = FALSE)
```

## Libraries

Load the necessary libraries:

```{r Libraries, message=FALSE}
library(kableExtra)
library(tidyverse)
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
#BiocManager::install("ComplexHeatmap")
require(ComplexHeatmap)
library(data.table)
library(readxl)
library(metan)
library(DataExplorer)
library(ggthemes)
library(GGally)
theme_set(theme_bw())
```

## Data import and manipulation

Import the phenotypic dataset, excluding traits without information and redundant traits:

```{r import-Data}
pheno <- read_excel("data/Phenotyping2.xlsx", na = "NA") %>%
  select_if(~ !all(is.na(.))) %>%  # Deleting traits without information
  select(-c("Local", "Tratamento"))
```

Convert character traits to factors, and numeric grades to integer factors:

```{r Process-Data}
pheno <- pheno %>%
  mutate(across(c(Clone, Ano, row, col, Bloco), as.factor))
```

## Exploratory Data Analysis

Provide an introductory analysis of the dataset:

```{r EDA-info}
introduce(pheno) %>%
  t() %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

We don't have any columns that have all of the missing observations, but we do have a lot of missing values in every dataset. Some manipulations should be performed to improve the quality of the data.

### Environment Analysis

Heatmap visualization of clone presence per year:

```{r heatmap}
pheno_year_count <- pheno %>%
  count(Ano, Clone)

genmat <- model.matrix(~ -1 + Clone, data = pheno_year_count)
envmat <- model.matrix(~ -1 + Ano, data = pheno_year_count)
genenvmat <- t(envmat) %*% genmat
genenvmat_ch <- ifelse(genenvmat == 1, "Present", "Absent")

Heatmap(genenvmat_ch, 
        col = c("white", "tomato"), 
        show_column_names = F, 
        heatmap_legend_param = list(title = ""),
        column_title = "Genotypes", 
        row_title = "Environments")
```

Filter out the year 2016 due to insufficient observations:

```{r Filter-Year}
pheno <- pheno %>%
  filter(Ano != 2016) %>%
  droplevels()
```

Reinspect the updated heatmap:

```{r reheatmap}
pheno_year_count <- pheno %>%
  count(Ano, Clone)

genmat <- model.matrix(~ -1 + Clone, data = pheno_year_count)
envmat <- model.matrix(~ -1 + Ano, data = pheno_year_count)
genenvmat <- t(envmat) %*% genmat
genenvmat_ch <- ifelse(genenvmat == 1, "Present", "Absent")

Heatmap(genenvmat_ch, 
        col = c("white", "tomato"), 
        show_column_names = F, 
        heatmap_legend_param = list(title = ""),
        column_title = "Genotypes", 
        row_title = "Environments")
```

Visualize the number of environments and genotypes:

```{r environment-genotype-table}
genotype_year_table <- genenvmat %*% t(genenvmat) %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
rm(pheno_year_count, genmat, envmat, genenvmat, genenvmat_ch)
genotype_year_table
```

Observe that some clones were evaluated in only one year. The number of clones evaluated across different years is summarized below:

```{r clone-evaluation-analysis}
pheno %>%
  count(Ano, Clone) %>%
  count(Clone) %>%
  count(n) %>%
  kbl(escape = F, align = 'c', col.names = c("N of Environment", "N of Genotypes")) %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```

Only 5 clones were evaluated in all years, which might affect the model's accuracy. Therefore, adopting mixed models via REML in the analysis is suitable for obtaining BLUPs.

### Analysis of traits

Provide descriptive statistics for each trait:

```{r trait-summary}
summary_table <- summary(pheno) %>%
  t() %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
summary_table
```

Exclude traits with high missing value ratios:

```{r filter-traits}
pheno <- pheno %>%
  select(-c(Incidence_Mites, Vigor, Flowering, Leaf_Lenght, Canopy_Width, Canopy_Lenght))
```

Ensure that the traits have acceptable missing value ratios:

```{r check-missing}
plot_missing(pheno)
```

Evaluate the distribution of traits by year with histograms for quantitative traits:

```{r histograms}
plot_histogram(pheno, ncol = 6)
```

Remove traits that lack normal distribution:

```{r filter-nonnormal-traits}
pheno <- pheno %>%
  select(-c(Branching, Leaf.Ret, Mite, PltArc, Stand6MAP, Staygreen))
```

### Analisys of Clone

Inspect missing values by clone and year:

```{r missing-values-clones}
pheno_missing_summary <- pheno %>%
  select(-Bloco, -row, -col) %>%
  group_by(Clone, Ano) %>%
  summarise_all(.funs = list(~ sum(is.na(.)))) %>%
  ungroup() %>%
  select_numeric_cols() %>%
  mutate(mean = rowMeans(.),
         Clone.Ano = factor(unique(interaction(pheno$Clone, pheno$Ano)))) %>%
  filter(mean > 2) %>%
  droplevels()

missing_genotypes <- nlevels(pheno_missing_summary$Clone.Ano) %>%
  kbl(escape = F, align = 'c', col.names = c("N of genotypes")) %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
missing_genotypes
```

Evaluate clone and year descriptive statistics:

```{r clone-year-descriptive}
clone_year_stats <- ge_details(pheno, Ano, Clone, resp = everything()) %>%
  t() %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
clone_year_stats
```

Again, some traits were not computed for the year 2017, so we have to eliminate that year when performing the analysis for these traits.

Evaluate the clone-only descriptive statistics for the traits.

```{r clone-descriptive-statistics}
cv_stats <- desc_stat(pheno, by = Ano, na.rm = TRUE, stats = "cv") %>%
  na.omit() %>%
  arrange(desc(cv)) %>%
  pivot_wider(names_from = "Ano", values_from = "cv") %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
cv_stats
```

Some traits were not computed for the year 2017, so we have to eliminate that year when performing the analysis for these traits and some traits presented hight cv, as StY and FRY.

## General Inspection

Identifying outliers in all non-categorical variables:

```{r general-inspection}
inspect(pheno %>%
          select_if(~ !is.factor(.)), verbose = FALSE) %>%
  arrange(desc(Outlier)) %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic(
    "hover",
    full_width = F,
    position = "center",
    fixed_thead = T
  )
```

Confirming what was previously described, most traits with high coefficients of variation (CV) have many outliers.

Inspect overall data correlations and save clean data:

```{r correlation-cleaning, fig.width=10, fig.height=10}
# Plotting correlations
corr_plot(pheno, col.by = Ano)
```


```{r include = FALSE}
# Saving the cleaned data
write.csv(pheno, "data/pheno_clean.csv", row.names = FALSE, quote = FALSE)

# Descriptive statistics (mean, minimum, maximum, CV)
pheno_mean_sd <- desc_stat(pheno, stats = c("mean", "min", "max", "cv"))
write.csv(pheno_mean_sd, "output/pheno_mean_sd.csv", row.names = FALSE, quote = FALSE)
```

## Climate data

```{r Climate-data}
# Load the climate data
climate_data <- read.csv("data/dados__temp_umi.csv", sep = ";", na = "null", dec = ",")

# Convert the date column to date format
climate_data$Data.Medicao <- dmy(climate_data$Data.Medicao)

# Extract the year and semester from the date
climate_data <- climate_data %>%
  mutate(Ano = year(Data.Medicao), Semestre = ifelse(month(Data.Medicao) <= 6, "1-6", "7-12"))

# Calculate the means per semester and year for each variable
semester_means <- climate_data %>%
  group_by(Ano, Semestre) %>%
  summarise_if(is.numeric, ~mean(., na.rm = TRUE))

# Display the results
semester_means %>%
  kbl(escape = F, align = 'c') %>%
  kable_classic("hover", full_width = F, position = "center", fixed_thead = T)
```


```{r include=FALSE}
# Save the results to a file
writexl::write_xlsx(semester_means, "output/semester_means.xlsx")
```

Then, now we go to execute the mixed models analisys script: [mixed_models.Rmd](https://wevertongomescosta.github.io/Genomic-Selection-for-Drought-Tolerance-Using-Genome-Wide-SNPs-in-Casava/mixed_models)
